<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { color: #00769e; background-color: #f1f3f5; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #00769e; } /* Normal */
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #657422; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #00769e; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #00769e; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #00769e; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

<style>
  div.csl-bib-body { }
  div.csl-entry {
    clear: both;
      margin-bottom: 0em;
    }
  .hanging div.csl-entry {
    margin-left:2em;
    text-indent:-2em;
  }
  div.csl-left-margin {
    min-width:2em;
    float:left;
  }
  div.csl-right-inline {
    margin-left:2em;
    padding-left:1em;
  }
  div.csl-indent {
    margin-left: 2em;
  }
</style>

  <!--radix_placeholder_meta_tags-->
  <title>RStudio Reference Guide: Useful Packages</title>




  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="RStudio Reference Guide: Useful Packages"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="RStudio Reference Guide"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="RStudio Reference Guide: Useful Packages"/>

  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","output","theme","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Useful Packages"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["toc","toc_depth"]}},"value":[{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[2]}]}]},{"type":"character","attributes":{},"value":["theme.css"]},{"type":"character","attributes":{},"value":["bib.bibtex"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
    font-size: 100%;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  hr.section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    margin: 0px;
  }


  d-byline {
    border-top: none;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
    border-top: none;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  /* tweak for Pandoc numbered line within distill */
  d-article pre.numberSource code > span {
      left: -2em;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // separator
    var separator = '<hr class="section-separator" style="clear: both"/>';
    // prepend separator above appendix
    $('.d-byline').before(separator);
    $('.d-article').before(separator);

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme, except when numbering line
    // in code chunk
    $('pre:not(.numberLines) code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        var author_name = front_matter.authors[i].author
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', author_name ? 'ORCID ID for ' + author_name : 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        const citeChild = $(this).children()[0]
        // Do not process if @xyz has been used without escaping and without bibliography activated
        // https://github.com/rstudio/distill/issues/466
        if (citeChild === undefined) return true

        if (citeChild.nodeName == "D-FOOTNOTE") {
          var fn = citeChild
          $(this).html(fn.shadowRoot.querySelector("sup"))
          $(this).id = fn.id
          fn.remove()
        }
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          // Could use CSS.escape too here, we insure backward compatibility in navigator
          return "<p>" + $('div[id="ref-' + ref + '"]').html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // fix footnotes in tables (#411)
      // replacing broken distill.pub feature
      $('table d-footnote').each(function() {
        // we replace internal showAtNode methode which is triggered when hovering a footnote
        this.hoverBox.showAtNode = function(node) {
          // ported from https://github.com/distillpub/template/pull/105/files
          calcOffset = function(elem) {
              let x = elem.offsetLeft;
              let y = elem.offsetTop;
              // Traverse upwards until an `absolute` element is found or `elem`
              // becomes null.
              while (elem = elem.offsetParent && elem.style.position != 'absolute') {
                  x += elem.offsetLeft;
                  y += elem.offsetTop;
              }

              return { left: x, top: y };
          }
          // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop
          const bbox = node.getBoundingClientRect();
          const offset = calcOffset(node);
          this.show([offset.left + bbox.width, offset.top + bbox.height]);
        }
      })

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      // ignore leaflet img layers (#106)
      figures = figures.filter(':not(img[class*="leaflet"])')
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace(/^index[.]html/, "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <style type="text/css">
  /* base variables */

  /* Edit the CSS properties in this file to create a custom
     Distill theme. Only edit values in the right column
     for each row; values shown are the CSS defaults.
     To return any property to the default,
     you may set its value to: unset
     All rows must end with a semi-colon.                      */

  /* Optional: embed custom fonts here with `@import`          */
  /* This must remain at the top of this file.                 */



  html {
    /*-- Main font sizes --*/
    --title-size:      50px;
    --body-size:       1.06rem;
    --code-size:       14px;
    --aside-size:      12px;
    --fig-cap-size:    13px;
    /*-- Main font colors --*/
    --title-color:     #000000;
    --header-color:    rgba(0, 0, 0, 0.8);
    --body-color:      rgba(0, 0, 0, 0.8);
    --aside-color:     rgba(0, 0, 0, 0.6);
    --fig-cap-color:   rgba(0, 0, 0, 0.6);
    /*-- Specify custom fonts ~~~ must be imported above   --*/
    --heading-font:    sans-serif;
    --mono-font:       monospace;
    --body-font:       sans-serif;
    --navbar-font:     sans-serif;  /* websites + blogs only */
  }

  /*-- ARTICLE METADATA --*/
  d-byline {
    --heading-size:    0.6rem;
    --heading-color:   rgba(0, 0, 0, 0.5);
    --body-size:       0.8rem;
    --body-color:      rgba(0, 0, 0, 0.8);
  }

  /*-- ARTICLE TABLE OF CONTENTS --*/
  .d-contents {
    --heading-size:    18px;
    --contents-size:   13px;
  }

  /*-- ARTICLE APPENDIX --*/
  d-appendix {
    --heading-size:    15px;
    --heading-color:   rgba(0, 0, 0, 0.65);
    --text-size:       0.8em;
    --text-color:      rgba(0, 0, 0, 0.5);
  }

  /*-- WEBSITE HEADER + FOOTER --*/
  /* These properties only apply to Distill sites and blogs  */

  .distill-site-header {
    --title-size:       18px;
    --text-color:       rgba(255, 255, 255, 0.8);
    --text-size:        15px;
    --hover-color:      white;
    --bkgd-color:       #0F2E3D;
  }

  .distill-site-footer {
    --text-color:       rgba(255, 255, 255, 0.8);
    --text-size:        15px;
    --hover-color:      white;
    --bkgd-color:       #0F2E3D;
  }

  /*-- Additional custom styles --*/
  /* Add any additional CSS rules below                      */
  </style>
  <style type="text/css">
  /* base variables */

  /* Edit the CSS properties in this file to create a custom
     Distill theme. Only edit values in the right column
     for each row; values shown are the CSS defaults.
     To return any property to the default,
     you may set its value to: unset
     All rows must end with a semi-colon.                      */

  /* Optional: embed custom fonts here with `@import`          */
  /* This must remain at the top of this file.                 */



  html {
    /*-- Main font sizes --*/
    --title-size:      50px;
    --body-size:       30px;
    --aside-size:      20px;
    --fig-cap-size:    20px;
    /*-- Main font colors --*/
    --title-color:     #000000;
    --header-color:    rgba(0, 0, 0, 0.8);
    --body-color:      rgba(0, 0, 0, 0.8);
    --aside-color:     rgba(0, 0, 0, 0.6);
    --fig-cap-color:   rgba(0, 0, 0, 0.6);
    /*-- Specify custom fonts ~~~ must be imported above   --*/
    --heading-font:    sans-serif;
    --mono-font:       monospace;
    --body-font:       sans-serif;
    --navbar-font:     sans-serif;  /* websites + blogs only */
  }

  /*-- ARTICLE METADATA --*/
  d-byline {
    --heading-size:    0.6rem;
    --heading-color:   rgba(0, 0, 0, 0.5);
    --body-size:       0.8rem;
    --body-color:      rgba(0, 0, 0, 0.8);
  }

  /*-- ARTICLE TABLE OF CONTENTS --*/
  .d-contents {
    --heading-size:    30px;
    --contents-size:   25px;
  }

  /*-- ARTICLE APPENDIX --*/
  d-appendix {
    --heading-size:    15px;
    --heading-color:   rgba(0, 0, 0, 0.65);
    --text-size:       0.8em;
    --text-color:      rgba(0, 0, 0, 0.5);
  }

  /*-- WEBSITE HEADER + FOOTER --*/
  /* These properties only apply to Distill sites and blogs  */

  .distill-site-header {
    --title-size:       25px;
    --text-color:       rgba(255, 255, 255, 0.8);
    --text-size:        25px;
    --hover-color:      white;
    --bkgd-color:       #0F2E3D;
  }

  .distill-site-footer {
    --text-color:       rgba(255, 255, 255, 0.8);
    --text-size:        15px;
    --hover-color:      white;
    --bkgd-color:       #0F2E3D;
  }

  /*-- Additional custom styles --*/
  /* Add any additional CSS rules below                      */

  h1{
    text-align: center;
    font-size: 40pt;
    font-weight: bold;
    }

  h2{
    text-align: center;
    font-size: 35pt;
    font-weight: bold;
  }

  h3{
    text-align: center;
    font-size: 30pt;
    font-weight: bold;
  }

  .citation {
    text-align: right;
  }

  .jump {
    text-align: center;
  }</style>
  <style type="text/css">
  /* base style */

  /* FONT FAMILIES */

  :root {
    --heading-default: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    --mono-default: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    --body-default: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  body,
  .posts-list .post-preview p,
  .posts-list .description p {
    font-family: var(--body-font), var(--body-default);
  }

  h1, h2, h3, h4, h5, h6,
  .posts-list .post-preview h2,
  .posts-list .description h2 {
    font-family: var(--heading-font), var(--heading-default);
  }

  d-article div.sourceCode code,
  d-article pre code {
    font-family: var(--mono-font), var(--mono-default);
  }


  /*-- TITLE --*/
  d-title h1,
  .posts-list > h1 {
    color: var(--title-color, black);
  }

  d-title h1 {
    font-size: var(--title-size, 50px);
  }

  /*-- HEADERS --*/
  d-article h1,
  d-article h2,
  d-article h3,
  d-article h4,
  d-article h5,
  d-article h6 {
    color: var(--header-color, rgba(0, 0, 0, 0.8));
  }

  /*-- BODY --*/
  d-article > p,  /* only text inside of <p> tags */
  d-article > ul, /* lists */
  d-article > ol {
    color: var(--body-color, rgba(0, 0, 0, 0.8));
    font-size: var(--body-size, 1.06rem);
  }


  /*-- CODE --*/
  d-article div.sourceCode code,
  d-article pre code {
    font-size: var(--code-size, 14px);
  }

  /*-- ASIDE --*/
  d-article aside {
    font-size: var(--aside-size, 12px);
    color: var(--aside-color, rgba(0, 0, 0, 0.6));
  }

  /*-- FIGURE CAPTIONS --*/
  figure .caption,
  figure figcaption,
  .figure .caption {
    font-size: var(--fig-cap-size, 13px);
    color: var(--fig-cap-color, rgba(0, 0, 0, 0.6));
  }

  /*-- METADATA --*/
  d-byline h3 {
    font-size: var(--heading-size, 0.6rem);
    color: var(--heading-color, rgba(0, 0, 0, 0.5));
  }

  d-byline {
    font-size: var(--body-size, 0.8rem);
    color: var(--body-color, rgba(0, 0, 0, 0.8));
  }

  d-byline a,
  d-article d-byline a {
    color: var(--body-color, rgba(0, 0, 0, 0.8));
  }

  /*-- TABLE OF CONTENTS --*/
  .d-contents nav h3 {
    font-size: var(--heading-size, 18px);
  }

  .d-contents nav a {
    font-size: var(--contents-size, 13px);
  }

  /*-- APPENDIX --*/
  d-appendix h3 {
    font-size: var(--heading-size, 15px);
    color: var(--heading-color, rgba(0, 0, 0, 0.65));
  }

  d-appendix {
    font-size: var(--text-size, 0.8em);
    color: var(--text-color, rgba(0, 0, 0, 0.5));
  }

  d-appendix d-footnote-list a.footnote-backlink {
    color: var(--text-color, rgba(0, 0, 0, 0.5));
  }

  /*-- WEBSITE HEADER + FOOTER --*/
  .distill-site-header .title {
    font-size: var(--title-size, 18px);
    font-family: var(--navbar-font), var(--heading-default);
  }

  .distill-site-header a,
  .nav-dropdown .nav-dropbtn {
    font-family: var(--navbar-font), var(--heading-default);
  }

  .nav-dropdown .nav-dropbtn {
    color: var(--text-color, rgba(255, 255, 255, 0.8));
    font-size: var(--text-size, 15px);
  }

  .distill-site-header a:hover,
  .nav-dropdown:hover .nav-dropbtn {
    color: var(--hover-color, white);
  }

  .distill-site-header {
    font-size: var(--text-size, 15px);
    color: var(--text-color, rgba(255, 255, 255, 0.8));
    background-color: var(--bkgd-color, #0F2E3D);
  }

  .distill-site-footer {
    font-size: var(--text-size, 15px);
    color: var(--text-color, rgba(255, 255, 255, 0.8));
    background-color: var(--bkgd-color, #0F2E3D);
  }

  .distill-site-footer a:hover {
    color: var(--hover-color, white);
  }</style>
  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
  <link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
  <script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
  <script src="site_libs/leaflet-1.3.1/leaflet.js"></script>
  <link href="site_libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
  <script src="site_libs/proj4-2.6.2/proj4.min.js"></script>
  <script src="site_libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
  <link href="site_libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
  <script src="site_libs/leaflet-binding-2.2.2/leaflet.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Useful Packages","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">RStudio Reference Guide</a>
</div>
<div class="nav-right">
<a href="index.html">Overview</a>
<a href="basics.html">Basics</a>
<a href="about.html">Keyboard Shortcuts</a>
<a href="usepacks.html">Useful Packages</a>
<a href="installpack.html">Installing Packages</a>
<a href="resources.html">Resources</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Useful Packages</h1>

<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->

</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#tidyverse" id="toc-tidyverse">Tidyverse</a>
<ul>
<li><a href="#readrpack" id="toc-readrpack"><em>readr</em></a></li>
<li><a href="#tidyrpack" id="toc-tidyrpack"><em>tidyr</em></a></li>
<li><a href="#dplyrpack" id="toc-dplyrpack"><em>dplyr</em></a></li>
<li><a href="#tibblepack" id="toc-tibblepack"><em>tibble</em></a></li>
<li><a href="#stringrpack" id="toc-stringrpack"><em>stringr</em></a></li>
<li><a href="#forcatspack" id="toc-forcatspack"><em>forcats</em></a></li>
<li><a href="#lubridatepack" id="toc-lubridatepack"><em>lubridate</em></a></li>
</ul></li>
<li><a href="#figures" id="toc-figures">Figures</a>
<ul>
<li><a href="#directlabelspack" id="toc-directlabelspack"><em>directlabels</em></a></li>
<li><a href="#esquissepack" id="toc-esquissepack"><em>esquisse</em></a></li>
<li><a href="#fishualizepack" id="toc-fishualizepack"><em>fishualize</em></a></li>
<li><a href="#geomtextpathpack" id="toc-geomtextpathpack"><em>geomtextpath</em></a></li>
<li><a href="#gganimatepack" id="toc-gganimatepack"><em>gganimate</em></a></li>
<li><a href="#ggbreakpack" id="toc-ggbreakpack"><em>ggbreak</em></a></li>
<li><a href="#ggbumppack" id="toc-ggbumppack"><em>ggbump</em></a></li>
<li><a href="#ggeasypack" id="toc-ggeasypack"><em>ggeasy</em></a></li>
<li><a href="#ggforcepack" id="toc-ggforcepack"><em>ggforce</em></a></li>
<li><a href="#gghighlightpack" id="toc-gghighlightpack"><em>gghighlight</em></a></li>
<li><a href="#ggiraphpack" id="toc-ggiraphpack"><em>ggiraph</em></a></li>
<li><a href="#ggpatternpack" id="toc-ggpatternpack"><em>ggpattern</em></a></li>
<li><a href="#ggraphpack" id="toc-ggraphpack"><em>ggraph</em></a></li>
<li><a href="#ggrepelpack" id="toc-ggrepelpack"><em>ggrepel</em></a></li>
<li><a href="#ggsankeyfierpack" id="toc-ggsankeyfierpack"><em>ggsankeyfier</em></a></li>
<li><a href="#ggstatsplotpack" id="toc-ggstatsplotpack"><em>ggstatsplot</em></a></li>
<li><a href="#ggtextpack" id="toc-ggtextpack"><em>ggtext</em></a></li>
<li><a href="#ggthemespack" id="toc-ggthemespack"><em>ggthemes</em></a></li>
<li><a href="#ggwordcloudpack" id="toc-ggwordcloudpack"><em>ggwordcloud</em></a></li>
<li><a href="#hrbrthemespack" id="toc-hrbrthemespack"><em>hrbrthemes</em></a></li>
<li><a href="#patchworkpack" id="toc-patchworkpack"><em>patchwork</em></a></li>
<li><a href="#paletteerpack" id="toc-paletteerpack"><em>paletteer</em></a></li>
<li><a href="#pbapplypack" id="toc-pbapplypack"><em>pbapply</em></a></li>
<li><a href="#plot3Dpack" id="toc-plot3Dpack"><em>plot3D</em></a></li>
<li><a href="#plotrixpack" id="toc-plotrixpack"><em>plotrix</em></a></li>
<li><a href="#shinypack" id="toc-shinypack"><em>shiny</em></a></li>
<li><a href="#showtextpack" id="toc-showtextpack"><em>showtext</em></a></li>
</ul></li>
<li><a href="#fisheries" id="toc-fisheries">Fisheries Packages</a>
<ul>
<li><a href="#BayesGrowthpack" id="toc-BayesGrowthpack"><em>BayesGrowth</em></a></li>
<li><a href="#CatDynpack" id="toc-CatDynpack"><em>CatDyn</em></a></li>
<li><a href="#FEStapack" id="toc-FEStapack"><em>FESta</em></a></li>
<li><a href="#fishModpack" id="toc-fishModpack"><em>fishMod</em></a></li>
<li><a href="#FSApack" id="toc-FSApack"><em>FSA</em></a></li>
<li><a href="#ggFishPlotspack" id="toc-ggFishPlotspack"><em>ggFishPlots</em></a></li>
<li><a href="#RFishBCpack" id="toc-RFishBCpack"><em>RFishBC</em></a></li>
</ul></li>
<li><a href="#geo" id="toc-geo">Geospatial Packages</a>
<ul>
<li><a href="#mapsmapdatapack" id="toc-mapsmapdatapack"><em>maps</em>/<em>mapdata</em></a></li>
<li><a href="#rayshaderpack" id="toc-rayshaderpack"><em>rayshader</em></a></li>
<li><a href="#sfpack" id="toc-sfpack"><em>sf</em></a></li>
<li><a href="#tmappack" id="toc-tmappack"><em>tmap</em></a></li>
</ul></li>
<li><a href="#misc" id="toc-misc">Miscellaneous</a>
<ul>
<li><a href="#EchoviewRpack" id="toc-EchoviewRpack"><em>EchoviewR</em></a></li>
<li><a href="#rticlespack" id="toc-rticlespack"><em>rticles</em></a></li>
<li><a href="#tinytexpack" id="toc-tinytexpack"><em>tinytex</em></a></li>
</ul></li>
<li><a href="#stats" id="toc-stats">Statistics</a>
<ul>
<li><a href="#AICcmodavgpack" id="toc-AICcmodavgpack"><em>AICcmodavg</em></a></li>
<li><a href="#bbmlepack" id="toc-bbmlepack"><em>bbmle</em></a></li>
<li><a href="#DHARMapack" id="toc-DHARMapack"><em>DHARMa</em></a></li>
<li><a href="#emmeanspack" id="toc-emmeanspack"><em>emmeans</em></a></li>
<li><a href="#gamlsspack" id="toc-gamlsspack"><em>gamlss</em></a></li>
<li><a href="#glmmTMBpack" id="toc-glmmTMBpack"><em>glmmTMB</em></a></li>
<li><a href="#lme4pack" id="toc-lme4pack"><em>lme4</em></a></li>
</ul></li>
<li><a href="#tables" id="toc-tables">Tables</a>
<ul>
<li><a href="#DTpack" id="toc-DTpack"><em>DT</em></a></li>
<li><a href="#gtsummarypack" id="toc-gtsummarypack"><em>gtsummary</em></a></li>
</ul></li>
<li><a href="#telem" id="toc-telem">Telemetry</a>
<ul>
<li><a href="#actelpack" id="toc-actelpack"><em>actel</em></a></li>
<li><a href="#adehabitatHRpack" id="toc-adehabitatHRpack"><em>adehabitatHR</em></a></li>
<li><a href="#adehabitatHSpack" id="toc-adehabitatHSpack"><em>adehabitatHS</em></a></li>
<li><a href="#adehabitatLTpack" id="toc-adehabitatLTpack"><em>adehabitatLT</em></a></li>
<li><a href="#adehabitatMApack" id="toc-adehabitatMApack"><em>adehabitatMA</em></a></li>
<li><a href="#amtpack" id="toc-amtpack"><em>amt</em></a></li>
<li><a href="#m2bpack" id="toc-m2bpack"><em>m2b</em></a></li>
<li><a href="#sftrackpack" id="toc-sftrackpack"><em>sftrack</em></a></li>
<li><a href="#SiMRivpack" id="toc-SiMRivpack"><em>SiMRiv</em></a></li>
<li><a href="#wildlifeDIpack" id="toc-wildlifeDIpack"><em>wildlifeDI</em></a></li>
</ul></li>
</ul>
</nav>
</div>
<p>There are a variety of helpful packages which are briefly listed below, along with links to documentation, tutorials, cheatsheets, and other potentially helpful resources to help you learn how to use the packages. They are broadly organized into the following sections:</p>
<ul>
<li>Tidyverse</li>
<li>Figures</li>
<li>Fisheries Packages</li>
<li>Geospatial Packages</li>
<li>Miscellaneous</li>
<li>Statistics</li>
<li>Tables</li>
<li>Telemetry</li>
</ul>
<p>The table below briefly lists the contents of each section, but it is highly recommended you familiarize yourself with the gist of many of the packages listed.</p>
<h1 id="tidyverse">Tidyverse</h1>
<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Package</th>
<th style="text-align: center;">Section Link</th>
<th style="text-align: center;">Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong><em>readr</em></strong></td>
<td style="text-align: center;"><a href="#readrpack">link to <em>readr</em></a></td>
<td style="text-align: center;">data import</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>tidyr</em></strong></td>
<td style="text-align: center;"><a href="#tidyrpack">link to <em>tidyr</em></a></td>
<td style="text-align: center;">data tidying</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>dplyr</em></strong></td>
<td style="text-align: center;"><a href="#dplyrpack">link to <em>dplyr</em></a></td>
<td style="text-align: center;">data manipulation</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>ggplot2</em></strong></td>
<td style="text-align: center;"><a href="#ggplot2pack">link to <em>ggplot2</em></a></td>
<td style="text-align: center;">data visualization</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>purrr</em></strong></td>
<td style="text-align: center;"><a href="#purrrpack">link to <em>purrr</em></a></td>
<td style="text-align: center;">functional programming</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>tibble</em></strong></td>
<td style="text-align: center;"><a href="#tibblepack">link to <em>tibble</em></a></td>
<td style="text-align: center;">re-imaging dataframes</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>stringr</em></strong></td>
<td style="text-align: center;"><a href="#stringrpack">link to <em>stringr</em></a></td>
<td style="text-align: center;">string manipulation</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>forcats</em></strong></td>
<td style="text-align: center;"><a href="#forcatspack">link to <em>forcats</em></a></td>
<td style="text-align: center;">factor manipulation</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>lubridate</em></strong></td>
<td style="text-align: center;"><a href="#lubridatepack">link to <em>lubridate</em></a></td>
<td style="text-align: center;">dates and times</td>
</tr>
</tbody>
</table>
<blockquote>
<p>The <strong><em>tidyverse</em></strong> <span class="citation" data-cites="tidyverse">(<a href="#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span> is a set of packages that work in harmony because they share common data representations and API [Application Programming Interfaces] design. The <strong><em>tidyverse</em></strong> package is designed to make it easy to install and load core packages from the <strong><em>tidyverse</em></strong> in a single command.</p>
If you’d like to learn how to use the <strong><em>tidyverse</em></strong> effectively, the best place to start is <a href="https://r4ds.hadley.nz/"><strong>R for Data Science (2e)</strong></a>.
<div class="citation">
<strong>Source:</strong> <a href="https://tidyverse.tidyverse.org/index.html">tidyverse</a>
</div>
</blockquote>
<p><strong><em>tidyverse</em></strong> is a collection of several key packages that together are extremely powerful at data manipulation and visualization. It is one of the most important packages to become familiar with in the field of fisheries.</p>
<p>Calling the <strong><em>tidyverse</em></strong> will also load the core <strong><em>tidyverse</em></strong> packages, including:</p>
<ul>
<li><p><strong><em>readr</em></strong>, for data import</p></li>
<li><p><strong><em>tidyr</em></strong>, for data tidying</p></li>
<li><p><strong><em>dplyr</em></strong>, for data manipulation</p></li>
<li><p><strong><em>ggplot2</em></strong>, for data visualization</p></li>
<li><p><strong><em>purrr</em></strong>, for functional programming</p></li>
<li><p><strong><em>tibble</em></strong>, for modern re-imagining of data frames</p></li>
<li><p><strong><em>stringr</em></strong>, for strings</p></li>
<li><p><strong><em>forcats</em></strong>, for factors</p></li>
<li><p><strong><em>lubridate</em></strong>, for dates/times</p></li>
</ul>
<p><a href="https://tidyverse.tidyverse.org/"><strong><em>tidyverse</em> website</strong></a></p>
<p><a href="https://tidyverse.tidyverse.org/reference/index.html"><strong><em>tidyverse</em> functions</strong></a></p>
<p><a href="https://tidyverse.tidyverse.org/articles/manifesto.html"><strong><em>tidyverse</em> article: the tidy tools manifesto</strong></a></p>
<p><a href="https://tidyverse.tidyverse.org/articles/paper.html"><strong><em>tidyverse</em> article: welcome to the</strong> <strong><em>tidyverse</em></strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/tidyverse/tidyverse.pdf"><strong><em>tidyverse</em> reference manual</strong></a></p>
<h2 id="readrpack"><em>readr</em></h2>
<blockquote>
The goal of <strong><em>readr</em></strong> <span class="citation" data-cites="readr">(<a href="#ref-readr" role="doc-biblioref">Wickham, Hester, and Bryan 2024</a>)</span> is to provide a fast and friendly way to read rectangular data from delimited files, such as comma-separated values (CSV) and tab-separated values (TSV). It is designed to parse many types of data found in the wild, while providing an informative problem report when parsing leads to unexpected results. If you are new to <strong><em>readr</em></strong>, the best place to start is the <a href="https://r4ds.hadley.nz/data-import">data import chapter</a> in R for Data Science.
<div class="citation">
<strong>Source:</strong> <a href="https://readr.tidyverse.org/">readr</a>
</div>
</blockquote>
<p><strong><em>readr</em></strong> allows you to read in various types of data files. Specifically, it is helpful when reading in comma separated values (CSV) files by using the following code:</p>
<blockquote>
<p>read_csv(“filename.csv”)</p>
</blockquote>
<p><a href="https://readr.tidyverse.org/"><strong><em>readr</em></strong> <strong>website</strong></a></p>
<p><a href="https://readr.tidyverse.org/reference/index.html"><strong><em>readr</em> functions</strong></a></p>
<p><a href="https://readr.tidyverse.org/articles/readr.html"><strong><em>readr</em> article: introduction</strong></a></p>
<p><a href="https://readr.tidyverse.org/articles/column-types.html"><strong><em>readr</em> article: column type</strong></a></p>
<p><a href="https://readr.tidyverse.org/articles/locales.html"><strong><em>readr</em> article: locales</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/readr/readr.pdf"><strong><em>readr</em> reference manual</strong></a></p>
<p><a href="https://rstudio.github.io/cheatsheets/html/data-import.html"><strong><em>readr</em> tutorial</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/readr1.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-2"></span>
<img src="images/readr2.png" alt="[***readr*** **cheatsheet**](https://readr.tidyverse.org/#cheatsheet)" width="1000" />
<p class="caption">
Figure 1: <a href="https://readr.tidyverse.org/#cheatsheet"><strong><em>readr</em></strong> <strong>cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#tidyverse">Return to Tidyverse</a></strong>
</div>
<h2 id="tidyrpack"><em>tidyr</em></h2>
<blockquote>
<p>The goal of <strong><em>tidyr</em></strong> <span class="citation" data-cites="tidyr">(<a href="#ref-tidyr" role="doc-biblioref">Wickham, Vaughan, and Girlich 2024</a>)</span> is to help you create tidy data. Tidy data is data where:</p>
<ol type="1">
<li><p>Each variable is a column; each column is a variable.</p></li>
<li><p>Each observation is a row; each row is an observation.</p></li>
<li><p>Each value is a cell; each cell is a single value.</p></li>
</ol>
Tidy data describes a standard way of storing data that is used wherever possible throughout the <strong><em>tidyverse</em></strong>. If you ensure that your data is tidy, you’ll spend less time fighting with the tools and more time working on your analysis. Learn more about tidy data in <a href="https://tidyr.tidyverse.org/articles/tidy-data.html"><strong>vignette (“tidy-data”</strong>)</a>.
<div class="citation">
<strong>Source:</strong> <a href="https://tidyr.tidyverse.org/">tidyr</a>
</div>
</blockquote>
<p><a href="https://tidyr.tidyverse.org/"><strong><em>tidyr</em></strong> <strong>website</strong></a></p>
<p><a href="https://tidyr.tidyverse.org/articles/pivot.html"><strong><em>tidyr</em></strong> <strong>article: pivoting</strong></a></p>
<p><a href="https://tidyr.tidyverse.org/articles/rectangle.html"><strong><em>tidyr</em></strong> <strong>article: rectangling</strong></a></p>
<p><a href="https://tidyr.tidyverse.org/articles/nest.html"><strong><em>tidyr</em></strong> <strong>article: nested data</strong></a></p>
<p><a href="https://tidyr.tidyverse.org/articles/tidy-data.html"><strong><em>tidyr</em></strong> <strong>article: tidy data (theory)</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/tidyr/tidyr.pdf"><strong><em>tidyr</em></strong> <strong>reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/tidyr.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="images/tidyr2.png" alt="[***tidyr*** **cheatsheet**](https://tidyr.tidyverse.org/#cheatsheet)" width="1000" />
<p class="caption">
Figure 2: <a href="https://tidyr.tidyverse.org/#cheatsheet"><strong><em>tidyr</em></strong> <strong>cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#tidyverse">Return to Tidyverse</a></strong>
</div>
<h2 id="dplyrpack"><em>dplyr</em></h2>
<blockquote>
<p><strong><em>dplyr</em></strong> <span class="citation" data-cites="dplyr">(<a href="#ref-dplyr" role="doc-biblioref">Wickham et al. 2023</a>)</span> is a grammar of data manipulation, providing a consistent set of verbs that help you solve the most common data manipulation challenges:</p>
<p><strong>mutate()</strong> adds new variables that are functions of existing variables</p>
<p><strong>select()</strong> picks variables based on their names.</p>
<p><strong>filter()</strong> picks cases based on their values.</p>
<p><strong>summarise()</strong> reduces multiple values down to a single summary.</p>
<p><strong>arrange()</strong> changes the ordering of the rows.</p>
<p>These all combine naturally with <a href="https://dplyr.tidyverse.org/reference/group_by.html"><strong>group_by()</strong></a> which allows you to perform any operation “by group”. You can learn more about them in <a href="https://dplyr.tidyverse.org/articles/dplyr.html"><strong>vignette (“<em>dplyr</em>”)</strong></a>. As well as these single-table verbs, <strong><em>dplyr</em></strong> also provides a variety of two-table verbs, which you can learn about in <a href="https://dplyr.tidyverse.org/articles/two-table.html"><strong>vignette (“two-table”</strong>)</a>.</p>
If you are new to <strong><em>dplyr</em></strong>, the best place to start is the <a href="https://r4ds.hadley.nz/data-transform">data transformation chapter</a> in R for Data Science.
<div class="citation">
<strong>Source:</strong> <a href="https://dplyr.tidyverse.org/">dplyr</a>
</div>
</blockquote>
<p><a href="https://dplyr.tidyverse.org/"><strong><em>dplyr</em></strong> <strong>website</strong></a></p>
<p><a href="https://dplyr.tidyverse.org/articles/dplyr.html"><strong><em>dplyr</em></strong> <strong>article: introduction</strong></a></p>
<p><a href="https://dplyr.tidyverse.org/articles/grouping.html"><strong><em>dplyr</em></strong> <strong>article: grouped data</strong></a></p>
<p><a href="https://dplyr.tidyverse.org/articles/two-table.html"><strong><em>dplyr</em></strong> <strong>article: two-table verbs</strong></a></p>
<p><a href="https://dplyr.tidyverse.org/articles/base.html"><strong><em>dplyr</em></strong> <strong>article: <em>dplyr</em> base R</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/dplyr/dplyr.pdf"><strong><em>dplyr</em> reference guide</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/dplyr1.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="images/dplyr2.png" alt="[***dplyr* cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/data-transformation.pdf)" width="1000" />
<p class="caption">
Figure 3: <a href="https://github.com/rstudio/cheatsheets/blob/main/data-transformation.pdf"><strong><em>dplyr</em> cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#tidyverse">Return to Tidyverse</a></strong>
</div>
<h3 id="ggplot2pack"><em>ggplot2</em></h3>
<blockquote>
<p><strong><em>ggplot2</em></strong> <span class="citation" data-cites="ggplot2">(<a href="#ref-ggplot2" role="doc-biblioref">Wickham 2016</a>)</span> is a system for declaratively creating graphics, based on The Grammar of Graphics. You provide the data, tell <strong><em>ggplot2</em></strong> how to map variables to aesthetics, what graphical primitives to use, and it takes care of the details….</p>
<p>If you are new to <strong><em>ggplot2</em></strong> you are better off starting with a systematic introduction, rather than trying to learn from reading individual documentation pages. Currently, there are three good places to start:</p>
<ul>
<li><p>The <a href="https://r4ds.hadley.nz/data-visualize"><strong>Data Visualization</strong></a> and <a href="https://r4ds.hadley.nz/communication"><strong>Communication</strong></a> chapters in <a href="https://r4ds.hadley.nz/"><strong>R for Data Science</strong></a>. R for Data Science is designed to give you a comprehensive introduction to the <a href="https://www.tidyverse.org/"><strong><em>tidyverse</em></strong></a>, and these two chapters will get you up to speed with the essentials of <strong><em>ggplot2</em></strong> as quickly as possible.</p></li>
<li><p>If you’d like to take an online course, try <a href="https://www.oreilly.com/videos/data-visualization-in/9781491963661/"><strong>Data Visualization in R With <em>ggplot2</em></strong></a> by Kara Woo.</p></li>
<li><p>If you’d like to follow a webinar, try <a href="https://www.youtube.com/watch?v=h29g21z0a68&amp;feature=youtu.be"><strong>Plotting Anything with <em>ggplot2</em></strong></a> by Thomas Lin Pedersen.</p></li>
</ul>
<p>If you want to dive into making common graphics as quickly as possible, I recommend the <a href="https://r-graphics.org/"><strong>R Graphics Cookbook</strong></a> by Winston Chang. It provides a set of recipes to solve common graphics problems.</p>
If you’ve mastered the basics and want to learn more, read <a href="https://ggplot2-book.org/"><strong><em>ggplot2</em>: Elegant Graphics for Data Analysis</strong></a>. It describes the theoretical underpinnings of <strong><em>ggplot2</em></strong> and shows you how all the pieces fit together. This book helps you understand the theory that underpins <strong><em>ggplot2</em></strong>, and will help you create new types of graphics specifically tailored to your needs.
<div class="citation">
<strong>Source:</strong> <a href="https://ggplot2.tidyverse.org/">ggplot2</a>
</div>
</blockquote>
<p><a href="https://ggplot2.tidyverse.org/"><strong><em>ggplot2</em></strong> <strong>website</strong></a></p>
<p><a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html"><strong><em>ggplot2</em></strong> <strong>tutorial: aesthetic specifications</strong></a></p>
<p><a href="https://ggplot2.tidyverse.org/articles/faq-faceting.html"><strong><em>ggplot2</em></strong> <strong>tutorial: faceting FAQ</strong></a></p>
<p><a href="https://ggplot2.tidyverse.org/articles/faq-customising.html"><strong><em>ggplot2</em> tutorial: customizing FAQ</strong></a></p>
<p><a href="https://ggplot2.tidyverse.org/articles/faq-annotation.html"><strong><em>ggplot2</em> tutorial: annotation FAQ</strong></a></p>
<p><a href="https://ggplot2.tidyverse.org/articles/faq-reordering.html"><strong><em>ggplot2</em> tutorial: reordering FAQ</strong></a></p>
<p><a href="https://ggplot2.tidyverse.org/articles/faq-bars.html"><strong><em>ggplot2</em> tutorial: barplots FAQ</strong></a></p>
<p><a href="https://rstudio.github.io/cheatsheets/html/data-visualization.html"><strong><em>ggplot2</em></strong> <strong>tutorial</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/ggplot2/ggplot2.pdf"><strong><em>ggplot2</em> reference manual</strong></a></p>
<p><a href="https://r-graph-gallery.com/ggplot2-package.html"><strong><em>ggplot2</em> tutorial</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/ggplot2.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="images/ggplot22.png" alt="[***ggplot2* cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/data-visualization.pdf)" width="1000" />
<p class="caption">
Figure 4: <a href="https://github.com/rstudio/cheatsheets/blob/main/data-visualization.pdf"><strong><em>ggplot2</em> cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#tidyverse">Return to Tidyverse</a></strong>
</div>
<h3 id="purrrpack"><em>purrr</em></h3>
<blockquote>
<strong><em>purrr</em></strong> <span class="citation" data-cites="purrr">(<a href="#ref-purrr" role="doc-biblioref">Wickham and Henry 2023</a>)</span> enhances R’s functional programming (FP) toolkit by providing a complete and consistent set of tools for working with functions and vectors. If you’ve never heard of FP before, the best place to start is the family of <a href="https://purrr.tidyverse.org/reference/map.html"><strong>map()</strong></a> functions which allow you to replace many for loops with code that is both more succinct and easier to read. The best place to learn about the <strong>map()</strong> functions is the <a href="https://r4ds.hadley.nz/iteration">iteration chapter</a> in R for Data Science.
<div class="citation">
<strong>Source:</strong> <a href="https://purrr.tidyverse.org/">purrr</a>
</div>
</blockquote>
<p><a href="https://purrr.tidyverse.org/"><strong><em>purrr</em> website</strong></a></p>
<p><a href="https://purrr.tidyverse.org/articles/base.html"><strong><em>purrr</em> article: base R</strong></a></p>
<p><a href="https://purrr.tidyverse.org/articles/other-langs.html"><strong><em>purrr</em> article: functional programming in other languages</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/purrr/purrr.pdf"><strong><em>purrr</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/purrr1.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-10"></span>
<img src="images/purrr2.png" alt="[***purrr*** **cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/purrr.pdf)" width="1000" />
<p class="caption">
Figure 5: <a href="https://github.com/rstudio/cheatsheets/blob/main/purrr.pdf"><strong><em>purrr</em></strong> <strong>cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#tidyverse">Return to Tidyverse</a></strong>
</div>
<h2 id="tibblepack"><em>tibble</em></h2>
<blockquote>
<p>A <strong><em>tibble</em></strong> <span class="citation" data-cites="tibble">(<a href="#ref-tibble" role="doc-biblioref">Müller and Wickham 2023</a>)</span>, or tbl_df, is a modern reimagining of the data.frame, keeping what time has proven to be effective, and throwing out what is not. Tibbles are data.frames that are lazy and surly: they do less (i.e. they don’t change variable names or types, and don’t do partial matching) and complain more (e.g. when a variable does not exist). This forces you to confront problems earlier, typically leading to cleaner, more expressive code. Tibbles also have an enhanced print() method which makes them easier to use with large datasets containing complex objects.</p>
If you are new to tibbles, the best place to start is the <a href="https://r4ds.had.co.nz/tibbles.html">tibbles chapter</a> in <em>R for data science</em>.
<div class="citation">
<strong>Source:</strong> <a href="https://tibble.tidyverse.org/">tibble</a>
</div>
</blockquote>
<p><a href="https://tibble.tidyverse.org/"><strong><em>tibble</em></strong> <strong>website</strong></a></p>
<p><a href="https://tibble.tidyverse.org/articles/tibble.html"><strong><em>tibble</em> article: getting started</strong></a></p>
<p><a href="https://tibble.tidyverse.org/articles/types.html"><strong><em>tibble</em> article: column types</strong></a></p>
<p><a href="https://tibble.tidyverse.org/articles/numbers.html"><strong><em>tibble</em> article: controlling display of numbers</strong></a></p>
<p><a href="https://tibble.tidyverse.org/articles/digits.html"><strong><em>tibble</em> article: comparing display with data frames</strong></a></p>
<p><a href="https://tibble.tidyverse.org/reference/index.html"><strong><em>tibble</em> article: functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/tibble/tibble.pdf"><strong><em>tibble</em></strong> <strong>reference manual</strong></a></p>
<div class="citation">
<strong><a href="#tidyverse">Return to Tidyverse</a></strong>
</div>
<h2 id="stringrpack"><em>stringr</em></h2>
<blockquote>
Strings are not glamorous, high-profile components of R, but they do play a big role in many data cleaning and preparation tasks. The <strong><em>stringr</em></strong> <span class="citation" data-cites="stringr">(<a href="#ref-stringr" role="doc-biblioref">Wickham 2023b</a>)</span> package provides a cohesive set of functions designed to make working with strings as easy as possible. If you’re not familiar with strings, the best place to start is the chapter on <a href="https://r4ds.hadley.nz/strings">strings</a> in R for Data Science.
<div class="citation">
<strong>Source:</strong> <a href="https://stringr.tidyverse.org/">stringr</a>
</div>
</blockquote>
<p><a href="https://stringr.tidyverse.org/"><strong><em>stringr</em></strong> <strong>website</strong></a></p>
<p><a href="https://stringr.tidyverse.org/articles/stringr.html"><strong><em>stringr</em> article: introduction to <em>stringr</em></strong></a></p>
<p><a href="https://stringr.tidyverse.org/articles/from-base.html"><strong><em>stringr</em> article: from base R</strong></a></p>
<p><a href="https://stringr.tidyverse.org/articles/regular-expressions.html"><strong><em>stringr</em> article: regular expressions</strong></a></p>
<p><a href="https://stringr.tidyverse.org/reference/index.html"><strong><em>stringr</em> functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/stringr/stringr.pdf"><strong><em>stringr</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/stringr1.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-12"></span>
<img src="images/stringr2.png" alt="[***stringr*** **cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/strings.pdf)" width="1000" />
<p class="caption">
Figure 6: <a href="https://github.com/rstudio/cheatsheets/blob/main/strings.pdf"><strong><em>stringr</em></strong> <strong>cheatsheet</strong></a>
</p>
</div>
</div>
<h2 id="forcatspack"><em>forcats</em></h2>
<blockquote>
<p>R uses <strong>factors</strong> to handle categorical variables, variables that have a fixed and known set of possible values. Factors are also helpful for reordering character vectors to improve display. The goal of the <strong><em>forcats</em></strong> <span class="citation" data-cites="forcats">(<a href="#ref-forcats" role="doc-biblioref">Wickham 2023a</a>)</span> package is to provide a suite of tools that solve common problems with factors, including changing the order of levels or the values. Some examples include:</p>
<p><strong>fct_reorder()</strong>: Reordering a factor by another variable.</p>
<p><strong>fct_infreq()</strong>: Reordering a factor by the frequency of values.</p>
<p><strong>fct_relevel()</strong>: Changing the order of a factor by hand.</p>
<p><strong>fct_lump()</strong>: Collapsing the least/most frequent values of a factor into “other”.</p>
You can learn more about each of these in <a href="https://forcats.tidyverse.org/articles/forcats.html">vignette (“forcats”)</a>. If you’re new to factors, the best place to start is the chapter on <a href="https://r4ds.hadley.nz/factors.html">factors</a> in R for Data Science.
<div class="citation">
<strong>Source:</strong> <a href="https://forcats.tidyverse.org/">forcats</a>
</div>
</blockquote>
<p><a href="https://forcats.tidyverse.org/"><strong><em>forcats</em></strong> <strong>website</strong></a></p>
<p><a href="https://forcats.tidyverse.org/articles/forcats.html"><strong><em>forcats</em> article: introduction</strong></a></p>
<p><a href="https://forcats.tidyverse.org/reference/index.html"><strong><em>forcats</em> functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/forcats/forcats.pdf"><strong><em>forcats</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-13"></span>
<img src="images/forcats1.png" alt="[***forcats*** **cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/factors.pdf)" width="1000" />
<p class="caption">
Figure 7: <a href="https://github.com/rstudio/cheatsheets/blob/main/factors.pdf"><strong><em>forcats</em></strong> <strong>cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#tidyverse">Return to Tidyverse</a></strong>
</div>
<h2 id="lubridatepack"><em>lubridate</em></h2>
<blockquote>
<p>Date-time data can be frustrating to work with in R. R commands for date-times are generally unintuitive and change depending on the type of date-time object being used. Moreover, the methods we use with date-times must be robust to time zones, leap days, daylight savings times, and other time related quirks, and R lacks these capabilities in some situations. <strong><em>lubridate</em></strong> <span class="citation" data-cites="lubridate">(<a href="#ref-lubridate" role="doc-biblioref">Grolemund and Wickham 2011</a>)</span> makes it easier to do the things R does with date-times and possible to do the things R does not.</p>
If you are new to <strong><em>lubridate</em></strong>, the best place to start is the <a href="https://r4ds.had.co.nz/dates-and-times.html">date and times chapter</a> in R for data science.
<div class="citation">
<strong>Source:</strong> <a href="https://lubridate.tidyverse.org/">lubridate</a>
</div>
</blockquote>
<p><a href="https://lubridate.tidyverse.org/"><strong><em>lubridate</em> website</strong></a></p>
<p><a href="https://lubridate.tidyverse.org/reference/index.html"><strong><em>lubridate</em> functions</strong></a></p>
<p><a href="https://rstudio.github.io/cheatsheets/html/lubridate.html"><strong><em>lubridate</em> tutorial</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/lubridate/lubridate.pdf"><strong><em>lubridate</em> reference manual</strong></a></p>
<p><a href="https://lubridate.tidyverse.org/articles/lubridate.html"><strong><em>lubridate</em> article: do more with dates and times in R</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/lubridate1.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-15"></span>
<img src="images/lubridate2.png" alt="[***lubridate*** **cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/lubridate.pdf)" width="1000" />
<p class="caption">
Figure 8: <a href="https://github.com/rstudio/cheatsheets/blob/main/lubridate.pdf"><strong><em>lubridate</em></strong> <strong>cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#tidyverse">Return to Tidyverse</a></strong>
</div>
<h1 id="figures">Figures</h1>
<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Package</th>
<th style="text-align: center;">Section Link</th>
<th style="text-align: center;">Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong><em>directlabels</em></strong></td>
<td style="text-align: center;"><a href="#directlabelspack">link to <em>directlabels</em></a></td>
<td style="text-align: center;">adjust label positions</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>esquisse</em></strong></td>
<td style="text-align: center;"><a href="#esquissepack">link to <em>esquisse</em></a></td>
<td style="text-align: center;">shiny app to visualize data</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>fishualize</em></strong></td>
<td style="text-align: center;"><a href="#fishualizepack">link to <em>fishualize</em></a></td>
<td style="text-align: center;">fish shapes and color palettes</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>geomtextpath</em></strong></td>
<td style="text-align: center;"><a href="#geomtextpathpack">link to <em>geomtextpath</em></a></td>
<td style="text-align: center;">adjust label position and angles</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>gganimate</em></strong></td>
<td style="text-align: center;"><a href="#gganimatepack">link to <em>gganimate</em></a></td>
<td style="text-align: center;">create interactive plots</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>ggbreak</em></strong></td>
<td style="text-align: center;"><a href="#ggbreakpack">link to <em>ggbreak</em></a></td>
<td style="text-align: center;">create axis breaks</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>ggbump</em></strong></td>
<td style="text-align: center;"><a href="#ggbumppack">link to <em>ggbump</em></a></td>
<td style="text-align: center;">create bump charts</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>ggeasy</em></strong></td>
<td style="text-align: center;"><a href="#ggeasypack">link to <em>ggeasy</em></a></td>
<td style="text-align: center;">makes <em>ggplot2</em> more intuitive to use</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>ggforce</em></strong></td>
<td style="text-align: center;"><a href="#ggforcepack">link to <em>ggforce</em></a></td>
<td style="text-align: center;">force points to specific place on plot</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>gghighlight</em></strong></td>
<td style="text-align: center;"><a href="#gghighlightpack">link to <em>gghighlight</em></a></td>
<td style="text-align: center;">highlight specific region of data</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>ggiraph</em></strong></td>
<td style="text-align: center;"><a href="#ggiraphpack">link to <em>ggiraph</em></a></td>
<td style="text-align: center;">create dynamic graphs</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>ggpattern</em></strong></td>
<td style="text-align: center;"><a href="#ggpatternpack">link to <em>ggpattern</em></a></td>
<td style="text-align: center;">fill regions with custom patterns</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>ggraph</em></strong></td>
<td style="text-align: center;"><a href="#ggraphpack">link to <em>ggraph</em></a></td>
<td style="text-align: center;">support relational data structures</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>ggrepel</em></strong></td>
<td style="text-align: center;"><a href="#ggrepelpack">link to <em>ggrepel</em></a></td>
<td style="text-align: center;">removes overlaps in text labels</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>ggsankeyfier</em></strong></td>
<td style="text-align: center;"><a href="#ggsankeyfierpack">link to <em>ggsankeyfier</em></a></td>
<td style="text-align: center;">create sankey bump plots</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>ggstatsplot</em></strong></td>
<td style="text-align: center;"><a href="#ggstatsplotpack">link to <em>ggstatsplot</em></a></td>
<td style="text-align: center;">add statistical labels directly to plots</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>ggtext</em></strong></td>
<td style="text-align: center;"><a href="#ggtextpack">link to <em>ggtext</em></a></td>
<td style="text-align: center;">adjust orientation of labels</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>ggthemes</em></strong></td>
<td style="text-align: center;"><a href="#ggthemespack">link to <em>ggthemes</em></a></td>
<td style="text-align: center;">enhance plot aesthetics</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>ggwordcloud</em></strong></td>
<td style="text-align: center;"><a href="#ggwordcloudpack">link to <em>ggwordcloud</em></a></td>
<td style="text-align: center;">create custom wordclouds</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>hrbrthemes</em></strong></td>
<td style="text-align: center;"><a href="#hrbrthemespack">link to <em>hrbrthemes</em></a></td>
<td style="text-align: center;">choose fonts on figures</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>patchwork</em></strong></td>
<td style="text-align: center;"><a href="#patchworkpack">link to <em>patchwork</em></a></td>
<td style="text-align: center;">combine multiple plots</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>paletteer</em></strong></td>
<td style="text-align: center;"><a href="#paletteerpack">link to <em>paletteer</em></a></td>
<td style="text-align: center;">collection of 2759 palettes</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>pbapply</em></strong></td>
<td style="text-align: center;"><a href="#pbapplypack">link to <em>pbapply</em></a></td>
<td style="text-align: center;">add progress bars</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>plot3D</em></strong></td>
<td style="text-align: center;"><a href="#plot3Dpack">link to <em>plot3D</em></a></td>
<td style="text-align: center;">tools for plotting 2D and 3D data</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>plotrix</em></strong></td>
<td style="text-align: center;"><a href="#plotrixpack">link to <em>plotrix</em></a></td>
<td style="text-align: center;">plotting, labeling, and scaling</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>shiny</em></strong></td>
<td style="text-align: center;"><a href="#shinypack">link to <em>shiny</em></a></td>
<td style="text-align: center;">create interactive web apps</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>showtext</em></strong></td>
<td style="text-align: center;"><a href="#showtextpack">link to <em>showtext</em></a></td>
<td style="text-align: center;">customize font on figures</td>
</tr>
</tbody>
</table>
<blockquote>
<p><strong><em>ggplot2</em></strong> is very powerful on its own. But it gets even better thanks to its extensions</p>
<p>Here is a curated a collection of the best <strong><em>ggplot2</em></strong> extensions that enhance and elevate the core capabilities of <strong><em>ggplot2</em></strong>.</p>
<p>These extensions, seamlessly integrated with the foundational <strong><em>ggplot2</em></strong> framework, unlock <strong>new visualization techniques</strong>, streamline intricate <strong>plotting tasks</strong>, and offer innovative ways to <strong>represent complex data</strong>.</p>
<div class="citation">
<strong>Source:</strong> <a href="https://r-graph-gallery.com/best-dataviz-packages.html">R Graph Gallery</a>
</div>
</blockquote>
<h2 id="directlabelspack"><em>directlabels</em></h2>
<h3 id="direct-labels-for-multicolor-plots">Direct Labels for Multicolor Plots</h3>
<blockquote>
An extensible framework for automatically placing direct labels onto multicolor <strong><em>lattice</em></strong> or <strong><em>ggplot2</em></strong> plots. Label positions are described using Positioning Methods which can be re-used across several different plots. There are heuristics for examining “trellis” and “ggplot” objects and inferring an appropriate Positioning Method.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/directlabels/index.html">directlabels</a>
</div>
</blockquote>
<blockquote>
<p>There are 2 major reasons why the <strong><em>directlabels</em></strong> <span class="citation" data-cites="directlabels">(<a href="#ref-directlabels" role="doc-biblioref">Hocking 2025</a>)</span> package is so useful:</p>
<p>Direct labeling is difficult to do “by hand” using <strong>standard base</strong>/<strong><em>lattice</em></strong>/<strong><em>ggplot2</em></strong> plotting functions. The <strong><em>directlabels</em></strong> package allows you to define your direct labels in terms of a simple Positioning Method. This makes your code simpler, more readable, and less repetitive.</p>
The functions that you will use to direct label a plot “by hand” depend on the plot package you use. The directlabels package offers a unified interface for both lattice and <strong><em>ggplot2</em></strong> plots, thus greatly simplifying your code.
<div class="citation">
<strong>Source:</strong> <a href="https://tdhock.github.io/directlabels/motivation.html">directlabels</a>
</div>
</blockquote>
<br />

<div class="layout-chunk" data-layout="l-body">
<p><img src="images/confusing.png" width="150%" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-17"></span>
<img src="images/directlabels.png" alt="[***directlabels* example**](https://tdhock.github.io/directlabels/motivation.html)" width="150%" />
<p class="caption">
Figure 9: <a href="https://tdhock.github.io/directlabels/motivation.html"><strong><em>directlabels</em> example</strong></a>
</p>
</div>
</div>
<p><a href="https://tdhock.github.io/directlabels/"><strong><em>directlabels</em> website</strong></a></p>
<p><a href="https://tdhock.github.io/directlabels/motivation.html"><strong><em>directlabels</em> article: motivation for direct labeling</strong></a></p>
<p><a href="https://tdhock.github.io/directlabels/smart.html"><strong><em>directlabels</em> article: smart labeling</strong></a></p>
<p><a href="https://tdhock.github.io/directlabels/examples.html"><strong><em>directlabels</em> article: advanced examples</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/directlabels/directlabels.pdf"><strong><em>directlabels</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="esquissepack"><em>esquisse</em></h2>
<blockquote>
This addin [esquisse]<span class="citation" data-cites="esquisse">(<a href="#ref-esquisse" role="doc-biblioref">Meyer and Perrier 2025</a>)</span> allows you to <strong>interactively explore your data</strong> by visualizing it with the <strong><em>ggplot2</em></strong> package. It allows you to draw bar plots, curves, scatter plots, histograms, boxplot and sf objects, then export the graph or retrieve the code to reproduce the graph.
<div class="citation">
<strong>Source:</strong> <a href="https://dreamrs.github.io/esquisse/">esquisse</a>
</div>
</blockquote>
<p><a href="https://dreamrs.github.io/esquisse/index.html"><strong><em>esquisse</em> website</strong></a></p>
<p><a href="https://dreamrs.github.io/esquisse/articles/get-started.html"><strong><em>esquisse</em> article: getting started</strong></a></p>
<p><a href="https://dreamrs.github.io/esquisse/articles/shiny-usage.html"><strong><em>esquisse</em> article: shiny usage</strong></a></p>
<p><a href="https://dreamrs.shinyapps.io/esquisse/"><strong><em>esquisse</em> shiny app</strong></a></p>
<p><a href="https://dreamrs.github.io/esquisse/reference/index.html"><strong><em>esquisse</em> functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/esquisse/esquisse.pdf"><strong><em>esquisse</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="fishualizepack"><em>fishualize</em></h2>
<blockquote>
The <strong><em>fishualize</em></strong> <span class="citation" data-cites="fishualize">(<a href="#ref-fishualize" role="doc-biblioref">Schiettekatte, Brandl, and Casey 2022</a>)</span> package provides color scales for plotting in R based on nature’s most stunning and colorful organisms: teleost fishes (with a few chondrichthyan cameos). #TeamFish in its colorful glory
<div class="citation">
<strong>Source:</strong> <a href="https://nschiett.github.io/fishualize/index.html">fishualize</a>
</div>
</blockquote>
<h3 id="available-palettes">Available Palettes</h3>
<blockquote>
<p>The package contains one scale per species, defined by five dominant colors. The number of fishualized species will expand over time. For a list of fish species that are currently available, run <strong>fish_palettes()</strong>.</p>
<p>A visual overview of the color scales can be found <a href="https://nschiett.github.io/fishualize/articles/overview_colors.html"><strong>here</strong></a>.</p>
<div class="citation">
<strong>Source:</strong> <a href="https://nschiett.github.io/fishualize/index.html#the-color-scales">fishualize</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-18"></span>
<img src="images/fishpal1.png" alt="[***fishualize* palette example**](https://nschiett.github.io/fishualize/articles/overview_colors.html)" width="150%" />
<p class="caption">
Figure 10: <a href="https://nschiett.github.io/fishualize/articles/overview_colors.html"><strong><em>fishualize</em> palette example</strong></a>
</p>
</div>
</div>
<details>
<summary>
<strong>All Species Palettes (2/18/25)</strong>
</summary>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<h5 id="species-by-order">Species by Order</h5>
<details>
<summary>
<strong>Acanthuriformes</strong>
</summary>
<p><strong>Acanthuridae</strong><br />
- <strong><em>Acanthurus chirurgus </em></strong><br />
- <strong><em>Acanthurus coeruleus </em></strong><br />
- <strong><em>Acanthurus leucosternon </em></strong><br />
- <strong><em>Acanthurus olivaceus</em></strong><br />
- <strong><em>Acanthurus sohal</em></strong><br />
- <strong><em>Acanthurus triostegus</em></strong><br />
- <strong><em>Naso lituratus</em></strong><br />
- <strong><em>Paracanthurus hepatus</em></strong><br />
- <strong><em>Zebrasoma velifer</em></strong><br />
- <strong><em>Zebrasoma xanthurum</em></strong><br />
</p>
<p><strong>Callanthiidae</strong><br />
- <strong><em>Callanthias australis</em></strong><br />
</p>
<p><strong>Chaetodontidae</strong><br />
- <strong><em>Chaetodon ephippium</em></strong><br />
- <strong><em>Chaetodon larvatus</em></strong><br />
- <strong><em>Chaetodon pelewensis</em></strong><br />
- <strong><em>Chaetodon sedentarius</em></strong><br />
- <strong><em>Prognathodes brasiliensis</em></strong><br />
- <strong><em>Prognathodes guyanensis</em></strong><br />
- <strong><em>Forcipiger longirostris</em></strong><br />
- <strong><em>Hemitaurichthys polylepis</em></strong><br />
</p>
<p><strong>Haemulidae</strong><br />
- <strong><em>Anisotremus virginicus</em></strong><br />
- <strong><em>Haemulon squamipinna</em></strong><br />
- <strong><em>Haemulon plumierii</em></strong><br />
</p>
<p><strong>Lutjanidae</strong><br />
- <strong><em>Lutjanus jocu</em></strong><br />
- <strong><em>Lutjanus sebae</em></strong><br />
</p>
<p><strong>Pomacanthidae</strong><br />
- <strong><em>Chaetodontoplus conspicillatus</em></strong><br />
- <strong><em>Centropyge loricula</em></strong><br />
- <strong><em>Holacanthus ciliaris</em></strong><br />
- <strong><em>Pomacanthus imperator</em></strong><br />
- <strong><em>Pomacanthus paru</em></strong><br />
- <strong><em>Pomacanthus xanthometopon</em></strong><br />
</p>
<p><strong>Priacanthidae</strong><br />
- <strong><em>Heteropriacanthus cruentatus</em></strong><br />
</p>
<p><strong>Zanclidae</strong><br />
- <strong><em>Zanclus cornutus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Anguilliformes</strong>
</summary>
<p><strong>Muraenidae</strong><br />
- <strong><em>Gymnothorax funebris</em></strong><br />
</p>
<p><strong>Ophichthidae</strong><br />
- <strong><em>Myrichthys ocellatus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Atheriniformes</strong>
</summary>
<p><strong>Atherinopsidae</strong><br />
- <strong><em>Atherinella brasilia</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Blenniiformes</strong>
</summary>
<p><strong>Blennidae</strong><br />
- <strong><em>Exallias brevis</em></strong><br />
- <strong><em>Hypleurochilus fissicornis</em></strong><br />
- <strong><em>Hypsoblennius invemar</em></strong><br />
- <strong><em>Parablennius marmoreus</em></strong><br />
- <strong><em>Parablennius pilicornis</em></strong><br />
</p>
<p><strong>Grammatidae</strong><br />
- <strong><em>Gramma brasiliensis</em></strong><br />
- <strong><em>Gramma loreto</em></strong><br />
</p>
<p><strong>Labrisomidae</strong><br />
- <strong><em>Labrisomus cricota</em></strong><br />
- <strong><em>Labrisomus nuchipinnis</em></strong><br />
</p>
<p><strong>Pomacentridae</strong><br />
- <strong><em>Chromis multilineata</em></strong><br />
- <strong><em>Chromis vanderbilti</em></strong><br />
- <strong><em>Hypsypops rubicundus</em></strong><br />
- <strong><em>Stegastes nigricans</em></strong><br />
- <strong><em>Stegastes partitus</em></strong><br />
- <strong><em>Stegastes variabilis</em></strong><br />
</p>
<p><strong>Pseudochromidae</strong><br />
- <strong><em>Pseudochromis aldabraensis </em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Callionymiformes</strong>
</summary>
<p><strong>Callionymidae</strong><br />
- <strong><em>Synchiropus splendidus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Carangiformes</strong>
</summary>
<p><strong>Carangidae</strong><br />
- <strong><em>Elagatis bipinnulata</em></strong><br />
</p>
<p><strong>Coryphaenidae</strong><br />
- <strong><em>Coryphaena hippurus</em></strong><br />
</p>
<p><strong>Pleuronectidae</strong><br />
- <strong><em>Pleuronectes platessa</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Carcharhiniformes</strong>
</summary>
<p><strong>Carcharhinidae</strong><br />
- <strong><em>Prionace glauca</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Centrarchiformes</strong>
</summary>
<p><strong>Centrarchidae</strong><br />
- <strong><em>Lepomis megalotis </em></strong><br />
- <strong><em>Micropterus punctulatus </em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Clupeiformes</strong>
</summary>
<p><strong>Alosidae</strong><br />
- <strong><em>Alosa fallax</em></strong><br />
</p>
<p><strong>Dorosomatidae</strong><br />
- <strong><em>Harengula jaguana</em></strong><br />
- <strong><em>Lile piquitidnga </em></strong><br />
- <strong><em>Opisthonema oglinum </em></strong><br />
- <strong><em>Sardinella brasiliensis </em></strong><br />
</p>
<p><strong>Engraulidae</strong><br />
- <strong><em>Anchoviella lepidentostole</em></strong><br />
- <strong><em>Cetengraulis edentulus</em></strong><br />
- <strong><em>Lycengraulis grossidens</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Cypriniformes</strong>
</summary>
<p><strong>Cyprinidae</strong><br />
- <strong><em>Barbus barbus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Esociformes</strong>
</summary>
<p><strong>Esocidae</strong><br />
- <strong><em>Esox lucius</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Gadiformes</strong>
</summary>
<p><strong>Gadidae</strong><br />
- <strong><em>Gadus morhua</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Gobiiformes</strong>
</summary>
<p><strong>Apogonidae</strong><br />
- <strong><em>Ostorhinchus angustatus </em></strong><br />
</p>
<p><strong>Gobiidae</strong><br />
- <strong><em>Bryaninops natans </em></strong><br />
- <strong><em>Elacatinus figaro</em></strong><br />
- <strong><em>Ostorhinchus angustatus </em></strong><br />
- <strong><em>Nemateleotris magnifica </em></strong><br />
- <strong><em>Neogobius melanostomus </em></strong><br />
- <strong><em>Trimma lantana </em></strong><br />
- <strong><em>Valenciennea strigata </em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Holocentriformes</strong>
</summary>
<p><strong>Holocentridae</strong><br />
- <strong><em>Holocentrus adscensionis</em></strong><br />
- <strong><em>Sargocentron bullisi</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Lampriformes</strong>
</summary>
<p><strong>Lampridae</strong><br />
- <strong><em>Lampris guttatus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Lophiiformes</strong>
</summary>
<p><strong>Antennariidae</strong><br />
- <strong><em>Antennarius commerson</em></strong><br />
- <strong><em>Antennarius multiocellatus</em></strong><br />
- <strong><em>Histiophryne psychedelica</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Mugiliformes</strong>
</summary>
<p><strong>Mugilidae</strong><br />
- <strong><em>Mugil liza </em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Myliobatiformes</strong>
</summary>
<p><strong>Dasyatidae</strong><br />
- <strong><em>Taeniura lymma </em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Orectolobiformes</strong>
</summary>
<p><strong>Ginglymostomatidae</strong><br />
- <strong><em>Ginglymostoma cirratum</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Perciformes</strong>
</summary>
<p><strong>Anthiadidae</strong><br />
- <strong><em>Pronotogrammus martinicensis</em></strong><br />
</p>
<p><strong>Hexagrammidae</strong><br />
- <strong><em>Hexagrammos lagocephalus</em></strong><br />
</p>
<p><strong>Labridae</strong><br />
- <strong><em>Bodianus pulchellus</em></strong><br />
- <strong><em>Bodianus rufus</em></strong><br />
- <strong><em>Cirrhilabrus solorensis</em></strong><br />
- <strong><em>Cirrhilabrus tonozukai</em></strong><br />
- <strong><em>Clepticus brasiliensis</em></strong><br />
- <strong><em>Clepticus parrae</em></strong><br />
- <strong><em>Coris gaimard</em></strong><br />
- <strong><em>Epibulus insidiator </em></strong><br />
- <strong><em>Gomphosus varius </em></strong><br />
- <strong><em>Halichoeres bivittatus</em></strong><br />
- <strong><em>Halichoeres brasiliensis</em></strong><br />
- <strong><em>Halichoeres dimidiatus</em></strong><br />
- <strong><em>Halichoeres garnoti</em></strong><br />
- <strong><em>Halichoeres radiatus</em></strong><br />
- <strong><em>Minilabrus striatus</em></strong><br />
- <strong><em>Pseudocheilinus tetrataenia </em></strong><br />
- <strong><em>Semicossyphus pulcher </em></strong><br />
- <strong><em>Stethojulis bandanensis </em></strong><br />
- <strong><em>Thalassoma bifasciatum</em></strong><br />
- <strong><em>Thalassoma hardwicke</em></strong><br />
- <strong><em>Thalassoma noronhanum</em></strong><br />
- <strong><em>Thalassoma pavo</em></strong><br />
- <strong><em>Xyrichtys novacula </em></strong><br />
</p>
<p><strong>Percidae</strong><br />
- <strong><em>Etheostoma barrenense</em></strong><br />
- <strong><em>Etheostoma spectabile</em></strong><br />
- <strong><em>Sander lucioperca </em></strong><br />
</p>
<p><strong>Scaridae</strong><br />
- <strong><em>Chlorurus microrhinos</em></strong><br />
- <strong><em>Chlorurus spilurus</em></strong><br />
- <strong><em>Scarus ghobban</em></strong><br />
- <strong><em>Scarus globiceps</em></strong><br />
- <strong><em>Scarus hoefleri</em></strong><br />
- <strong><em>Scarus quoyi</em></strong><br />
- <strong><em>Scarus tricolor</em></strong><br />
- <strong><em>Scarus zelindae</em></strong><br />
- <strong><em>Sparisoma frondosum_m</em></strong><br />
- <strong><em>Sparisoma tuyupiranga_f</em></strong><br />
- <strong><em>Sparisoma tuyupiranga_m</em></strong><br />
- <strong><em>Sparisoma viride</em></strong><br />
</p>
<p><strong>Scorpaenidae</strong><br />
- <strong><em>Pterois volitans </em></strong><br />
</p>
<p><strong>Serranidae</strong><br />
- <strong><em>Acanthisthius brasilianus</em></strong><br />
- <strong><em>Cephalopholis argus</em></strong><br />
- <strong><em>Cephalopholis fulva</em></strong><br />
- <strong><em>Dermatolepis inermis</em></strong><br />
- <strong><em>Epinephelus fasciatus</em></strong><br />
- <strong><em>Epinephelus lanceolatus</em></strong><br />
- <strong><em>Epinephelus marginatus</em></strong><br />
- <strong><em>Epinephelus striatus</em></strong><br />
- <strong><em>Hypoplectrus puella </em></strong><br />
- <strong><em>Mycteroperca bonaci </em></strong><br />
- <strong><em>Paralabrax clathratus </em></strong><br />
- <strong><em>Paranthias furcifer </em></strong><br />
- <strong><em>Serranus baldwini</em></strong><br />
- <strong><em>Serranus scriba</em></strong><br />
- <strong><em>Variola louti </em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Petromyzontiformes</strong>
</summary>
<p><strong>Petromyzontidae</strong><br />
- <strong><em>Petromyzon marinus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Rhinopristiformes</strong>
</summary>
<p><strong>Trygonorrhinidae</strong><br />
- <strong><em>Zapteryx brevirostris</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Salmoniformes</strong>
</summary>
<p><strong>Salmonidae</strong><br />
- <strong><em>Oncorhynchus gorbuscha</em></strong><br />
- <strong><em>Oncorhynchus keta</em></strong><br />
- <strong><em>Oncorhynchus kisutch</em></strong><br />
- <strong><em>Oncorhynchus mykiss</em></strong><br />
- <strong><em>Oncorhynchus nerka</em></strong><br />
- <strong><em>Oncorhynchus tshawytscha</em></strong><br />
- <strong><em>Salmo salar</em></strong><br />
- <strong><em>Salmo trutta</em></strong><br />
- <strong><em>Salvelinus fontinalis </em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Scombriformes</strong>
</summary>
<p><strong>Scombridae</strong><br />
- <strong><em>Thunnus obesus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Siluriformes</strong>
</summary>
<p><strong>Loricariidae</strong><br />
- <strong><em>Pareiorhaphis garbei</em></strong><br />
</p>
<p><strong>Pimelodidae</strong><br />
- <strong><em>Phractocephalus hemioliopterus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Syngnathiformes</strong>
</summary>
<p><strong>Aulostomidae</strong><br />
- <strong><em>Aulostomus chinensis </em></strong><br />
</p>
<p><strong>Mullidae</strong><br />
- <strong><em>Parupeneus insularis </em></strong><br />
- <strong><em>Pseudupeneus maculatus </em></strong><br />
</p>
<p><strong>Syngnathidae</strong><br />
- <strong><em>Hippocampus reidi</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Tetraodontiformes</strong>
</summary>
<p><strong>Balistidae</strong><br />
- <strong><em>Balistapus undulatus</em></strong><br />
- <strong><em>Balistes vetula</em></strong><br />
- <strong><em>Balistoides conspicillum</em></strong><br />
- <strong><em>Rhinecanthus aculeatus</em></strong><br />
- <strong><em>Rhinecanthus assasi</em></strong><br />
- <strong><em>Melichthys vidua </em></strong><br />
- <strong><em>Odonus niger </em></strong><br />
</p>
<p><strong>Monacanthidae</strong><br />
- <strong><em>Alterus scriptus</em></strong><br />
- <strong><em>Cantherhines macrocerus</em></strong><br />
- <strong><em>Oxymonacanthus longirostris</em></strong><br />
</p>
<p><strong>Ostraciidae</strong><br />
- <strong><em>Acanthostracion polygonius</em></strong><br />
- <strong><em>Acanthostracion_polygonius_y</em></strong><br />
- <strong><em>Ostracion cubicus</em></strong><br />
- <strong><em>Ostracion whitleyi</em></strong><br />
</p>
</details>
<h4 id="fish-shapes">Fish Shapes</h4>
<blockquote>
<p>fishualize 2.0 enables you to add fish silhouettes to a ggplot object with the function <strong>add_fishape()</strong>. To get an overview of all available silhouettes, run <strong>fishapes()</strong>.</p>
<div class="citation">
<strong>Source:</strong> <a href="https://nschiett.github.io/fishualize/index.html#fish-shapes">fishualize</a>
</div>
</blockquote>
<p><br />
</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-19"></span>
<img src="images/fishape.png" alt="[***fishualize* shape example**](https://nschiett.github.io/fishualize/articles/overview_shapes.html)" width="150%" />
<p class="caption">
Figure 11: <a href="https://nschiett.github.io/fishualize/articles/overview_shapes.html"><strong><em>fishualize</em> shape example</strong></a>
</p>
</div>
</div>
<details>
<summary>
<strong>All Shapes (2/18/25)</strong>
</summary>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<h4 id="shapes-by-order">Shapes by Order</h4>
<details>
<summary>
<strong>Acanthuriformes</strong>
</summary>
<p><strong>Acanthuridae</strong><br />
- <strong><em>Acanthurus nigricauda</em></strong><br />
- <strong><em>Ctenochaetus striatus</em></strong><br />
- <strong><em>Naso lituratus</em></strong><br />
- <strong><em>Naso unicornis</em></strong><br />
- <strong><em>Zebrasoma scopas</em></strong><br />
- <strong><em>Zebrasoma velifer</em></strong><br />
</p>
<p><strong>Chaetodontidae</strong><br />
- <strong><em>Chaetodon auriga</em></strong><br />
- <strong><em>Chaetodon ornatissimus</em></strong><br />
- <strong><em>Chaetodon plebius</em></strong><br />
- <strong><em>Chaetodon trifasciatus</em></strong><br />
- <strong><em>Chaetodon vagabundus</em></strong><br />
</p>
<p><strong>Ephippidae</strong><br />
- <strong><em>Platax teira</em></strong><br />
</p>
<p><strong>Haemulidae</strong><br />
- <strong><em>Plectorhinchus gibbosus</em></strong><br />
</p>
<p><strong>Lethrinidae: </strong><br />
- <strong><em>Lethrinus nebulosus</em></strong><br />
</p>
<p><strong>Lutjanidae</strong><br />
- <strong><em>Lutjanus gibbus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Acipenseriformes</strong>
</summary>
<p><strong>Acipenseridae</strong><br />
- <strong><em>Acipenser fulvescens</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Atheriniformes</strong>
</summary>
<p><strong>Atherinopsidae</strong><br />
- <strong><em>Atherinella brasiliensis</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Batrachoidiformes</strong>
</summary>
<p><strong>Batrachoididae</strong><br />
- <strong><em>Opsanus beta</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Beloniformes</strong>
</summary>
<p><strong>Belonidae</strong><br />
- <strong><em>Belone belone</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Blenniiformes</strong>
</summary>
<p><strong>Blennidae</strong><br />
- <strong><em>Antennablennius adenensis</em></strong><br />
- <strong><em>Ecsenius pulcher</em></strong><br />
- <strong><em>Hypleurochilus pseudoaequipinnis</em></strong><br />
- <strong><em>Lupinoblennius nicholsi</em></strong><br />
- <strong><em>Bothus mancus</em></strong><br />
</p>
<p><strong>Chaenopsidae</strong><br />
- <strong><em>Acanthemblemaria spinosa</em></strong><br />
</p>
<p><strong>Clinidae</strong><br />
- <strong><em>Heteroclinus heptaeolus</em></strong><br />
</p>
<p><strong>Dactyloscopidae</strong><br />
- <strong><em>Gillellus uranidea</em></strong><br />
</p>
<p><strong>Gobiesocidae</strong><br />
- <strong><em>Acyrtus lanthanum</em></strong><br />
</p>
<p><strong>Grammatidae</strong><br />
- <strong><em>Lipogramma barrettorum</em></strong><br />
</p>
<p><strong>Labrisomidae</strong><br />
- <strong><em>Labrisomus guppyi</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Callionymiformes</strong>
</summary>
<p><strong>Callionymidae</strong><br />
- <strong><em>Callionymus bairdi</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Characiformes</strong>
</summary>
<p><strong>Characidae</strong><br />
- <strong><em>Salminus brasiliensis</em></strong><br />
</p>
<p><strong>Erythrinidae</strong><br />
- <strong><em>Hoplias malabaricus</em></strong><br />
</p>
<p><strong>Serrasalmidae</strong><br />
- <strong><em>Colossoma macropomum</em></strong><br />
- <strong><em>Piaractus brachypomus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Carangiformes</strong>
</summary>
<p><strong>Carangidae</strong><br />
- <strong><em>Caranx melampygus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Carcharhiniformes</strong>
</summary>
<p><strong>Carcharhinidae</strong><br />
- <strong><em>Triaenodon obesus</em></strong><br />
</p>
<p><strong>Ctenoluciidae</strong><br />
- <strong><em>Boulengerella cuvieri</em></strong><br />
</p>
<p><strong>Cynodontidae</strong><br />
- <strong><em>Hydrolycus armatus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Centrarchiformes</strong>
</summary>
<p><strong>Centrarchidae</strong><br />
- <strong><em>Lepomis megalotis</em></strong><br />
</p>
<p><strong>Cheilodactylidae</strong><br />
- <strong><em>Cheilodactylus fuscus</em></strong><br />
</p>
<p><strong>Cirrhitidae</strong><br />
- <strong><em>Paracirrhites forsteri</em></strong><br />
</p>
<p><strong>Kyphosidae</strong><br />
- <strong><em>Kyphosus cinerascens</em></strong><br />
- <strong><em>Kyphosus vaigiensis</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Cichliformes</strong>
</summary>
<p><strong>Cichlidae</strong><br />
- <strong><em>Apistogramma hippolytae</em></strong><br />
- <strong><em>Cichla piquiti</em></strong><br />
- <strong><em>Cichla temensis</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Clupeiformes</strong>
</summary>
<p><strong>Alosidae</strong><br />
- <strong><em>Alosa fallax</em></strong><br />
</p>
<p><strong>Dorosomatidae</strong><br />
- <strong><em>Harengula clupeola</em></strong><br />
- <strong><em>Lile piquitinga</em></strong><br />
- <strong><em>Opisthonema oglinum</em></strong><br />
- <strong><em>Sardinella brasiliensis</em></strong><br />
</p>
<p><strong>Engraulidae</strong><br />
- <strong><em>Anchoviella lepidentostole</em></strong><br />
- <strong><em>Cetengraulis edentulus</em></strong><br />
- <strong><em>Lycengraulis grossidens</em></strong><br />
</p>
<p><strong>Spratelloididae</strong><br />
- <strong><em>Jenkinsia lamprotaenia</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Cypriniformes</strong>
</summary>
<p><strong>Cyprinidae</strong><br />
- <strong><em>Barbus barbus</em></strong><br />
- <strong><em>Danio rerio</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Cyprinodontiformes </strong>
</summary>
<p><strong>Fundulidae</strong><br />
- <strong><em>Fundulus grandis</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Esociformes</strong>
</summary>
<p><strong>Esocidae</strong><br />
- <strong><em>Esox lucius</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Gobiiformes</strong>
</summary>
<p><strong>Apogonidae</strong><br />
- <strong><em>Apogon binotatus</em></strong><br />
</p>
<p><strong>Gobiidae</strong><br />
- <strong><em>Coryogalops anomolus</em></strong><br />
- <strong><em>Eviota guttata</em></strong><br />
- <strong><em>Eviota melasma</em></strong><br />
- <strong><em>Fusigobius neophytus</em></strong><br />
- <strong><em>Gnatholepis cauerensis</em></strong><br />
- <strong><em>Gobiodon okinawae</em></strong><br />
- <strong><em>Hetereleotris vulgaris</em></strong><br />
- <strong><em>Priolepis hippoliti</em></strong><br />
- <strong><em>Risor ruber</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Heterodontiformes </strong>
</summary>
<p><strong>Heterodontidae </strong><br />
- <strong><em>Heterodontus portusjacksoni</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Holocentriformes </strong>
</summary>
<p><strong>Holocentridae</strong><br />
- <strong><em>Myripristis violacea</em></strong><br />
- <strong><em>Sargocentron spiniferum</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Lamniformes</strong>
</summary>
<p><strong>Alopiidae</strong><br />
- <strong><em>Alopias vulpinus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Myliobatiformes</strong>
</summary>
<p><strong>Dasyatidae</strong><br />
- <strong><em>Bathytoshia brevicaudata</em></strong><br />
</p>
<p><strong>Mobulidae</strong><br />
- <strong><em>Mobula birostris</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Ophidiiformes</strong>
</summary>
<p><strong>Bythitidae</strong><br />
- <strong><em>Dinematichthys iluocoeteoides</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Orectolobiformes </strong>
</summary>
<p><strong>Ginglymostomatidae</strong><br />
- <strong><em>Nebrius ferrugineus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Perciformes</strong>
</summary>
<p><strong>Channichthyidae</strong><br />
- <strong><em>Chaenocephalus aceratus</em></strong><br />
</p>
<p><strong>Cottidae</strong><br />
- <strong><em>Artedius harringtoni</em></strong><br />
</p>
<p><strong>Labridae</strong><br />
- <strong><em>Chlorurus microrhinus</em></strong><br />
- <strong><em>Chlorurus sordidus</em></strong><br />
- <strong><em>Coris gaimard</em></strong><br />
- <strong><em>Epibulus insidiator</em></strong><br />
- <strong><em>Gomphosus caeruleus</em></strong><br />
- <strong><em>Labroides bicolor</em></strong><br />
- <strong><em>Labroides dimidiatus</em></strong><br />
- <strong><em>Labroides pectoralis</em></strong><br />
- <strong><em>Scarus oviceps</em></strong><br />
- <strong><em>Scarus rivulatus</em></strong><br />
- <strong><em>Scarus schlegeli</em></strong><br />
</p>
<p><strong>Liparidae </strong><br />
- <strong><em>Careproctus staufferi</em></strong><br />
</p>
<p><strong>Scaridae</strong><br />
- <strong><em>Sparisoma cretense</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Syngnathiformes</strong>
</summary>
<p><strong>Aulostomidae</strong><br />
- <strong><em>Aulostomus chinensis</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Tetraodontiformes</strong>
</summary>
<p><strong>Balistidae</strong><br />
- <strong><em>Balistapus undulatus</em></strong><br />
- <strong><em>Odonus niger</em></strong><br />
</p>
<p><strong>Diodontidae</strong><br />
- <strong><em>Dicotylichthys punctulatus</em></strong><br />
</p>
</details>
<details>
<summary>
<strong>Torpediniformes</strong>
</summary>
<p><strong>Hypnidae</strong><br />
- <strong><em>Hypnos monopterygius</em></strong><br />
</p>
</details>
<p><br />
</p>
<p><a href="https://nschiett.github.io/fishualize/index.html"><strong><em>fishualize</em> website</strong></a></p>
<p><a href="https://nschiett.github.io/fishualize/articles/overview_shapes.html"><strong>fish shapes overview</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/fishualize/fishualize.pdf"><strong><em>fishualize</em> reference manual</strong></a></p>
<p><a href="https://nschiett.github.io/fishualize/articles/contribute.html"><strong>contribute to <em>fishualize</em></strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="geomtextpathpack"><em>geomtextpath</em></h2>
<h3 id="create-curved-text-in-ggplot2">Create curved text in <em>ggplot2</em></h3>
<blockquote>
<p>The existing text-based geom layers in <strong><em>ggplot2</em></strong> (<strong><em>geom_text</em></strong> and <strong><em>geom_label</em></strong>) are ideal for the majority of plots, since typically textual annotations are short, straight and in line with the axes. However, there are some occasions when it is useful to have text follow a curved path. This may be to create or recreate a specific visual effect, or it may be to label a circular / polar plot in a more “natural” way. Direct and automatic text labels that adhere to their associated line can also provide a neat alternative to legends, without the need for specifying exact label positions, and with a lower risk of overplotting.</p>
Using <strong><em>geomtextpath</em></strong> <span class="citation" data-cites="geomtextpath">(<a href="#ref-geomtextpath" role="doc-biblioref">Cameron and van den Brand 2022</a>)</span>, your text can follow any path, and will remain correctly spaced and angled, even if you change the size and aspect ratio of your plotting device. It does so without the need to redraw your plot each time, as shown in the introduction vignette.
<div class="citation">
<strong>Source:</strong> <a href="https://allancameron.github.io/geomtextpath/">geomtextpath</a>
</div>
</blockquote>
<p><a href="https://allancameron.github.io/geomtextpath/index.html"><strong><em>geomtextpath</em> website</strong></a></p>
<p><a href="https://allancameron.github.io/geomtextpath/articles/geomtextpath.html"><strong><em>geomtextpath</em> article: introduction</strong></a></p>
<p><a href="https://allancameron.github.io/geomtextpath/articles/aesthetics.html"><strong><em>geomtextpath</em> article: aesthetics, parameters, and scales</strong></a></p>
<p><a href="https://allancameron.github.io/geomtextpath/articles/curved_polar.html"><strong><em>geomtextpath</em> article: curved text in polar coordinates</strong></a></p>
<p><a href="https://allancameron.github.io/geomtextpath/articles/gallery.html"><strong><em>geomtextpath</em> article: gallery of examples</strong></a></p>
<p><a href="https://allancameron.github.io/geomtextpath/reference/index.html"><strong><em>geomtextpath</em> functions</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/geomtextpath/geomtextpath.pdf"><strong><em>geomtextpath</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-20"></span>
<img src="images/geomtext.png" alt="[***geomtextpath* example**](https://allancameron.github.io/geomtextpath/)" width="150%" />
<p class="caption">
Figure 12: <a href="https://allancameron.github.io/geomtextpath/"><strong><em>geomtextpath</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="gganimatepack"><em>gganimate</em></h2>
<blockquote>
<strong><em>gganimate</em></strong> <span class="citation" data-cites="gganimate">(<a href="#ref-gganimate" role="doc-biblioref">Pedersen and Robinson 2024</a>)</span> extends the grammar of graphics as implemented by <strong><em>ggplot2</em></strong> to include the description of animation. It does this by providing a range of new grammar classes that can be added to the plot object in order to customise how it should change with time.
<div class="citation">
<strong>Source:</strong> <a href="https://gganimate.com/index.html">gganimate</a>
</div>
</blockquote>
<p><a href="https://gganimate.com/"><strong><em>gganimate</em> website</strong></a></p>
<p><a href="https://gganimate.com/articles/gganimate.html"><strong><em>gganimate</em> article: getting started</strong></a></p>
<p><a href="https://gganimate.com/reference/index.html"><strong><em>gganimate</em> functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/gganimate/gganimate.pdf"><strong><em>gganimate</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/gganimate1.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-22"></span>
<img src="images/gganimate2.png" alt="[***gganimate*** **cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/gganimate.pdf)" width="1000" />
<p class="caption">
Figure 13: <a href="https://github.com/rstudio/cheatsheets/blob/main/gganimate.pdf"><strong><em>gganimate</em></strong> <strong>cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggbreakpack"><em>ggbreak</em></h2>
<h3 id="set-axis-break-for-ggplot2">Set Axis Break for <em>ggplot2</em></h3>
<blockquote>
<a href="#ggbreakpack"><em>ggbreak</em></a> <span class="citation" data-cites="ggbreak">(<a href="#ref-ggbreak" role="doc-biblioref">Shuangbin Xu et al. 2021</a>)</span> - An implementation of scale functions for setting axis breaks of a ‘gg’ plot.
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/YuLab-SMU/ggbreak?tab=readme-ov-file">ggbreak</a>
</div>
</blockquote>
<p><a href="https://cran.r-project.org/web/packages/ggbreak/vignettes/ggbreak.html"><strong><em>ggbreak</em> website</strong></a></p>
<p><a href="https://github.com/YuLab-SMU/ggbreak/blob/master/vignettes/ggbreak.Rmd"><strong><em>ggbreak</em> vignette</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/ggbreak/ggbreak.pdf"><strong><em>ggbreak</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-23"></span>
<img src="images/ggbreak.png" alt="[***ggbreak* example**](https://cran.r-project.org/web/packages/ggbreak/vignettes/ggbreak.html)" width="150%" />
<p class="caption">
Figure 14: <a href="https://cran.r-project.org/web/packages/ggbreak/vignettes/ggbreak.html"><strong><em>ggbreak</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggbumppack"><em>ggbump</em></h2>
<h3 id="bump-chart-and-sigmoid-curves">Bump Chart and Sigmoid Curves</h3>
<blockquote>
The R package <strong><em>ggbump</em></strong> <span class="citation" data-cites="ggbump">(<a href="#ref-ggbump" role="doc-biblioref">D. Sjoberg 2020</a>)</span> creates elegant bump charts in <strong><em>ggplot</em></strong>. Bump charts are good to use to plot ranking over time, or other examples when the path between two nodes have no statistical significance. Also includes functions to create custom smooth lines called sigmoid curves.
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/davidsjoberg/ggbump">ggbump</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-24"></span>
<img src="images/ggbump.png" alt="***ggbump* example from [ documentation](https://github.com/davidsjoberg/ggbump/)**" width="3600" />
<p class="caption">
Figure 15: <strong><em>ggbump</em> example from <a href="https://github.com/davidsjoberg/ggbump/">documentation</a></strong>
</p>
</div>
</div>
<p><a href="https://github.com/davidsjoberg/ggbump/tree/master"><strong><em>ggbump</em> GitHub</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/ggbump/ggbump.pdf"><strong><em>ggbump</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggeasypack"><em>ggeasy</em></h2>
<blockquote>
<a href="#ggeasypack"><em>ggeasy</em></a> <span class="citation" data-cites="ggeasy">(<a href="#ref-ggeasy" role="doc-biblioref">Carroll, Schep, and Sidi 2024</a>)</span> Provides a series of aliases to commonly used but difficult to remember <strong><em>ggplot2</em></strong> sequences.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/ggeasy/ggeasy.pdf">ggeasy</a>
</div>
</blockquote>
<p><a href="https://jonocarroll.github.io/ggeasy/"><strong><em>ggeasy</em> website</strong></a></p>
<p><a href="https://jonocarroll.github.io/ggeasy/reference/index.html"><strong><em>ggeasy</em> functions</strong></a></p>
<p><a href="https://jonocarroll.github.io/ggeasy/articles/shortcuts.html"><strong><em>ggeasy</em> article: <em>ggplot2</em> shortcuts</strong></a></p>
<p><a href="https://jonocarroll.github.io/ggeasy/articles/tests_and_coverage.htmlcit"><strong><em>ggeasy</em> article: tests and coverage</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/ggeasy/ggeasy.pdf"><strong><em>ggeasy</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-25"></span>
<img src="images/ggeasy.png" alt="[***ggeasy* example**](https://jonocarroll.github.io/ggeasy/articles/shortcuts.html)" width="768" />
<p class="caption">
Figure 16: <a href="https://jonocarroll.github.io/ggeasy/articles/shortcuts.html"><strong><em>ggeasy</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggforcepack"><em>ggforce</em></h2>
<blockquote>
The aim of <strong><em>ggplot2</em></strong> is to aid in visual data investigations. This focus has led to a lack of facilities for composing specialised plots. <strong><em>ggforce</em></strong> <span class="citation" data-cites="ggforce">(<a href="#ref-ggforce" role="doc-biblioref">Pedersen 2024a</a>)</span> aims to be a collection of mainly new stats and geoms that fills this gap. All additional functionality is aimed to come through the official extension system so using <strong><em>ggforce</em></strong> should be a stable experience.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/ggforce/index.html">ggforce</a>
</div>
</blockquote>
<p><a href="https://ggforce.data-imaginist.com/"><strong><em>ggforce</em> website</strong></a></p>
<p><a href="https://ggforce.data-imaginist.com/reference/index.html"><strong><em>ggforce</em> functions</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/ggforce/ggforce.pdf"><strong><em>ggforce</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-26"></span>
<img src="images/ggforce.png" alt="[***ggforce* example**](https://rviews.rstudio.com/2019/09/19/intro-to-ggforce/)" width="1152" />
<p class="caption">
Figure 17: <a href="https://rviews.rstudio.com/2019/09/19/intro-to-ggforce/"><strong><em>ggforce</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="gghighlightpack"><em>gghighlight</em></h2>
<h3 id="highlight-lines-and-points-in-ggplot2">Highlight Lines and Points in <em>ggplot2</em></h3>
<blockquote>
The <strong><em>gghighlight</em></strong> <span class="citation" data-cites="gghighlight">(<a href="#ref-gghighlight" role="doc-biblioref">Yutani 2024</a>)</span> package in R is an extension of the <strong><em>ggplot2</em></strong> package, designed to simplify the process of highlighting elements.
<div class="citation">
<strong>Source:</strong> <a href="https://r-graph-gallery.com/package/gghighlight.html">R Graph Gallery</a>
</div>
</blockquote>
<p><a href="https://yutannihilation.github.io/gghighlight/index.html"><strong><em>gghighlight</em> website</strong></a></p>
<p><a href="https://yutannihilation.github.io/gghighlight/articles/gghighlight.html"><strong><em>gghighlight</em> article: introduction</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/gghighlight/gghighlight.pdf"><strong><em>gghighlight</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/gghighmessy.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-28"></span>
<img src="images/gghighclean.png" alt="[***gghighlight* example**](https://rviews.rstudio.com/2019/09/19/intro-to-ggforce/)" width="150%" />
<p class="caption">
Figure 18: <a href="https://rviews.rstudio.com/2019/09/19/intro-to-ggforce/"><strong><em>gghighlight</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggiraphpack"><em>ggiraph</em></h2>
<h3 id="make-ggplot2-dynamic-and-interactive">Make <em>ggplot2</em> dynamic and interactive</h3>
<blockquote>
The <strong><em>ggiraph</em></strong> <span class="citation" data-cites="ggiraph">(<a href="#ref-ggiraph" role="doc-biblioref">Gohel and Skintzos 2024</a>)</span> package in R is an extension of the <strong><em>ggplot2</em></strong> package, designed to simplify the process of creating interactive and dynamic graphs. It’s an htmlwidget, which means it’s highly compatible with RMarkdown/Quarto document and Shiny application.
<div class="citation">
<strong>Source:</strong> <a href="https://r-graph-gallery.com/package/ggiraph.html">R Graph Gallery</a>
</div>
</blockquote>
<p><a href="https://davidgohel.github.io/ggiraph/index.html"><strong><em>ggiraph</em> website</strong></a></p>
<p><a href="https://ardata.fr/ggiraph-book/"><strong><em>ggiraph</em> book</strong></a></p>
<p><a href="https://davidgohel.github.io/ggiraph/reference/index.html"><strong><em>ggiraph</em> functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/ggiraph/ggiraph.pdf"><strong><em>ggiraph</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggpatternpack"><em>ggpattern</em></h2>
<blockquote>
[ggpattern] <span class="citation" data-cites="ggpattern">(<a href="#ref-ggpattern" role="doc-biblioref">FC, Davis, and ggplot2 authors 2025</a>)</span> Provides <strong><em>ggplot2</em></strong> geoms filled with various patterns. Includes a patterned version of every <strong><em>ggplot2</em></strong> geom that has a region that can be filled with a pattern. Provides a suite of <strong><em>ggplot2</em></strong> aesthetics and scales for controlling pattern appearances. Supports over a dozen builtin patterns (every pattern implemented by <strong><em>gridpattern</em></strong>) as well as allowing custom user-defined patterns.
<div class="citation">
<strong>Source:</strong> <a href="https://cloud.r-project.org/web/packages/ggpattern/index.html">ggpattern</a>
</div>
</blockquote>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/"><strong><em>ggpattern</em> website</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/create-logo.html"><strong><em>ggpattern</em> article: creating the logo in R</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/developing-patterns.html"><strong><em>ggpattern</em> article: developing patterns</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/geom-gallery-array.html"><strong><em>ggpattern</em> article: geom gallery - array-based patterns</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/geom-gallery-geometry.html"><strong><em>ggpattern</em> article: geom gallery - geometry-based patterns</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/geometry-based-pattern-parameters.html"><strong><em>ggpattern</em> article: parameters for geometry-based patterns</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/gganimate.html"><strong><em>ggpattern</em> article: animating patterns with <em>gganimate</em></strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/patterns-image.html"><strong><em>ggpattern</em> article: patterns ‘image’ and ‘placeholder’ parameters and examples</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/patterns-noise.html"><strong><em>ggpattern</em> article: patterns ‘gradient’, ‘ambient’, and ‘plasma’ patterns</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/patterns-points.html"><strong><em>ggpattern</em> article: patterns ‘circle’, ‘pch’, ‘regular_polygon’, and ‘rose’ parameters and examples</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/patterns-stripes.html"><strong><em>ggpattern</em> article: patterns ‘stripe’, ‘wave’, ‘crosshatch’, and ‘weave’ parameters and examples</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/articles/patterns-tilings.html"><strong><em>ggpattern</em> article: patterns ‘magick’ and ‘polygon_tiling’ parameters and examples</strong></a></p>
<p><a href="https://trevorldavis.com/R/ggpattern/dev/reference/index.html"><strong><em>ggpattern</em> functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/ggpattern/ggpattern.pdf"><strong><em>ggpattern</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-29"></span>
<img src="images/gggpattern.jpg" alt="[***ggpattern* example**](https://trevorldavis.com/R/ggpattern/dev/)" width="150%" />
<p class="caption">
Figure 19: <a href="https://trevorldavis.com/R/ggpattern/dev/"><strong><em>ggpattern</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggraphpack"><em>ggraph</em></h2>
<blockquote>
<strong><em>ggraph</em></strong> <span class="citation" data-cites="ggraph">(<a href="#ref-ggraph" role="doc-biblioref">Pedersen 2024b</a>)</span> is an extension of <strong><em>ggplot2</em></strong> aimed at supporting relational data structures such as networks, graphs, and trees. While it builds upon the foundation of <strong><em>ggplot2</em></strong> and its API it comes with its own self-contained set of geoms, facets, etc., as well as adding the concept of layouts to the grammar
<div class="citation">
<strong>Source:</strong> <a href="https://ggraph.data-imaginist.com/index.html">ggraph</a>
</div>
</blockquote>
<p><a href="https://ggraph.data-imaginist.com/index.html"><strong><em>ggraph</em> website</strong></a></p>
<p><a href="https://ggraph.data-imaginist.com/reference/index.html"><strong><em>ggraph</em> functions</strong></a></p>
<p><a href="https://ggraph.data-imaginist.com/articles/Layouts.html"><strong><em>ggraph</em> article: layouts</strong></a></p>
<p><a href="https://ggraph.data-imaginist.com/articles/Nodes.html"><strong><em>ggraph</em> article: nodes</strong></a></p>
<p><a href="https://ggraph.data-imaginist.com/articles/Edges.html"><strong><em>ggraph</em> article: edges</strong></a></p>
<p><a href="https://ggraph.data-imaginist.com/articles/tidygraph.html"><strong><em>ggraph</em> article: tidygraph interface</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/ggraph/ggraph.pdf"><strong><em>ggraph</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-30"></span>
<img src="images/ggraph.png" alt="[***ggraph* example**](https://ggraph.data-imaginist.com/articles/tidygraph.html)" width="150%" />
<p class="caption">
Figure 20: <a href="https://ggraph.data-imaginist.com/articles/tidygraph.html"><strong><em>ggraph</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggrepelpack"><em>ggrepel</em></h2>
<blockquote>
<p><strong><em>ggrepel</em></strong> <span class="citation" data-cites="ggrepel">(<a href="#ref-ggrepel" role="doc-biblioref">Slowikowski 2024</a>)</span> provides geoms for <strong><em>ggplot2</em></strong> to repel overlapping text labels:</p>
<ul>
<li><strong>geom_text_repel()</strong><br />
</li>
<li><strong>geom_label_repel()</strong></li>
</ul>
Text labels repel away from each other, away from data points, and away from edges of the plotting area.
<div class="citation">
<strong>Source:</strong> <a href="https://ggrepel.slowkow.com/">ggrepel</a>
</div>
</blockquote>
<p><a href="https://ggrepel.slowkow.com/"><strong><em>ggrepel</em> website</strong></a></p>
<p><a href="https://ggrepel.slowkow.com/articles/examples"><strong><em>ggrepel</em> examples</strong></a></p>
<p><a href="https://ggrepel.slowkow.com/reference/"><strong><em>ggrepel</em> functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/ggrepel/ggrepel.pdf"><strong><em>ggrepel</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/ggrepel1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-32"></span>
<img src="images/ggrepel2.png" alt="[***ggrepel* example**](https://ggrepel.slowkow.com/articles/examples#hide-some-of-the-labels)" width="150%" />
<p class="caption">
Figure 21: <a href="https://ggrepel.slowkow.com/articles/examples#hide-some-of-the-labels"><strong><em>ggrepel</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggsankeyfierpack"><em>ggsankeyfier</em></h2>
<blockquote>
The goal of <strong><em>ggsankey</em></strong> <span class="citation" data-cites="ggsankey">(<a href="#ref-ggsankey" role="doc-biblioref">de Vries 2024</a>)</span> is to make beautiful sankey, alluvial and sankey bump plots in <strong><em>ggplot2</em></strong>
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/davidsjoberg/ggsankey">ggsankey</a>
</div>
</blockquote>
<p><a href="https://github.com/davidsjoberg/ggsankey"><strong><em>ggsankey</em> website</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/ggsankeyfier/ggsankeyfier.pdf"><strong><em>ggsankey</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-33"></span>
<img src="images/ggsankey.png" alt="[***ggsankey* example**](https://github.com/davidsjoberg/ggsankey)" width="150%" />
<p class="caption">
Figure 22: <a href="https://github.com/davidsjoberg/ggsankey"><strong><em>ggsankey</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggstatsplotpack"><em>ggstatsplot</em></h2>
<h3 id="ggplot2-based-plots-with-statistical-details"><strong><em>ggplot2</em></strong> Based Plots with Statistical Details</h3>
<blockquote>
<strong><em>ggstatsplot</em></strong> <span class="citation" data-cites="ggstatsplot">(<a href="#ref-ggstatsplot" role="doc-biblioref">Patil 2021</a>)</span> is an extension of <strong><em>ggplot2</em></strong> package for creating graphics with details from statistical tests included in the information-rich plots themselves. In a typical exploratory data analysis workflow, data visualization and statistical modeling are two different phases: visualization informs modeling, and modeling in its turn can suggest a different visualization method, and so on and so forth. The central idea of <strong><em>ggstatsplot</em></strong> is simple: combine these two phases into one in the form of graphics with statistical details, which makes data exploration simpler and faster.
<div class="citation">
<strong>Source:</strong> <a href="https://indrajeetpatil.github.io/ggstatsplot/">ggstatsplot</a>
</div>
</blockquote>
<p><a href="https://indrajeetpatil.github.io/ggstatsplot/"><strong><em>ggstatsplot</em> website</strong></a></p>
<p><a href="https://indrajeetpatil.github.io/ggstatsplot/#summary-of-available-plots"><strong><em>ggstatsplot</em> plot list</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/ggstatsplot/ggstatsplot.pdf"><strong><em>ggstatsplot</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-34"></span>
<img src="images/ggstatplot.png" alt="[***ggstatsplot* example**](https://indrajeetpatil.github.io/ggstatsplot/index.html)" width="150%" />
<p class="caption">
Figure 23: <a href="https://indrajeetpatil.github.io/ggstatsplot/index.html"><strong><em>ggstatsplot</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggtextpack"><em>ggtext</em></h2>
<blockquote>
The <strong><em>ggtext</em></strong> <span class="citation" data-cites="ggtext">(<a href="#ref-ggtext" role="doc-biblioref">Wilke and Wiernik 2025</a>)</span> package provides simple Markdown and HTML rendering for <strong><em>ggplot2</em></strong>. Under the hood, the package uses the <strong><em>gridtext</em></strong> package for the actual rendering, and consequently it is limited to the feature set provided by <strong><em>gridtext</em></strong>.
<div class="citation">
<strong>Source:</strong> <a href="https://wilkelab.org/ggtext/">ggtext</a>
</div>
</blockquote>
<p><a href="https://wilkelab.org/ggtext/"><strong><em>ggtext</em> website</strong></a></p>
<p><a href="https://wilkelab.org/ggtext/articles/plotting_text.html"><strong><em>ggtext</em> article: plotting with markdown text</strong></a></p>
<p><a href="https://wilkelab.org/ggtext/articles/theme_elements.html"><strong><em>ggtext</em> article: markdown theme elements</strong></a></p>
<p><a href="https://wilkelab.org/ggtext/reference/index.html"><strong><em>ggtext</em> functions</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/ggtext/ggtext.pdf"><strong><em>ggtext</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-35"></span>
<img src="images/ggtext.png" alt="[***ggtext* example**](https://wilkelab.org/ggtext/articles/plotting_text.html)" width="150%" />
<p class="caption">
Figure 24: <a href="https://wilkelab.org/ggtext/articles/plotting_text.html"><strong><em>ggtext</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggthemespack"><em>ggthemes</em></h2>
<blockquote>
Some extra geoms, scales, and themes for <strong><em>ggplot</em></strong>. <span class="citation" data-cites="ggthemes">(<a href="#ref-ggthemes" role="doc-biblioref">Arnold 2024</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://jrnold.github.io/ggthemes/">ggthemes</a>
</div>
</blockquote>
<p><a href="https://jrnold.github.io/ggthemes/"><strong><em>ggthemes</em> website</strong></a></p>
<p><a href="https://jrnold.github.io/ggthemes/reference/index.html"><strong><em>ggthemes</em> functions</strong></a></p>
<p><a href="https://rafalab.dfci.harvard.edu/dsbook/ggplot2.html#add-on-packages"><strong><em>ggthemes</em> tutorial</strong></a></p>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/"><strong><em>ggthemes</em> examples</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/ggthemes/ggthemes.pdf"><strong><em>ggthemes</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-36"></span>
<img src="images/ggtheme.png" alt="[***ggthemes* example**](https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/)" width="150%" />
<p class="caption">
Figure 25: <a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/"><strong><em>ggthemes</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="ggwordcloudpack"><em>ggwordcloud</em></h2>
<blockquote>
<strong><em>ggwordcloud</em></strong> <span class="citation" data-cites="ggwordcloud">(<a href="#ref-ggwordcloud" role="doc-biblioref">Le Pennec and Slowikowski 2024</a>)</span> provides a word cloud text geom for <strong><em>ggplot2</em></strong>. The placement algorithm implemented in C++ is an hybrid between the one of <strong><em>wordcloud</em></strong> and the one of <strong><em>wordcloud2.js</em></strong>. The cloud can grow according to a shape and stay within a mask. The size aesthetic is used either to control the font size or the printed area of the words. <strong><em>ggwordcloud</em></strong> also supports arbitrary text rotation. The faceting scheme of <strong><em>ggplot2</em></strong> can also be used. Two functions meant to be the equivalent of <strong><em>wordcloud</em></strong> and <strong><em>wordcloud2</em></strong> are proposed. Last but not least you can use <strong><em>gridtext</em></strong> markdown/html syntax in the labels
<div class="citation">
<strong>Source:</strong> <a href="">ggwordcloud</a>
</div>
</blockquote>
<p><a href="https://lepennec.github.io/ggwordcloud/"><strong><em>ggwordcloud</em> website</strong></a></p>
<p><a href="https://lepennec.github.io/ggwordcloud/reference/index.html"><strong><em>ggwordcloud</em> functions</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/ggwordcloud/ggwordcloud.pdf"><strong><em>ggwordcloud</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-37"></span>
<img src="images/ggwordcloud.png" alt="[***ggwordcloud* example**](https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/)" width="150%" />
<p class="caption">
Figure 26: <a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/"><strong><em>ggwordcloud</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="hrbrthemespack"><em>hrbrthemes</em></h2>
<blockquote>
<p>The <strong><em>hrbrthemes</em></strong> <span class="citation" data-cites="hrbrthemes">(<a href="#ref-hrbrthemes" role="doc-biblioref">Rudis 2024</a>)</span> package in R serves as a valuable extension to the renowned <strong><em>ggplot2</em></strong> package. Created by Bob Rudis, <strong><em>hrbrthemes</em></strong> offers a collection of opinionated themes and typography settings that aim to elevate the visual appeal of your <strong><em>ggplot2</em></strong> plots. Whether you are a data scientist, a researcher, or anyone interested in data visualization, this package helps you create publication-quality plots with minimal effort.</p>
The package is particularly useful for those who want to create publication-quality plots without spending too much time on customization. It provides a set of themes that are not only visually appealing but also adhere to good visualization practices.
<div class="citation">
<strong>Source:</strong> <a href="https://r-graph-gallery.com/package/hrbrthemes.html">R Graph Gallery</a>
</div>
</blockquote>
<blockquote>
<p>This is a very focused package that provides typography-centric themes and theme components for <strong><em>ggplot2</em></strong>.</p>
The core theme: <strong>theme_ipsum</strong> (“ipsum” is Latin for “precise”) uses Arial Narrow which should be installed on practically any modern system, so it’s “free”-ish. This font is condensed, has solid default kerning pairs and geometric numbers. That’s what I consider the “font trifecta” must-have for charts. An additional quality for fonts for charts is that they have a diversity of weights. Arial Narrow (the one on most systems, anyway) does not have said diversity but this quality is not (IMO) a “must have”.
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/hrbrmstr/hrbrthemes">hrbrthemes</a>
</div>
</blockquote>
<p><a href="https://github.com/hrbrmstr/hrbrthemes"><strong><em>hrbrthemes</em> GitHub</strong></a></p>
<p><a href="https://hrbrmstr.github.io/hrbrthemes/"><strong><em>hrbrthemes</em> website</strong></a></p>
<p><a href="https://hrbrmstr.github.io/hrbrthemes/reference/index.html"><strong><em>hrbrthemes</em> functions</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/hrbrthemes/hrbrthemes.pdf"><strong><em>hrbrthemes</em> reference manual</strong></a></p>
<p><a href="https://r-graph-gallery.com/package/hrbrthemes.html"><strong>elevate your <em>ggplot2</em> visuals with <em>hrbrthemes</em></strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-38"></span>
<img src="images/hrbrthemes.png" alt="[***hrbrthemes* example**](https://github.com/hrbrmstr/hrbrthemes)" width="150%" />
<p class="caption">
Figure 27: <a href="https://github.com/hrbrmstr/hrbrthemes"><strong><em>hrbrthemes</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="patchworkpack"><em>patchwork</em></h2>
<blockquote>
The <strong><em>ggplot2</em></strong> package provides a strong API for sequentially building up a plot, but does not concern itself with composition of multiple plots. <strong><em>patchwork</em></strong> <span class="citation" data-cites="patchwork">(<a href="#ref-patchwork" role="doc-biblioref">Pedersen 2024c</a>)</span> is a package that expands the API to allow for arbitrarily complex composition of plots by, among others, providing mathematical operators for combining multiple plots.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/patchwork/patchwork.pdf">patchwork</a>
</div>
</blockquote>
<p>Use this package to modify figures made with <strong><em>ggplot2</em></strong>, to get a more aesthetically appealing figure.</p>
<p><a href="https://patchwork.data-imaginist.com/"><strong><em>patchwork</em> website</strong></a></p>
<p><a href="https://patchwork.data-imaginist.com/articles/patchwork.html"><strong><em>patchwork</em></strong> <strong>tutorial: getting started</strong></a></p>
<p><a href="https://patchwork.data-imaginist.com/articles/guides/assembly.html"><strong><em>patchwork</em></strong> <strong>tutorial: plot Assembly</strong></a></p>
<p><a href="https://patchwork.data-imaginist.com/articles/guides/layout.html"><strong><em>patchwork</em></strong> <strong>tutorial: controlling Layouts</strong></a></p>
<p><a href="https://patchwork.data-imaginist.com/articles/guides/annotation.html"><strong><em>patchwork</em></strong> <strong>tutorial: adding annotation and style</strong></a></p>
<p><a href="https://patchwork.data-imaginist.com/articles/guides/multipage.html"><strong><em>patchwork</em></strong> <strong>tutorial: alignment across multiple Pages</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/patchwork/patchwork.pdf"><strong><em>patchwork</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-39"></span>
<img src="images/patchwork.png" alt="[***patchwork* example**](https://patchwork.data-imaginist.com/)" width="150%" />
<p class="caption">
Figure 28: <a href="https://patchwork.data-imaginist.com/"><strong><em>patchwork</em> example</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="paletteerpack"><em>paletteer</em></h2>
<h3 id="comprehensive-collection-of-color-palettes">Comprehensive Collection of Color Palettes</h3>
<blockquote>
The choices of color palettes in R can be quite overwhelming with palettes spread over many packages with many different APIs. <strong><em>paletteer</em></strong> <span class="citation" data-cites="paletteer">(<a href="#ref-paletteer" role="doc-biblioref">Hvitfeldt 2021</a>)</span> aims to collect all color palettes across the R ecosystem under the same package with a streamlined API.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/paletteer/index.html">paletteer</a>
</div>
</blockquote>
<h3 id="palettes">Palettes</h3>
<blockquote>
<p>The palettes are divided into 2 groups; <strong>discrete</strong> and <strong>continuous</strong>. For <strong>discrete palette you have the choice between the fixed width palettes and dynamic palettes</strong>. Most common of the two are the fixed width palettes which have a set amount of colors which doesn’t change when the number of colors requested vary like the following palettes:</p>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/paletteer1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<blockquote>
<p>on the other hand we have the dynamic palettes where the colors of the palette depend on the number of colors you need like the green.pal palette from the cartography package:</p>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/paletteer2.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<blockquote>
<p>Lastly we have the continuous palettes which provides as many colors as you need for a smooth transition of color:</p>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/paletteer3.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<blockquote>
This package includes <strong>2759 palettes from 75 different packages</strong> and information about these can be found in the following data.frames: <strong>palettes_c_names, palettes_d_names</strong> and <strong>palettes_dynamic_names</strong>. Additionally <a href="https://github.com/EmilHvitfeldt/r-color-palettes"><strong>this github repo</strong></a> showcases all the palettes included in the package and more.
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/EmilHvitfeldt/paletteer?tab=readme-ov-file#palettes"><strong>paletteer</strong></a>
</div>
</blockquote>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h3 id="palette-explorer">Palette Explorer</h3>
<blockquote>
A <a href="https://r-graph-gallery.com/color-palette-finder"><strong>web application</strong></a> is available to quickly explore the available palettes.
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/EmilHvitfeldt/paletteer?tab=readme-ov-file#palette-explorer"><strong>paletteer</strong></a>
</div>
</blockquote>
<blockquote>
<p><strong><em>paletteer</em></strong> is one of my favorite R packages. It consolidates all available color palettes in the R ecosystem into a single package. This means you can access over 2500 palettes with just one line of code.</p>
<p>The palettes are categorized as <strong>continuous</strong> or <strong>qualitative</strong>. The R Graph Gallery has a <a href="https://r-graph-gallery.com/package/paletteer.html"><strong>dedicated page</strong></a> to help you get started.</p>
<p>To use <strong>paletteer</strong>, the basic syntax looks like this:</p>
</blockquote>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="citation">
<strong>Source:</strong> <a href="https://r-graph-gallery.com/color-palette-finder"><strong>Color Palette Finder</strong></a>
</div>
<p><br />
This webpage defaults to a palette called Acadia. If you click the name it opens to a dropdown where you can select other palettes, and then see examples of graphics made using that palette to test it before you take time building your own. There is also a dropdown that allows you to view the palettes with colorblind filters on to ensure that the palette will be legible.</p>
<p><a href="https://github.com/EmilHvitfeldt/paletteer"><strong><em>paletteer</em> website</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/paletteer/paletteer.pdf"><strong><em>paletteer</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h3 id="available-palettes-1">Available Palettes</h3>
<p>Courtesy of <a href="https://emilhvitfeldt.github.io/r-color-palettes/"><strong>R Color Palettes</strong></a></p>
<details>
<summary>
<strong><em>ButterflyColors</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-43"></span>
<img src="images/butterflycolors.png" alt="[***ButterflyColors***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ButterflyColors)" width="150%" />
<p class="caption">
Figure 29: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ButterflyColors"><strong><em>ButterflyColors</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>DresdenColor</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-44"></span>
<img src="images/DresdenColor.png" alt="[***DresdenColor***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20DresdenColor)" width="150%" />
<p class="caption">
Figure 30: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20DresdenColor"><strong><em>DresdenColor</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>IslamicArt</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-45"></span>
<img src="images/IslamicArt.png" alt="[***IslamicArt***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20IslamicArt)" width="150%" />
<p class="caption">
Figure 31: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20IslamicArt"><strong><em>IslamicArt</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>LaCroixColoR</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-46"></span>
<img src="images/LaCroixColoR.png" alt="[***LaCroixColoR***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20LaCroixColoR)" width="150%" />
<p class="caption">
Figure 32: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20LaCroixColoR"><strong><em>LaCroixColoR</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>Manu</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-47"></span>
<img src="images/Manu.png" alt="[***Manu***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20Manu)" width="150%" />
<p class="caption">
Figure 33: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20Manu"><strong><em>Manu</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>MapPalettes</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-48"></span>
<img src="images/MapPalettes.png" alt="[***MapPalettes***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20MapPalettes)" width="150%" />
<p class="caption">
Figure 34: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20MapPalettes"><strong><em>MapPalettes</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>MetBrewer</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/MetBrewer.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-50"></span>
<img src="images/MetBrewer2.png" alt="[***MetBrewer***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20MetBrewer)" width="150%" />
<p class="caption">
Figure 35: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20MetBrewer"><strong><em>MetBrewer</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>MexBrewer</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-51"></span>
<img src="images/MexBrewer.png" alt="[***MexBrewer***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20MexBrewer)" width="150%" />
<p class="caption">
Figure 36: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20MexBrewer"><strong><em>MexBrewer</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>MoMAColors</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/MoMAColors1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-53"></span>
<img src="images/MoMAColors2.png" alt="[***MoMAColors***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20MoMAColors)" width="150%" />
<p class="caption">
Figure 37: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20MoMAColors"><strong><em>MoMAColors</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>NatParksPalettes</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-54"></span>
<img src="images/NatParksPalettes.png" alt="[***NatParksPalettes***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20NatParksPalettes)" width="150%" />
<p class="caption">
Figure 38: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20NatParksPalettes"><strong><em>NatParksPalettes</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>NineteenEightyR</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-55"></span>
<img src="images/NineteenEightyR.png" alt="[***NineteenEightyR***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20NineteenEightyR)" width="150%" />
<p class="caption">
Figure 39: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20NineteenEightyR"><strong><em>NineteenEightyR</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>PNWColors</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-56"></span>
<img src="images/PNWColors.png" alt="[***PNWColors***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20PNWColors)" width="150%" />
<p class="caption">
Figure 40: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20PNWColors"><strong><em>PNWColors</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>PrettyCols</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/PrettyCols1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-58"></span>
<img src="images/PrettyCols2.png" alt="[***PrettyCols***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20PrettyCols)" width="150%" />
<p class="caption">
Figure 41: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20PrettyCols"><strong><em>PrettyCols</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>RColorBrewer</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/RColorBrewer1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-60"></span>
<img src="images/RColorBrewer2.png" alt="[***RColorBrewer***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20RColorBrewer)" width="150%" />
<p class="caption">
Figure 42: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20RColorBrewer"><strong><em>RColorBrewer</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>RSkittleBrewer</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-61"></span>
<img src="images/RSkittleBrewer.png" alt="[***RSkittleBrewer***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20RSkittleBrewer)" width="150%" />
<p class="caption">
Figure 43: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20RSkittleBrewer"><strong><em>RSkittleBrewer</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>Redmonder</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-62"></span>
<img src="images/Redmonder2.png" alt="[***Redmonder***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20Redmonder)" width="150%" />
<p class="caption">
Figure 44: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20Redmonder"><strong><em>Redmonder</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>awtools</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-63"></span>
<img src="images/awtools.png" alt="[***awtools***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20awtools)" width="150%" />
<p class="caption">
Figure 45: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20awtools"><strong><em>awtools</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>basetheme</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-64"></span>
<img src="images/basetheme.png" alt="[***basetheme***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20basetheme)" width="150%" />
<p class="caption">
Figure 46: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20basetheme"><strong><em>basetheme</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>beyonce</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/beyonce1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/beyonce2.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/beyonce3.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/beyonce4.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-69"></span>
<img src="images/beyonce5.png" alt="[***beyonce***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20beyonce )" width="150%" />
<p class="caption">
Figure 47: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20beyonce"><strong><em>beyonce</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>calecopal</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/calecopal1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-71"></span>
<img src="images/calecopal2.png" alt="[***calecopal***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20calecopal)" width="150%" />
<p class="caption">
Figure 48: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20calecopal"><strong><em>calecopal</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>colRoz</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/colRoz1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-73"></span>
<img src="images/colRoz2.png" alt="[***colRoz***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20colRoz)" width="150%" />
<p class="caption">
Figure 49: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20colRoz"><strong><em>colRoz</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>colorBlindness</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-74"></span>
<img src="images/colorBlindness.png" alt="[***colorBlindness***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20colorBlindness)" width="150%" />
<p class="caption">
Figure 50: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20colorBlindness"><strong><em>colorBlindness</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>colorblindr</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-75"></span>
<img src="images/colorblindr.png" alt="[***colorblindr***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20colorblindr)" width="150%" />
<p class="caption">
Figure 51: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20colorblindr"><strong><em>colorblindr</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>dichromat</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-76"></span>
<img src="images/dichromat.png" alt="[***dichromat***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20dichromat)" width="150%" />
<p class="caption">
Figure 52: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20dichromat"><strong><em>dichromat</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>dutchmasters</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-77"></span>
<img src="images/dutchmasters.png" alt="[***dutchmasters***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20dutchmasters)" width="150%" />
<p class="caption">
Figure 53: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20dutchmasters"><strong><em>dutchmasters</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>feathers</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-78"></span>
<img src="images/feathers.png" alt="[***feathers***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20feathers)" width="150%" />
<p class="caption">
Figure 54: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20feathers"><strong><em>feathers</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>fishualize</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/fishualize1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/fishualize2.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/fishualize3.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/fishualize4.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/fishualize5.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-84"></span>
<img src="images/fishualize6.png" alt="[***fishualize***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20fishualize)" width="150%" />
<p class="caption">
Figure 55: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20fishualize"><strong><em>fishualize</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>futurevisions</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-85"></span>
<img src="images/futurevisions.png" alt="[***futurevisions***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20futurevisions)" width="150%" />
<p class="caption">
Figure 56: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20futurevisions"><strong><em>futurevisions</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>ggpomological</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-86"></span>
<img src="images/ggpomological.png" alt="[***ggpomological***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggpomological)" width="150%" />
<p class="caption">
Figure 57: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggpomological"><strong><em>ggpomological</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>ggprism</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/ggprism1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/ggprism2.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-89"></span>
<img src="images/ggprism3.png" alt="[***ggprism***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggprism)" width="150%" />
<p class="caption">
Figure 58: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggprism"><strong><em>ggprism</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>ggsci</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/ggsci1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-91"></span>
<img src="images/ggsci2.png" alt="[***ggsci***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggsci)" width="150%" />
<p class="caption">
Figure 59: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggsci"><strong><em>ggsci</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>ggthemes</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/ggthemes1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/ggthemes2.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/ggthemes3.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-95"></span>
<img src="images/ggthemes4.png" alt="[***ggthemes***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggthemes)" width="150%" />
<p class="caption">
Figure 60: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggthemes"><strong><em>ggthemes</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>ggthemr</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-96"></span>
<img src="images/ggthemr.png" alt="[***ggthemr***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggthemr)" width="150%" />
<p class="caption">
Figure 61: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ggthemr"><strong><em>ggthemr</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>ghibli</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-97"></span>
<img src="images/ghibli.png" alt="[***ghibli***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ghibli)" width="150%" />
<p class="caption">
Figure 62: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ghibli"><strong><em>ghibli</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>grDevices</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-98"></span>
<img src="images/grDevices.png" alt="[***grDevices***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20grDevices)" width="150%" />
<p class="caption">
Figure 63: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20grDevices"><strong><em>grDevices</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>impressionist.colors</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-99"></span>
<img src="images/impressionist.colors.png" alt="[***impressionist.colors***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20impressionist.colors)" width="150%" />
<p class="caption">
Figure 64: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20impressionist.colors"><strong><em>impressionist.colors</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>jcolors</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-100"></span>
<img src="images/jcolors.png" alt="[***jcolors***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20jcolors)" width="150%" />
<p class="caption">
Figure 65: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20jcolors"><strong><em>jcolors</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>khroma</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-101"></span>
<img src="images/khroma.png" alt="[***khroma***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20khroma)" width="150%" />
<p class="caption">
Figure 66: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20khroma"><strong><em>khroma</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>lisa</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/lisa1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/lisa2.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/lisa3.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/lisa4.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-106"></span>
<img src="images/lisa5.png" alt="[***lisa***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20lisa&amp;listing-listing-page=1)" width="150%" />
<p class="caption">
Figure 67: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20lisa&amp;listing-listing-page=1"><strong><em>lisa</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>ltc</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-107"></span>
<img src="images/ltc.png" alt="[***ltc***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ltc)" width="150%" />
<p class="caption">
Figure 68: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ltc"><strong><em>ltc</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>miscpalettes</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-108"></span>
<img src="images/miscpalettes.png" alt="[***miscpalettes***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20miscpalettes)" width="150%" />
<p class="caption">
Figure 69: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20miscpalettes"><strong><em>miscpalettes</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>musculusColors</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-109"></span>
<img src="images/musculusColors.png" alt="[***musculusColors***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20musculusColors)" width="150%" />
<p class="caption">
Figure 70: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20musculusColors"><strong><em>musculusColors</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>nationalparkcolors</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-110"></span>
<img src="images/nationalparkcolors.png" alt="[***nationalparkcolors***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20nationalparkcolors)" width="150%" />
<p class="caption">
Figure 71: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20nationalparkcolors"><strong><em>nationalparkcolors</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>nbapalettes</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/nbapalettes1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/nbapalettes2.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/nbapalettes3.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/nbapalettes4.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-115"></span>
<img src="images/nbapalettes5.png" alt="[***nbapalettes***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20nbapalettes)" width="150%" />
<p class="caption">
Figure 72: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20nbapalettes"><strong><em>nbapalettes</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>nord</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-116"></span>
<img src="images/nord.png" alt="[***nord***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20nord)" width="150%" />
<p class="caption">
Figure 73: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20nord"><strong><em>nord</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>ochRe</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-117"></span>
<img src="images/ochRe.png" alt="[***ochRe***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ochRe)" width="150%" />
<p class="caption">
Figure 74: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20ochRe"><strong><em>ochRe</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>palettesForR</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-118"></span>
<img src="images/palettesForR.png" alt="[***palettesForR***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20palettesForR)" width="150%" />
<p class="caption">
Figure 75: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20palettesForR"><strong><em>palettesForR</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>palettetown</em></strong>
</summary>
<blockquote>
<p>An R package providing pokemon colour palettes.</p>
This package is very much inspired by <a href="https://pokepalettes.com"><strong>pokepalettes</strong></a>.
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/timcdlucas/palettetown"><strong>palettetown</strong></a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown2.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown3.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown4.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown5.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown6.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown7.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown8.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown9.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown10.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown11.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/palettetown12.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-131"></span>
<img src="images/palettetown13.png" alt="[***palettetown***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20palettetown)" width="150%" />
<p class="caption">
Figure 76: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20palettetown"><strong><em>palettetown</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>pals</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-132"></span>
<img src="images/pals.png" alt="[***pals***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20pals)" width="150%" />
<p class="caption">
Figure 77: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20pals"><strong><em>pals</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>peRReo</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-133"></span>
<img src="images/peRReo.png" alt="[***peRReo***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20peRReo)" width="150%" />
<p class="caption">
Figure 78: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20peRReo"><strong><em>peRReo</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>rcartocolor</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/rcartocolor1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-135"></span>
<img src="images/rcartocolor2.png" alt="[***rcartocolor***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20rcartocolor)" width="150%" />
<p class="caption">
Figure 79: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20rcartocolor"><strong><em>rcartocolor</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>rockthemes</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-136"></span>
<img src="images/rockthemes.png" alt="[***rockthemes***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20rockthemes)" width="150%" />
<p class="caption">
Figure 80: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20rockthemes"><strong><em>rockthemes</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>rtist</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-137"></span>
<img src="images/rtist.png" alt="[***rtist***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20rtist)" width="150%" />
<p class="caption">
Figure 81: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20rtist"><strong><em>rtist</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>severance</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-138"></span>
<img src="images/severance.png" alt="[***severance***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20severance)" width="150%" />
<p class="caption">
Figure 82: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20severance"><strong><em>severance</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>soilpalettes</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-139"></span>
<img src="images/soilpalettes.png" alt="[***soilpalettes***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20soilpalettes)" width="150%" />
<p class="caption">
Figure 83: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20soilpalettes"><strong><em>soilpalettes</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>suffrager</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-140"></span>
<img src="images/suffrager.png" alt="[***suffrager***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20suffrager)" width="150%" />
<p class="caption">
Figure 84: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20suffrager"><strong><em>suffrager</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>tayloRswift</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-141"></span>
<img src="images/tayloRswift.png" alt="[***tayloRswift***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20tayloRswift)" width="150%" />
<p class="caption">
Figure 85: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20tayloRswift"><strong><em>tayloRswift</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>tidyquant</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-142"></span>
<img src="images/tidyquant.png" alt="[***tidyquant***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20tidyquant)" width="150%" />
<p class="caption">
Figure 86: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20tidyquant"><strong><em>tidyquant</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>trekcolors</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/trekcolors1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-144"></span>
<img src="images/trekcolors2.png" alt="[***trekcolors***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20trekcolors)" width="150%" />
<p class="caption">
Figure 87: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20trekcolors"><strong><em>trekcolors</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>tvthemes</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/tvthemes1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-146"></span>
<img src="images/tvthemes2.png" alt="[***tvthemes***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20tvthemes)" width="150%" />
<p class="caption">
Figure 88: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20tvthemes"><strong><em>tvthemes</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>unikn</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-147"></span>
<img src="images/unikn.png" alt="[***unikn***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20unikn)" width="150%" />
<p class="caption">
Figure 89: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20unikn"><strong><em>unikn</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>vangogh</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-148"></span>
<img src="images/vangogh.png" alt="[***vangogh***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20vangogh)" width="150%" />
<p class="caption">
Figure 90: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20vangogh"><strong><em>vangogh</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>vapeplot</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-149"></span>
<img src="images/vapeplot.png" alt="[***vapeplot***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20vapeplot)" width="150%" />
<p class="caption">
Figure 91: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20vapeplot"><strong><em>vapeplot</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>vapoRwave</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-150"></span>
<img src="images/vapoRwave.png" alt="[***vapoRwave***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20vapoRwave)" width="150%" />
<p class="caption">
Figure 92: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20vapoRwave"><strong><em>vapoRwave</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>waRhol</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-151"></span>
<img src="images/waRhol.png" alt="[***waRhol***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20waRhol)" width="150%" />
<p class="caption">
Figure 93: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20waRhol"><strong><em>waRhol</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>werpals</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-152"></span>
<img src="images/werpals.png" alt="[***werpals***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20werpals)" width="150%" />
<p class="caption">
Figure 94: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20werpals"><strong><em>werpals</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>wesanderson</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-153"></span>
<img src="images/wesanderson.png" alt="[***wesanderson***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20wesanderson)" width="150%" />
<p class="caption">
Figure 95: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20wesanderson"><strong><em>wesanderson</em></strong></a>
</p>
</div>
</div>
</details>
<details>
<summary>
<strong><em>yarrr</em></strong>
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-154"></span>
<img src="images/yarrr.png" alt="[***yarrr***](https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20yarrr)" width="150%" />
<p class="caption">
Figure 96: <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html#category=Package%20-%20yarrr"><strong><em>yarrr</em></strong></a>
</p>
</div>
</div>
</details>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="pbapplypack"><em>pbapply</em></h2>
<blockquote>
A lightweight package that adds progress bar to vectorized R functions (<strong><em>apply</em></strong>). The implementation can easily be added to functions where showing the progress is useful (e.g. bootstrap). The type and style of the progress bar (with percentages or remaining time) can be set through options. The package supports several parallel processing backends, such as snow-type clusters, multicore-type forking, and future.
<div class="citation">
<strong>Source:</strong> <a href="https://peter.solymos.org/pbapply/">pbapply</a>
</div>
</blockquote>
<p>Prior to using the <strong><em>pbapply</em></strong> <span class="citation" data-cites="pbapply">(<a href="#ref-pbapply" role="doc-biblioref">Solymos and Zawadzki 2023</a>)</span> package, it is important you learn how to vectorize/parallelize your data. The author wrote a complementary step-by-step guide on how to do so titled <a href="https://github.com/psolymos/the-road-to-progress?tab=readme-ov-file#the-road-to-progress"><strong>the road to progress</strong></a>.</p>
<p><a href="https://peter.solymos.org/pbapply/"><strong><em>pbapply</em> website</strong></a></p>
<p><a href="https://peter.solymos.org/pbapply/#how-to-get-started"><strong><em>pbapply</em> article: how to get started </strong></a></p>
<p><a href="https://peter.solymos.org/pbapply/#how-to-add-pbapply-to-a-package"><strong><em>pbapply</em> article: how to add <em>pbapply</em> to a package</strong></a></p>
<p><a href="https://peter.solymos.org/pbapply/#examples"><strong><em>pbapply</em> article: examples</strong></a></p>
<p><a href="https://peter.solymos.org/pbapply/reference/index.html"><strong><em>pbapply</em> functions</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/pbapply/pbapply.pdf"><strong><em>pbapply</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="plot3Dpack"><em>plot3D</em></h2>
<h3 id="plotting-multi-dimensional-data">Plotting Multi-Dimensional Data</h3>
<blockquote>
<span class="citation" data-cites="plot3D">(<a href="#ref-plot3D" role="doc-biblioref">Soetaert 2024</a>)</span> Functions for viewing 2-D and 3-D data, including perspective plots, slice plots, surface plots, scatter plots, etc. Includes data sets from oceanography.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/plot3D/index.html">plot3D</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-155"></span>
<img src="images/plot3D.png" alt="[***plot3D* example**](https://www.rforscience.com/plot3d-items/articles/plot3D-mtcars.html)" width="150%" />
<p class="caption">
Figure 97: <a href="https://www.rforscience.com/plot3d-items/articles/plot3D-mtcars.html"><strong><em>plot3D</em> example</strong></a>
</p>
</div>
</div>
<p><a href="https://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization"><strong><em>plot3D</em> tutorial</strong></a></p>
<p><a href="https://www.rforscience.com/plot3d.html"><strong><em>plot3D</em> R for Science article</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/plot3D/plot3D.pdf"><strong><em>plot3D</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="plotrixpack"><em>plotrix</em></h2>
<blockquote>
<strong><em>plotrix</em></strong> <span class="citation" data-cites="plotrix">(<a href="#ref-plotrix" role="doc-biblioref">J 2006</a>)</span> is an R package that provides many plotting, labeling, and axis &amp; color scaling functions.
<div class="citation">
<strong>Source: plotrix</strong> <a href="https://plotrix.github.io/plotrix/"></a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/plotrix1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-157"></span>
<img src="images/plotrix2.png" alt="[***plotrix* examples**](https://jtr13.github.io/cc19/plotrix-for-complex-visualizations.html)" width="150%" />
<p class="caption">
Figure 98: <a href="https://jtr13.github.io/cc19/plotrix-for-complex-visualizations.html"><strong><em>plotrix</em> examples</strong></a>
</p>
</div>
</div>
<p><a href="https://plotrix.github.io/plotrix/"><strong><em>plotrix</em> website</strong></a></p>
<p><a href="https://plotrix.github.io/plotrix/reference/index.html"><strong><em>plotrix</em> functions</strong></a></p>
<p><a href="https://github.com/plotrix/plotrix"><strong><em>plotrix</em> GitHub</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/plotrix/plotrix.pdf"><strong><em>plotrix</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="shinypack"><em>shiny</em></h2>
<h3 id="web-application-framework-for-r">Web Application Framework for R</h3>
<blockquote>
<span class="citation" data-cites="shiny">(<a href="#ref-shiny" role="doc-biblioref">Chang et al. 2024</a>)</span> Makes it incredibly easy to build interactive web applications with R. Automatic “reactive” binding between inputs and outputs and extensive prebuilt widgets make it possible to build beautiful, responsive, and powerful applications with minimal effort.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/shiny/index.html">shiny</a>
</div>
</blockquote>
<p><a href="https://shiny.posit.co/"><strong><em>shiny</em> website</strong></a></p>
<p><a href="https://shiny.posit.co/r/articles/"><strong><em>shiny</em> article hub</strong></a></p>
<p><a href="https://shiny.posit.co/r/getstarted/shiny-basics/lesson1/"><strong><em>shiny</em> article: shiny basics</strong></a></p>
<p><a href="https://shiny.posit.co/r/getstarted/build-an-app/hello-shiny/getting-started.html"><strong><em>shiny</em> article: build an app</strong></a></p>
<p><a href="https://shiny.posit.co/r/articles/share/shinyapps/"><strong><em>shiny</em> article: shinyapps.io</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/shiny/shiny.pdf"><strong><em>shiny</em> reference manual</strong></a></p>
<p><a href="https://rdatatoolbox.github.io/course-renv/#/"><strong>Mastering <em>shiny</em>: build interactive apps, reports, &amp; dashboards powered by R</strong></a> <strong>by Hadley Wickham</strong></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/shiny1.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-159"></span>
<img src="images/shiny2.png" alt="[***shiny*** **cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/shiny.pdf)" width="1000" />
<p class="caption">
Figure 99: <a href="https://github.com/rstudio/cheatsheets/blob/main/shiny.pdf"><strong><em>shiny</em></strong> <strong>cheatsheet</strong></a>
</p>
</div>
</div>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h2 id="showtextpack"><em>showtext</em></h2>
<h3 id="using-fonts-more-easily-in-r-graphs">Using Fonts More Easily in R Graphs</h3>
<blockquote>
<span class="citation" data-cites="showtext">(<a href="#ref-showtext" role="doc-biblioref">Qiu 2024</a>)</span> Making it easy to use various types of fonts (‘TrueType’, ‘OpenType’, Type 1, web fonts, etc.) in R graphs, and supporting most output formats of R graphics including PNG, PDF and SVG. Text glyphs will be converted into polygons or raster images, hence after the plot has been created, it no longer relies on the font files. No external software such as ‘Ghostscript’ is needed to use this package.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/showtext/showtext.pdf">showtext</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/showtext.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-161"></span>
<img src="images/showtext2.png" alt="[***showtext* example **](https://cran.r-project.org/web/packages/showtext/vignettes/introduction.html#introduction)" width="150%" />
<p class="caption">
Figure 100: <a href="https://cran.r-project.org/web/packages/showtext/vignettes/introduction.html#introduction"><strong><em>showtext</em> example </strong></a>
</p>
</div>
</div>
</details>
<p><a href="https://cran.r-project.org/web/packages/showtext/vignettes/introduction.html#introduction"><strong><em>showtext</em> website</strong></a></p>
<p><a href="https://github.com/yixuan/showtext"><strong><em>showtext</em> GitHub</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/showtext/showtext.pdf"><strong><em>showtext</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#figures">Return to Figures</a></strong>
</div>
<h1 id="fisheries">Fisheries Packages</h1>
<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Package</th>
<th style="text-align: center;">Section Link</th>
<th style="text-align: center;">Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong><em>BayesGrowth</em></strong></td>
<td style="text-align: center;"><a href="#BayesGrowthpack">link to <em>BayesGrowth</em></a></td>
<td style="text-align: center;">length-at-age growth curves</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>CatDyn</em></strong></td>
<td style="text-align: center;"><a href="#CatDynpack">link to <em>CatDyn</em></a></td>
<td style="text-align: center;">stock assessment by catch models</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>FESta</em></strong></td>
<td style="text-align: center;"><a href="#FEStapack">link to <em>FESta</em></a></td>
<td style="text-align: center;">fishing effort standardization</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>fishMod</em></strong></td>
<td style="text-align: center;"><a href="#fishModpack">link to <em>fishMod</em></a></td>
<td style="text-align: center;">fits models to CPUE data</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>FSA</em></strong></td>
<td style="text-align: center;"><a href="#FSApack">link to <em>FSA</em></a></td>
<td style="text-align: center;">Fisheries Stock Assessment</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>ggFishPlots</em></strong></td>
<td style="text-align: center;"><a href="#ggFishPlotspack">link to <em>ggFishPlots</em></a></td>
<td style="text-align: center;">visualize life history parameters</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>RFishBC</em></strong></td>
<td style="text-align: center;"><a href="#RFishBCpack">link to <em>RFishBC</em></a></td>
<td style="text-align: center;">back-calculation of fish length</td>
</tr>
</tbody>
</table>
<h2 id="BayesGrowthpack"><em>BayesGrowth</em></h2>
<h3 id="estimate-fish-growth-using-mcmc-analysis">Estimate fish growth using MCMC analysis</h3>
<blockquote>
<p><strong><em>BayesGrowth</em></strong> <span class="citation" data-cites="BayesGrowth1">(<a href="#ref-BayesGrowth1" role="doc-biblioref">Smart 2023</a>)</span> <span class="citation" data-cites="BayesGrowth2">(<a href="#ref-BayesGrowth2" role="doc-biblioref">Smart and Grammer 2021</a>)</span> combines length-at-age modelling for fisheries with MCMC implemented using Stan and the <strong><em>rstan</em></strong> package. Growth modelling using models such as the von Bertalanffy growth model involves three parameters: L∞, k and either L0 or t0. Two of these parameters: L0 and L∞ have direct biological meaning as the size-at-birth and maximum length, respectively. This package provides the tools to run an MCMC model with these two parameters treated as size-at-birth and maximum length using a rstan model with a No U-Turn Sampling (NUTS) algorithm. This MCMC model is pre-specified and built into wrapper functions.</p>
The user can therefore run an MCMC growth model using knowledge of species length-at-birth and maximum size as priors.
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/jonathansmart/BayesGrowth">BayesGrowth</a>
</div>
</blockquote>
<p><a href="https://github.com/jonathansmart/BayesGrowth"><strong><em>BayesGrowth</em> website</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/BayesGrowth/BayesGrowth.pdf"><strong><em>BayesGrowth</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#fisheries">Return to Fisheries Packages</a></strong>
</div>
<h2 id="CatDynpack"><em>CatDyn</em></h2>
<h3 id="fishery-stock-assessment-by-catch-dynamics-models">Fishery Stock Assessment by Catch Dynamics Models</h3>
<blockquote>
Based on fishery Catch Dynamics instead of fish Population Dynamics (hence <strong><em>CatDyn</em></strong>) <span class="citation" data-cites="CatDyn1">(<a href="#ref-CatDyn1" role="doc-biblioref">Roa-Ureta 2011</a>)</span> <span class="citation" data-cites="CatDyn2">(<a href="#ref-CatDyn2" role="doc-biblioref">Roa-Ureta 2015</a>)</span> <span class="citation" data-cites="CatDyn3">(<a href="#ref-CatDyn3" role="doc-biblioref">Roa-Ureta et al. 2015</a>)</span> <span class="citation" data-cites="CatDyn4">(<a href="#ref-CatDyn4" role="doc-biblioref">Lin et al. 2017</a>)</span> and using high-frequency or medium-frequency catch in biomass or numbers, fishing nominal effort, and mean fish body weight by time step, from one or two fishing fleets, estimate stock abundance, natural mortality rate, and fishing operational parameters. It includes methods for data organization, plotting standard exploratory and analytical plots, predictions, for 100 types of models of increasing complexity, and 72 likelihood models for the data.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/CatDyn/index.html">CatDyn</a>
</div>
</blockquote>
<p><a href="https://rdrr.io/cran/CatDyn/man/CatDyn-package.html"><strong><em>CatDyn</em> article: overview</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/CatDyn/CatDyn.pdf"><strong><em>CatDyn</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#fisheries">Return to Fisheries Packages</a></strong>
</div>
<h2 id="FEStapack"><em>FESta</em></h2>
<h3 id="fishing-effort-standardisation-festa">Fishing Effort Standardisation <span class="citation" data-cites="FESta">(<a href="#ref-FESta" role="doc-biblioref">Varghese, T V, and J 2020</a>)</span></h3>
<blockquote>
Original idea was presented in the reference paper. Varghese et al. (2020, 74(1):35-42) “Bayesian State-space Implementation of Schaefer Production Model for Assessment of Stock Status for Multi-gear Fishery”. Marine fisheries governance and management practices are very essential to ensure the sustainability of the marine resources. A widely accepted resource management strategy towards this is to derive sustainable fish harvest levels based on the status of marine fish stock. Various fish stock assessment models that describe the biomass dynamics using time series data on fish catch and fishing effort are generally used for this purpose. In the scenario of complex multi-species marine fishery in which different species are caught by a number of fishing gears and each gear harvests a number of species make it difficult to obtain the fishing effort corresponding to each fish species. Since the capacity of the gears varies, the effort made to catch a resource cannot be considered as the sum of efforts expended by different fishing gears. This necessitates standardisation of fishing effort in unit base.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/FESta/index.html">FESta</a>
</div>
</blockquote>
<p><a href="https://cran.r-project.org/web/packages/FESta/FESta.pdf"><strong><em>FESta</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#fisheries">Return to Fisheries Packages</a></strong>
</div>
<h2 id="fishModpack"><em>fishMod</em></h2>
<h3 id="fits-poisson-sum-of-gammas-glms-tweedie-glms-and-delta-log-normal-models">Fits Poisson-Sum-of-Gammas GLMs, Tweedie GLMs, and Delta Log-Normal Models</h3>
<blockquote>
<p><span class="citation" data-cites="fishMod">(<a href="#ref-fishMod" role="doc-biblioref">Foster 2024</a>)</span> Fits models to catch and effort data. Single-species models are</p>
<ol type="1">
<li><p>delta log-normal</p></li>
<li><p>Tweedie, or</p></li>
<li><p>Poisson-gamma (G)LMs.</p>
<div class="citation">
<strong>Source:</strong> <a href="https://scott-foster.r-universe.dev/fishMod">fishMod</a>
</div></li>
</ol>
</blockquote>
<p><a href="https://scott-foster.r-universe.dev/fishMod"><strong><em>fishMod</em> website</strong></a></p>
<p><a href="https://scott-foster.r-universe.dev/fishMod#manual"><strong><em>fishMod</em> article: help manuals</strong></a></p>
<p><a href="https://scott-foster.r-universe.dev/fishMod/doc/manual.html#00fishMod-package"><strong><em>fishMod</em> article: how to use</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/fishMod/fishMod.pdf"><strong><em>fishMod</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#fisheries">Return to Fisheries Packages</a></strong>
</div>
<h2 id="FSApack"><em>FSA</em></h2>
<h3 id="fisheries-stock-assessment">Fisheries Stock Assessment</h3>
<blockquote>
The <strong><em>FSA</em></strong> <span class="citation" data-cites="FSA">(<a href="#ref-FSA" role="doc-biblioref">Ogle et al. 2025</a>)</span> package provides R functions to conduct typical introductory fisheries analyses. Example analyses that use FSA can be found in the <a href="https://derekogle.com/IFAR/">Introductory Fisheries Analyses with R book</a> (see note below) and on the <a href="https://fishr-core-team.github.io/fishR/">fishR website</a>.
<div class="citation">
<strong>Source:</strong> <a href="https://fishr-core-team.github.io/FSA/">FSA</a>
</div>
</blockquote>
<p><a href="https://fishr-core-team.github.io/FSA/"><strong><em>FSA</em> website</strong></a></p>
<p><a href="https://fishr-core-team.github.io/FSA/reference/index.html"><strong><em>FSA</em> functions</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/FSA/FSA.pdf"><strong><em>FSA</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#fisheries">Return to Fisheries Packages</a></strong>
</div>
<h2 id="ggFishPlotspack"><em>ggFishPlots</em></h2>
<h3 id="visualise-and-calculate-life-history-parameters-for-fisheries-science">Visualise and calculate life history parameters for fisheries science</h3>
<blockquote>
At the time of writing, the package <span class="citation" data-cites="ggFishPlots">(<a href="#ref-ggFishPlots" role="doc-biblioref">Vihtakari 2024</a>)</span> produces four kinds of plots: <strong>growth curves, maturity plots, length-weight relationships, and catch curves</strong>. Each function returns a <strong><em>ggplot2</em></strong> plot and the estimated parameters as a text string that can be used in Rmarkdown and Shiny applications as well as a data frame for further use of the parameters. The elements are returned as a list. The package contains example data to illustrate the functionality.
<div class="citation">
<strong>Source:</strong> <a href="https://deepwaterimr.github.io/ggFishPlots/">ggFishPlots</a>
</div>
</blockquote>
<p><a href="https://deepwaterimr.github.io/ggFishPlots/"><strong><em>ggFishPlots</em> website</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/ggFishPlots/ggFishPlots.pdf"><strong><em>ggFishPlots</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#fisheries">Return to Fisheries Packages</a></strong>
</div>
<h2 id="RFishBCpack"><em>RFishBC</em></h2>
<h3 id="back-calculation-of-fish-length">Back-Calculation of Fish Length</h3>
<blockquote>
Helps fisheries scientists <strong>collect measurements from calcified
structures and back-calculate estimated lengths at previous ages</strong> using
standard procedures and models. This is intended to replace much of the
functionality provided by the now out-dated ‘fishBC’ software
(<a href="https://fisheries.org/bookstore/all-titles/software/70317/" class="uri">https://fisheries.org/bookstore/all-titles/software/70317/</a>). <span class="citation" data-cites="RFishBC">(<a href="#ref-RFishBC" role="doc-biblioref">Ogle 2023</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://cloud.r-project.org/web/packages/RFishBC/RFishBC.pdf">RFishBC</a>
</div>
</blockquote>
<p><a href="https://fishr-core-team.github.io/RFishBC/"><strong><em>RFishBC</em> website</strong></a></p>
<p><a href="https://fishr-core-team.github.io/RFishBC/articles/BCIntro.html"><strong><em>RFishBC</em> article: short introduction to back-calculation</strong></a></p>
<p><a href="https://fishr-core-team.github.io/RFishBC/articles/collectRadiiData.html"><strong><em>RFishBC</em> article: collect radial measurements from a calcified structure by interactively selecting annuli</strong></a></p>
<p><a href="https://fishr-core-team.github.io/RFishBC/articles/seeRadiiData.html"><strong><em>RFishBC</em> article: visualize points selected on a calcified structure</strong></a></p>
<p><a href="https://fishr-core-team.github.io/RFishBC/articles/BCCalc.html"><strong><em>RFishBC</em> article: compute back-calculated lengths</strong></a></p>
<p><a href="https://fishr-core-team.github.io/RFishBC/articles/BCWorkflow.html"><strong><em>RFishBC</em> article: suggested workflow for back-calculation</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/RFishBC/RFishBC.pdf"><strong><em>RFishBC</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#fisheries">Return to Fisheries Packages</a></strong>
</div>
<h1 id="geo">Geospatial Packages</h1>
<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Package</th>
<th style="text-align: center;">Section Link</th>
<th style="text-align: center;">Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong><em>leaflet</em></strong></td>
<td style="text-align: center;"><a href="#leafletpack">link to <em>leaflet</em></a></td>
<td style="text-align: center;">create interactive maps</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>maps/mapdata</em></strong></td>
<td style="text-align: center;"><a href="#mapsmapdatapack">link to <em>maps/mapdata</em></a></td>
<td style="text-align: center;">create maps with overlaid rivers</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>rayshader</em></strong></td>
<td style="text-align: center;"><a href="#rayshaderpack">link to <em>rayshader</em></a></td>
<td style="text-align: center;">create 2D and 3D maps</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>sf</em></strong></td>
<td style="text-align: center;"><a href="#sfpack">link to <em>sf</em></a></td>
<td style="text-align: center;">work with spatial data</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>tmap</em></strong></td>
<td style="text-align: center;"><a href="#tmappack">link to <em>tmap</em></a></td>
<td style="text-align: center;">create thematic maps</td>
</tr>
</tbody>
</table>
<h3 id="leafletpack"><em>leaflet</em></h3>
<blockquote>
<p>Leaflet is one of the most popular open-source JavaScript libraries for interactive maps. It’s used by websites ranging from The New York Times and The Washington Post to GitHub and Flickr, as well as GIS specialists like OpenStreetMap, Mapbox, and CartoDB.</p>
This R package <span class="citation" data-cites="leaflet">(<a href="#ref-leaflet" role="doc-biblioref">Cheng et al. 2024</a>)</span> makes it easy to integrate and control Leaflet maps in R.
<div class="citation">
<strong>Source:</strong> <a href="https://rstudio.github.io/leaflet/articles/leaflet.html">leaflet</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="leaflet html-widget html-fill-item" id="htmlwidget-6a3f1f57d8d609126126" style="width:624px;height:384px;"></div>
<script type="application/json" data-for="htmlwidget-6a3f1f57d8d609126126">{"x":{"options":{"crs":{"crsClass":"L.CRS.EPSG3857","code":null,"proj4def":null,"projectedBounds":null,"options":{}}},"calls":[{"method":"addTiles","args":["https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",null,null,{"minZoom":0,"maxZoom":18,"tileSize":256,"subdomains":"abc","errorTileUrl":"","tms":false,"noWrap":false,"zoomOffset":0,"zoomReverse":false,"opacity":1,"zIndex":1,"detectRetina":false,"attribution":"&copy; <a href=\"https://openstreetmap.org/copyright/\">OpenStreetMap<\/a>,  <a href=\"https://opendatacommons.org/licenses/odbl/\">ODbL<\/a>"}]},{"method":"addMarkers","args":[-36.852,174.768,null,null,null,{"interactive":true,"draggable":false,"keyboard":true,"title":"","alt":"","zIndexOffset":0,"opacity":1,"riseOnHover":false,"riseOffset":250},"The birthplace of R",null,null,null,null,{"interactive":false,"permanent":false,"direction":"auto","opacity":1,"offset":[0,0],"textsize":"10px","textOnly":false,"className":"","sticky":true},null]}],"limits":{"lat":[-36.852,-36.852],"lng":[174.768,174.768]}},"evals":[],"jsHooks":[]}</script>
</div>
<p><a href="https://leafletjs.com/"><strong>Leaflet website</strong></a></p>
<p><a href="https://rstudio.github.io/leaflet/index.html"><strong><em>leaflet</em> package website</strong></a></p>
<p><a href="https://rstudio.github.io/leaflet/articles/widget.html"><strong><em>leaflet</em> article: the map widget </strong></a></p>
<p><a href="https://rstudio.github.io/leaflet/articles/colors.html"><strong><em>leaflet</em> article: colors</strong></a></p>
<p><a href="https://rstudio.github.io/leaflet/articles/popups.html"><strong><em>leaflet</em> article: popups and labels</strong></a></p>
<p><a href=""><strong><em>leaflet</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#geo">Return to Geospatial Packages</a></strong>
</div>
<h2 id="mapsmapdatapack"><em>maps</em>/<em>mapdata</em></h2>
<blockquote>
Display of maps. Projection code and larger maps are in separate packages (‘<strong><em>mapproj</em></strong>’ and ‘<strong><em>mapdata</em></strong>’).
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/maps/index.html">maps</a>
</div>
</blockquote>
<p><strong><em>maps</em></strong> <span class="citation" data-cites="maps">(<a href="#ref-maps" role="doc-biblioref">Richard A. Becker, Ray Brownrigg. Enhancements by Thomas P Minka, and Deckmyn. 2024</a>)</span> includes <strong>maps of the USA</strong>, Canada, and other countries with various add-ons available (such as cities, counties, etc.) and also a database of <strong>lakes</strong> around the world.</p>
<p><strong><em>mapdata</em></strong> <span class="citation" data-cites="mapdata">(<a href="#ref-mapdata" role="doc-biblioref">Richard A. Becker and Ray Brownrigg. 2022</a>)</span> is an add-on package that provides information for additional countries and also <strong>rivers</strong>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-162"></span>
<img src="images/maps.png" alt="[***maps***](https://eriqande.github.io/rep-res-web/lectures/making-maps-with-R.html)" width="150%" />
<p class="caption">
Figure 101: <a href="https://eriqande.github.io/rep-res-web/lectures/making-maps-with-R.html"><strong><em>maps</em></strong></a>
</p>
</div>
</div>
<p><a href="https://r-graph-gallery.com/278-the-maps-library.html"><strong>use the <em>maps</em>, <em>mapdata</em> and <em>oz</em> packages to get most common boundaries</strong></a></p>
<p><a href="https://eriqande.github.io/rep-res-web/lectures/making-maps-with-R.html"><strong>tutorial: making maps with R</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/maps/maps.pdf"><strong><em>maps</em> reference manual</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/mapdata/mapdata.pdf"><strong><em>mapdata</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#geo">Return to Geospatial Packages</a></strong>
</div>
<h2 id="rayshaderpack"><em>rayshader</em></h2>
<blockquote>
<p><strong><em>rayshader</em></strong> <span class="citation" data-cites="rayshader">(<a href="#ref-rayshader" role="doc-biblioref">Morgan-Wall 2025</a>)</span> is an open source package for producing 2D and 3D data visualizations in R. <strong><em>rayshader</em></strong> uses elevation data in a base R matrix and a combination of raytracing, hillshading algorithms, and overlays to generate stunning 2D and 3D maps. In addition to maps, <strong><em>rayshader</em></strong> also allows the user to translate <strong><em>ggplot2</em></strong> objects into beautiful 3D data visualizations.</p>
The models can be rotated and examined interactively or the camera movement can be scripted to create animations. Scenes can also be rendered using a high-quality pathtracer, rayrender. The user can also create a cinematic depth of field post-processing effect to direct the user’s focus to important regions in the figure. The 3D models can also be exported to a 3D-printable format with a built-in STL export function, and can be exported to an OBJ file.
<div class="citation">
<strong>Source:</strong> <a href="https://www.rayshader.com/">rayshader</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-163"></span>
<img src="images/rayshader.png" alt="[***rayshader* example**](https://www.rayshader.com/)" width="150%" />
<p class="caption">
Figure 102: <a href="https://www.rayshader.com/"><strong><em>rayshader</em> example</strong></a>
</p>
</div>
</div>
<p><a href="https://www.rayshader.com/"><strong><em>rayshader</em> website</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/rayshader/rayshader.pdf"><strong><em>rayshader</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#geo">Return to Geospatial Packages</a></strong>
</div>
<h2 id="sfpack"><em>sf</em></h2>
<blockquote>
<p>Package <strong><em>sf</em></strong> <span class="citation" data-cites="sf">(<a href="#ref-sf" role="doc-biblioref">Pebesma and Bivand 2023</a>)</span> :</p>
<ul>
<li><p>represents simple features as records in a data.frame or tibble with a geometry list-column</p></li>
<li><p>represents natively in R all 17 simple feature types for all dimensions (XY, XYZ, XYM, XYZM)</p></li>
<li><p>interfaces to GEOS for geometrical operations on projected coordinates, and (through R package <strong><em>s2</em></strong>) to <strong><em>s2geometry</em></strong> for geometrical operations on ellipsoidal coordinates</p></li>
<li><p>interfaces to GDAL, supporting all driver options, Date and POSIXct and list-columns</p></li>
<li><p>interfaces to PRØJ for coordinate reference system conversion and transformation &gt;</p></li>
<li><p>uses well-known-binary serialisations written in C++/Rcpp for fast I/O with GDAL and GEOS</p></li>
<li><p>reads from and writes to spatial databases such as PostGIS using DBI
is extended by</p></li>
<li><p>lwgeom for selected liblwgeom/PostGIS functions</p></li>
<li><p>stars for raster data, and raster or vector data cubes (spatial time series)</p></li>
<li><p>sfnetworks for geospatial network data</p>
<div class="citation">
<strong>Source:</strong> <a href="https://r-spatial.github.io/sf/index.html">sf</a>
</div></li>
</ul>
</blockquote>
<p><a href="https://r-spatial.github.io/sf/index.html"><strong><em>sf</em> website</strong></a></p>
<p><a href="https://r-spatial.github.io/sf/reference/index.html"><strong><em>sf</em> functions</strong></a></p>
<p><a href="https://r-spatial.github.io/sf/articles/sf1.html"><strong><em>sf</em> article: simple features for R</strong></a></p>
<p><a href="https://r-spatial.github.io/sf/articles/sf2.html"><strong><em>sf</em> article: reading, writing, and converting simple Features</strong></a></p>
<p><a href="https://r-spatial.github.io/sf/articles/sf3.html"><strong><em>sf</em> article: manipulating simple feature geometries</strong></a></p>
<p><a href="https://r-spatial.github.io/sf/articles/sf4.html"><strong><em>sf</em> article: manipulating simple features</strong></a></p>
<p><a href="https://r-spatial.github.io/sf/articles/sf5.html"><strong><em>sf</em> article: plotting simple figures</strong></a></p>
<p><a href="https://r-spatial.github.io/sf/articles/sf6.html"><strong><em>sf</em> article: miscellaneous</strong></a></p>
<p><a href="https://r-spatial.github.io/sf/articles/sf7.html"><strong><em>sf</em> article: spherical geometry in <em>sf</em> using s2geometry</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/sf/sf.pdf"><strong><em>sf</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/sf1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-165"></span>
<img src="images/sf2.png" alt="[***sf* cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/sf.pdf )" width="150%" />
<p class="caption">
Figure 103: <a href="https://github.com/rstudio/cheatsheets/blob/main/sf.pdf"><strong><em>sf</em> cheatsheet</strong></a>
</p>
</div>
</div>
</details>
<div class="citation">
<strong><a href="#geo">Return to Geospatial Packages</a></strong>
</div>
<h2 id="tmappack"><em>tmap</em></h2>
<h3 id="thematic-maps-in-r">thematic maps in R</h3>
<blockquote>
<strong><em>tmap</em></strong> <span class="citation" data-cites="tmap">(<a href="#ref-tmap" role="doc-biblioref">Tennekes 2018</a>)</span> is an R package for drawing thematic maps. The API is based on A Layered Grammar of Graphics and resembles the syntax of <strong><em>ggplot2</em></strong>, a popular R-library for drawing charts.
<div class="citation">
<strong>Source:</strong> <a href="https://r-tmap.github.io/tmap/">tmap</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-166"></span>
<img src="images/tmap.png" alt="[***tmap***](https://r-tmap.github.io/tmap/)" width="150%" />
<p class="caption">
Figure 104: <a href="https://r-tmap.github.io/tmap/"><strong><em>tmap</em></strong></a>
</p>
</div>
</div>
<p><a href="https://r-tmap.github.io/tmap/"><strong><em>tmap</em> website</strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/foundations_gg"><strong><em>tmap</em> article: foundations - grammar of graphics</strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/foundations_crs"><strong><em>tmap</em> article: foundations - map projections (crs)</strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/basics_vv"><strong><em>tmap</em> article: visual variables</strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/basics_scales"><strong><em>tmap</em> article: scales</strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/basics_legends"><strong><em>tmap</em> article: legends</strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/basics_basemaps"><strong><em>tmap</em> article: basemaps</strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/basics_basemaps"><strong><em>tmap</em> article: <em>tmap</em> versus <em>mapsf</em></strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/adv_positions"><strong><em>tmap</em> article: positioning components</strong></a></p>
<p><a href="https://r-tmap.github.io/tmap/articles/adv_margins"><strong><em>tmap</em> article: margins and aspect ratio</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/tmap/tmap.pdf"><strong><em>tmap</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#geo">Return to Geospatial Packages</a></strong>
</div>
<h1 id="misc">Miscellaneous</h1>
<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Package</th>
<th style="text-align: center;">Section Link</th>
<th style="text-align: center;">Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong><em>EchoviewR</em></strong></td>
<td style="text-align: center;"><a href="#EchoviewRpack">link to <em>EchoviewR</em></a></td>
<td style="text-align: center;">Echoview and R interface</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>rticles</em></strong></td>
<td style="text-align: center;"><a href="#rticlespack">link to <em>rticles</em></a></td>
<td style="text-align: center;">format articles for publication</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>tinytex</em></strong></td>
<td style="text-align: center;"><a href="#tinytexpack">link to <em>tinytex</em></a></td>
<td style="text-align: center;">enables LaTeX coding in R</td>
</tr>
</tbody>
</table>
<h2 id="EchoviewRpack"><em>EchoviewR</em></h2>
<blockquote>
<strong><em>EchoviewR</em></strong> <span class="citation" data-cites="EchoviewR">(<a href="#ref-EchoviewR" role="doc-biblioref">Harrison et al. 2015</a>)</span> is a free and extensive software package that acts as an interface between R and Echoview, developed by the Australian Antarctic Division.
<div class="citation">
<strong>Source:</strong> <a href="https://echoview.com/support/example-scripts/">Echoview</a>
</div>
</blockquote>
<p><a href="https://australianantarcticdivision.github.io/EchoviewR/index.html"><strong><em>EchoviewR</em> website</strong></a></p>
<p><a href="https://australianantarcticdivision.github.io/EchoviewR/articles/Getting_Started.html"><strong><em>EchoviewR</em> article: getting started</strong></a></p>
<p><a href="https://australianantarcticdivision.github.io/EchoviewR/articles/Introduction.html"><strong><em>EchoviewR</em> article: basic functionality</strong></a></p>
<p><a href="https://australianantarcticdivision.github.io/EchoviewR/articles/BroadbandST.html"><strong><em>EchoviewR</em> article: broadband single targets</strong></a></p>
<div class="citation">
<strong><a href="#misc">Return to Miscellaneous Packages</a></strong>
</div>
<h2 id="rticlespack"><em>rticles</em></h2>
<blockquote>
The <strong><em>rticles</em></strong> <span class="citation" data-cites="rticles">(<a href="#ref-rticles" role="doc-biblioref">Allaire et al. 2024</a>)</span> package provides a suite of custom R Markdown LaTeX formats and templates for various formats. Most of the templates are provided and maintained by the community, and anyone can contribute a new template.
<div class="citation">
<strong>Source:</strong> <a href="https://github.com/rstudio/rticles">rticles</a>
</div>
</blockquote>
<p>This package is <strong>helpful in formatting articles for publication</strong> in scientific journals without having to manually reformat all your citations. In particular it may be helpful if you want to publish in <strong>PLOS, PNAS, or PeerJ</strong> but a full list of journals available is listed below.</p>
<h3 id="usage"><strong>Usage</strong></h3>
<blockquote>
<p>To use <strong><em>rticles</em></strong> from RStudio, you can access the templates through <strong>File -&gt; New File -&gt; R Markdown</strong>. This will open the dialog box where you can select from one of the available templates…</p>
<p>You can also get the list of available journal names with <strong>rticles::journals()</strong>.</p>
<table style="width:94%;">
<colgroup>
<col style="width: 13%" />
<col style="width: 13%" />
<col style="width: 13%" />
<col style="width: 13%" />
<col style="width: 13%" />
<col style="width: 13%" />
<col style="width: 13%" />
</colgroup>
<tbody>
<tr class="odd">
<td style="text-align: center;">ACM</td>
<td style="text-align: center;">ACS</td>
<td style="text-align: center;">AEA</td>
<td style="text-align: center;">AGU</td>
<td style="text-align: center;">AJS</td>
<td style="text-align: center;">AMQ</td>
<td style="text-align: center;">AMS</td>
</tr>
<tr class="even">
<td style="text-align: center;">ARXIV</td>
<td style="text-align: center;">ASA</td>
<td style="text-align: center;">Bioinf.</td>
<td style="text-align: center;">Biomet.</td>
<td style="text-align: center;">Copernic.</td>
<td style="text-align: center;">CTEX</td>
<td style="text-align: center;">Elsevier</td>
</tr>
<tr class="odd">
<td style="text-align: center;">frontiers</td>
<td style="text-align: center;">glossa</td>
<td style="text-align: center;">IEEE</td>
<td style="text-align: center;">IMS</td>
<td style="text-align: center;">Informs</td>
<td style="text-align: center;">IOP</td>
<td style="text-align: center;">ISBA</td>
</tr>
<tr class="even">
<td style="text-align: center;">JASA</td>
<td style="text-align: center;">JEDM</td>
<td style="text-align: center;">JOSS</td>
<td style="text-align: center;">JSS</td>
<td style="text-align: center;">LIPICS</td>
<td style="text-align: center;">LNCS</td>
<td style="text-align: center;">MDPI</td>
</tr>
<tr class="odd">
<td style="text-align: center;">MNRAS</td>
<td style="text-align: center;">OUP_v0</td>
<td style="text-align: center;">OUP_v1</td>
<td style="text-align: center;">PeerJ</td>
<td style="text-align: center;">PIHPH</td>
<td style="text-align: center;">PLOS</td>
<td style="text-align: center;">PNAS</td>
</tr>
<tr class="even">
<td style="text-align: center;">R Journal</td>
<td style="text-align: center;">RSOS</td>
<td style="text-align: center;">RSS</td>
<td style="text-align: center;">SAGE</td>
<td style="text-align: center;">SIM</td>
<td style="text-align: center;">Springer</td>
<td style="text-align: center;">TF</td>
</tr>
<tr class="odd">
<td style="text-align: center;">TRB</td>
<td style="text-align: center;">Wellcom.</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
… LaTeX templates are used to ensure that documents conform precisely to submission standards.
<div class="citation">
<strong>Source: </strong> <a href="https://github.com/rstudio/rticles?tab=readme-ov-file#templates">rticles</a>
</div>
</blockquote>
<p><a href="https://pkgs.rstudio.com/rticles/"><strong><em>rticles</em> website</strong></a></p>
<p><a href="https://pkgs.rstudio.com/rticles/articles/rticles.html"><strong><em>rticles</em> article: get started</strong></a></p>
<p><a href="https://pkgs.rstudio.com/rticles/#templates"><strong><em>rticles</em> article: available templates</strong></a></p>
<p><a href="https://pkgs.rstudio.com/rticles/articles/examples.html"><strong><em>rticles</em> article: examples of journal templates</strong></a></p>
<p><a href="https://pkgs.rstudio.com/rticles/"><strong><em>rticles</em> website</strong></a></p>
<p><a href="https://yihui.org/tinytex/r/#debugging"><strong>debugging LaTeX to PDF</strong></a></p>
<ul>
<li>step-by-step guide to address errors if your desired template doesn’t knit properly.</li>
</ul>
<div class="citation">
<strong><a href="#misc">Return to Miscellaneous Packages</a></strong>
</div>
<h2 id="tinytexpack"><em>tinytex</em></h2>
<blockquote>
<strong><em>tinytex</em></strong> <span class="citation" data-cites="tinytex1">(<a href="#ref-tinytex1" role="doc-biblioref">Xie 2024</a>)</span> <span class="citation" data-cites="tinytex2">(<a href="#ref-tinytex2" role="doc-biblioref">Xie 2019</a>)</span> is a custom <strong>LaTeX</strong> distribution based on <strong>TeX Live</strong> that is small in size, but functions well in most cases, especially for R users. If you run into the problem of missing LaTeX packages, it should be super clear to you what you need to do (in fact, R users won’t need to do anything). You only install LaTeX packages you actually need.
<div class="citation">
<strong>Source:</strong> <a href="https://yihui.org/tinytex/">TinyTeX</a>
</div>
</blockquote>
<p>This package allows .Rmd files to be exported as PDF documents. It is also necessary if you want to use the <strong><em>rticles</em></strong> package to format articles for publication.</p>
<p><a href="https://yihui.org/tinytex/"><strong><em>tinytex</em> website</strong></a></p>
<p><a href="https://yihui.org/tinytex/#motivation"><strong><em>tinytex</em> article: motivation</strong></a></p>
<p><a href="https://yihui.org/tinytex/faq/"><strong><em>tinytex</em> article: FAQs</strong></a></p>
<p><a href="https://yihui.org/tinytex/r/#compile-latex-documents"><strong><em>tinytex</em> article: compile LaTeX documents</strong></a></p>
<p><a href="https://yihui.org/tinytex/r/#debugging"><strong><em>tinytex</em> article: debugging</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/tinytex/tinytex.pdf"><strong><em>tinytex</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#misc">Return to Miscellaneous Packages</a></strong>
</div>
<h1 id="stats">Statistics</h1>
<h2 id="AICcmodavgpack"><em>AICcmodavg</em></h2>
<h3 id="model-selection-and-multimodel-inference-based-on-qaicc">Model Selection and Multimodel Inference Based on (Q)AIC(c)</h3>
<blockquote>
<span class="citation" data-cites="AICcmodavg">(<a href="#ref-AICcmodavg" role="doc-biblioref">Mazerolle 2023</a>)</span> Functions to implement model selection and multimodel inference based on Akaike’s information criterion (AIC) and the second-order AIC (AICc), as well as their quasi-likelihood counterparts (QAIC, QAICc) from various model object classes. The package implements classic model averaging for a given parameter of interest or predicted values, as well as a shrinkage version of model averaging parameter estimates or effect sizes. The package includes diagnostics and goodness-of-fit statistics for certain model types including those of ‘unmarkedFit’ classes estimating demographic parameters after accounting for imperfect detection probabilities. Some functions also allow the creation of model selection tables for Bayesian models of the ‘bugs’, ‘rjags’, and ‘jagsUI’ classes. Functions also implement model selection using BIC. Objects following model selection and multimodel inference can be formatted to LaTeX using <strong><em>xtable</em></strong> methods included in the package.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/AICcmodavg/index.html">AICcmodavg</a>
</div>
</blockquote>
<p><a href="https://www.rdocumentation.org/packages/AICcmodavg/versions/2.3-3"><strong><em>AICcmodavg</em> functions</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/AICcmodavg/AICcmodavg.pdf"><strong><em>AICcmodavg</em> reference manual</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/AICcmodavg/vignettes/AICcmodavg.pdf"><strong>vignette: model selection and multimodal inference using the <em>AICmodavg</em> package</strong></a></p>
<p><a href="http://127.0.0.1:31516/library/AICcmodavg/doc/AICcmodavg-unmarked.pdf"><strong>vignette: example session of model selection tools and diagnostics for model fit with the <em>unmarked</em> package</strong></a></p>
</details>
<div class="citation">
<strong><a href="#stats">Return to Statistics Packages</a></strong>
</div>
<h2 id="bbmlepack"><em>bbmle</em></h2>
<h3 id="tools-for-general-maximum-likelihood-estimation">Tools for General Maximum Likelihood Estimation</h3>
<blockquote>
<span class="citation" data-cites="bbmle">(<a href="#ref-bbmle" role="doc-biblioref">Bolker and R Development Core Team 2025</a>)</span> Methods and functions for fitting maximum likelihood models in R. This package modifies and extends the <strong><em>mle</em></strong> classes in the <strong><em>stats4</em></strong> package.
<div class="citation">
<strong>Source:</strong> <a href="https://bbolker.r-universe.dev/bbmle">bbmle</a>
</div>
</blockquote>
<p><a href="https://bbolker.r-universe.dev/bbmle"><strong><em>bbmle</em> website</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/bbmle/bbmle.pdf"><strong><em>bbmle</em> reference manual</strong></a></p>
<p><a href="https://bbolker.r-universe.dev/bbmle/doc/quasi.pdf"><strong><em>bbmle</em> article: dealing with quasi models in R</strong></a></p>
<p><a href="https://bbolker.r-universe.dev/bbmle/doc/mle2.pdf"><strong><em>bbmle</em> article: maximum likelihood estimation and analysis with the <em>bbmle</em> package</strong></a></p>
<div class="citation">
<strong><a href="#stats">Return to Statistics Packages</a></strong>
</div>
<h2 id="DHARMapack"><em>DHARMa</em></h2>
<h3 id="residual-diagnostics-for-hierarchical-multi-level-mixed-regression-models">Residual Diagnostics for Hierarchical (Multi-Level / Mixed) Regression Models</h3>
<blockquote>
The <strong><em>DHARMa</em></strong> <span class="citation" data-cites="DHARMa">(<a href="#ref-DHARMa" role="doc-biblioref">Hartig 2024</a>)</span> package uses a simulation-based approach to create readily interpretable scaled (quantile) residuals for fitted (generalized) linear mixed models. Currently supported are linear and generalized linear (mixed) models from <strong><em>lme4</em></strong> (classes ‘lmerMod’, ‘glmerMod’), <strong><em>glmmTMB</em></strong>, <strong><em>GLMMadaptive</em></strong>, and <strong><em>spaMM</em></strong>; phylogenetic linear models from ‘phylolm’ (classes ‘phylolm’ and ‘phyloglm’); generalized additive models (‘gam’ from ‘mgcv’); ‘glm’ (including ‘negbin’ from ‘MASS’, but excluding quasi-distributions) and ‘lm’ model classes. Moreover, externally created simulations, e.g. posterior predictive simulations from Bayesian software such as ‘JAGS’, ‘STAN’, or ‘BUGS’ can be processed as well. The resulting residuals are standardized to values between 0 and 1 and can be interpreted as intuitively as residuals from a linear regression. The package also provides a number of plot and test functions for typical model misspecification problems, such as over/underdispersion, zero-inflation, and residual spatial, phylogenetic and temporal autocorrelation.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/DHARMa/index.html">DHARMa</a>
</div>
</blockquote>
<p><a href="https://florianhartig.github.io/DHARMa/"><strong><em>DHARMa</em> website</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/DHARMa/DHARMa.pdf"><strong><em>DHARMa</em> reference manual</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/DHARMa/vignettes/DHARMaForBayesians.html"><strong><em>DHARMa</em> article: for Bayesians</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/DHARMa/vignettes/DHARMa.html"><strong><em>DHARMa</em> article: residual diagnositics for hierarchical (multi-level/mixed) regression models</strong></a></p>
<div class="citation">
<strong><a href="#stats">Return to Statistics Packages</a></strong>
</div>
<h2 id="emmeanspack"><em>emmeans</em></h2>
<h3 id="estimated-marginal-means-aka-least-squares-means">Estimated Marginal Means, aka Least-Squares Means</h3>
<blockquote>
Obtain estimated marginal means (EMMs) <span class="citation" data-cites="emmeans">(<a href="#ref-emmeans" role="doc-biblioref">Lenth 2024</a>)</span> for many linear, generalized linear, and mixed models. Compute contrasts or linear functions of EMMs, trends, and comparisons of slopes. Plots and other displays. Least-squares means are discussed, and the term “estimated marginal means” is suggested, in Searle, Speed, and Milliken (1980) Population marginal means in the linear model: An alternative to least squares means, The American Statistician 34(4), 216-221 <a href="doi:10.1080/00031305.1980.10483031" class="uri">doi:10.1080/00031305.1980.10483031</a>.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/emmeans/index.html">emmeans</a>
</div>
</blockquote>
<p><a href="https://rvlenth.github.io/emmeans/"><strong><em>emmeans</em> website</strong></a></p>
<p><a href="https://rvlenth.github.io/emmeans/articles/AQuickStart.html"><strong><em>emmeans</em></strong> <strong>article: quick start guide</strong></a></p>
<p><a href="https://rvlenth.github.io/emmeans/articles/basics.html"><strong><em>emmeans</em></strong> <strong>article: basics of estimated marginal means</strong></a></p>
<p><a href="https://rvlenth.github.io/emmeans/articles/comparisons.html"><strong><em>emmeans</em> article: comparisons and contrasts</strong></a></p>
<p><a href="https://rvlenth.github.io/emmeans/articles/confidence-intervals.html"><strong><em>emmeans</em> article: confidence intervals and tests</strong></a></p>
<p><a href="https://rvlenth.github.io/emmeans/articles/FAQs.html"><strong><em>emmeans</em> FAQs</strong></a></p>
<p><a href="https://rvlenth.github.io/emmeans/articles/vignette-topics.html"><strong><em>emmeans</em> index of vignette topics</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/emmeans/emmeans.pdf"><strong><em>emmeans</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#stats">Return to Statistics Packages</a></strong>
</div>
<h2 id="gamlsspack"><em>gamlss</em></h2>
<h3 id="generalized-additive-models-for-location-scale-and-shape">Generalized Additive Models for Location Scale and Shape</h3>
<blockquote>
<p><strong>01. What is GAMLSS</strong> <span class="citation" data-cites="gamlss">(<a href="#ref-gamlss" role="doc-biblioref">R. A. Rigby and D. M. Stasinopoulos 2005</a>)</span></p>
<p>GAMLSS are <strong>distributional regression models</strong>. In classical regression models, the explanatory variables, <strong>X</strong>, affect the expected value of the response <strong>y, X -&gt; E(y)</strong>. In a distributional regression the <strong>X</strong>’s effects all parts of the distribution of <strong>y, X -&gt; D(y)</strong>. The GAMLSS models are appropriate when the focus of the study is not only shifts in the mean (or location) of the distribution for y but possibly other parts like the variance (volatility), the skewness, the kurtosis (heavy tails), the quantiles. All aspects of the the distribution of the response can be modelled as functions of the explanatory variables.</p>
<p><strong>03. What distributions can be used within GAMLSS</strong></p>
GAMLSS provide over 100 <strong>continuous, discrete and mixed distributions for modelling the response variable. Truncated, censored (interval response), log and logit transformed and finite mixture versions of these distributions can be also used.</strong>
&gt;
<div class="citation">
<strong>Source:</strong> <a href="https://www.gamlss.com/">gamlss</a>
</div>
</blockquote>
<p><a href="https://www.gamlss.com/"><strong><em>gamlss</em> website</strong></a></p>
<p><a href="https://www.rdocumentation.org/packages/gamlss/versions/5.4-12/topics/gamlss"><strong><em>gamlss</em>: generalized additive models for location scale and shape</strong></a></p>
<p><a href="https://www.gamlss.com/wp-content/uploads/2023/06/gamlssreferencecard.pdf"><strong><em>gamlss</em> R packages reference card</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/gamlss/gamlss.pdf"><strong><em>gamlss</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#stats">Return to Statistics Packages</a></strong>
</div>
<h2 id="glmmTMBpack"><em>glmmTMB</em></h2>
<h3 id="fitting-generalized-linear-mixed-models-glmms-using-template-model-builder-tmb">Fitting Generalized Linear Mixed Models (GLMMs) using Template Model Builder (TMB)</h3>
<blockquote>
<strong><em>glmmTMB</em></strong> <span class="citation" data-cites="glmmTMB">(<a href="#ref-glmmTMB" role="doc-biblioref">Brooks et al. 2017</a>)</span> is an R package built on the Template Model Builder automatic
differentiation engine, for fitting generalized linear mixed models and extensions.
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/glmmTMB/vignettes/glmmTMB.pdf">glmmTMB</a>
</div>
</blockquote>
<h3 id="response-distributions">Response Distributions</h3>
<p><strong>- Gaussian </strong><br />
<strong>- binomial </strong><br />
<strong>- beta-binomial </strong><br />
<strong>- Poisson </strong><br />
<strong>- negative binomial (NB1 and NB2 parameterizations) </strong><br />
<strong>- Conway-Maxwell-Poisson </strong><br />
<strong>- generalized Poisson </strong><br />
<strong>- Gamma </strong><br />
<strong>- Beta </strong><br />
<strong>- Tweedie </strong><br />
<strong>- zero-truncated Poisson </strong><br />
</p>
<h3 id="usage-1">Usage</h3>
<blockquote>
<p>In order to use <strong><em>glmmTMB</em></strong> effectively you should already be reasonably familiar with generalized linear mixed models (GLMMs), which in turn requires familiarity with (i) generalized linear models (e.g. the special cases of logistic, binomial, and Poisson regression) and (ii) ‘modern’ mixed models (those
working via maximization of the marginal likelihood rather than by manipulating sums of squares). Bolker et al. (2009) and Bolker (2015) are reasonable starting points in this area (especially geared to biologists and less-technical readers), as are Zuur et al. (2009), Millar (2011), and Zuur et al. (2013).</p>
<p>In order to fit a model in <strong><em>glmmTMB</em></strong> you need to:</p>
<ul>
<li><p>specify a model for the conditional effects, in the standard R (Wilkinson-Rogers) formula notation (see ?formula or Section 11.1 of the Introduction to R. Formulae can also include offsets.)</p></li>
<li><p>specify a model for the random effects, in the notation that is common to the nlme and lme4 packages. Random effects are specified as x|g, where x is an effect and g is a grouping factor (which must be a factor variable, or a nesting of/interaction among factor variables)….</p></li>
<li><p>choose the error distribution by specifying the family (family argument). In general, you can specify the function (binomial, gaussian, poisson, Gamma from base R, or one of the options listed at family glmmTMB [nbinom2, beta family(), betabinomial, …])).</p></li>
<li><p>choose the error distribution by specifying the family (family argument).</p>
<div class="citation">
<strong>Source:</strong> <a href="https://cran.r-project.org/web/packages/glmmTMB/vignettes/glmmTMB.pdf">glmmTMB</a>
</div></li>
</ul>
</blockquote>
<p><a href="https://github.com/glmmTMB/glmmTMB"><strong><em>glmmTMB</em></strong> <strong>GitHub Page</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/glmmTMB/vignettes/glmmTMB.pdf"><strong><em>glmmTMB</em> reference manual</strong></a></p>
<p><a href="https://www.rdocumentation.org/packages/glmmTMB/versions/1.1.10/topics/glmmTMB"><strong><em>glmmTMB</em>: fit models with TMB</strong></a></p>
<p><a href="https://bbolker.github.io/mixedmodels-misc/ecostats_chap.html"><strong><em>glmmTMB</em> article: GLMM Worked Examples</strong></a></p>
<ul>
<li>tutorial with examples and a breakdown of what the code is doing.</li>
</ul>
<div class="citation">
<strong><a href="#stats">Return to Statistics Packages</a></strong>
</div>
<h2 id="lme4pack"><em>lme4</em></h2>
<h3 id="linear-mixed-effects-models-using-eigen-and-s4">Linear Mixed-Effects Models using ‘Eigen’ and S4</h3>
<blockquote>
Fit linear and generalized linear mixed-effects models. The models and their components are represented using S4 classes and methods. The core computational algorithms are implemented using the ‘Eigen’ C++ library for numerical linear algebra and ‘RcppEigen’ “glue”. <span class="citation" data-cites="lme4">(<a href="#ref-lme4" role="doc-biblioref">Bates et al. 2015</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://lme4.r-universe.dev/lme4">lme4</a>
</div>
</blockquote>
<p><a href="https://lme4.r-universe.dev/lme4"><strong><em>lme4</em> website</strong></a></p>
<p><a href="https://lme4.r-universe.dev/lme4/doc/Theory.pdf"><strong><em>lme4</em> article: computational methods for mixed models</strong></a></p>
<p><a href="https://lme4.r-universe.dev/lme4/doc/lmer.pdf"><strong><em>lme4</em> article: fitting linear mixed-effects models using <em>lme4</em></strong></a></p>
<p><a href="https://lme4.r-universe.dev/articles/lme4/lmerperf.html"><strong><em>lme4</em> article: lme4 performance tips</strong></a></p>
<p><a href="https://lme4.r-universe.dev/lme4/doc/PLSvGLS.pdf"><strong><em>lme4</em> article: penalized least squares versus generalized least squares representations of linear mixed models</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/lme4/lme4.pdf"><strong><em>lme4</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#stats">Return to Statistics Packages</a></strong>
</div>
<h1 id="tables">Tables</h1>
<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Package</th>
<th style="text-align: center;">Section Link</th>
<th style="text-align: center;">Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong><em>DT</em></strong></td>
<td style="text-align: center;"><a href="#DTpack">link to <em>DT</em></a></td>
<td style="text-align: center;">sort and filter data tables</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>gtsummary</em></strong></td>
<td style="text-align: center;"><a href="#gtsummarypack">link to <em>gtsummary</em></a></td>
<td style="text-align: center;">analytical and summary tables</td>
</tr>
</tbody>
</table>
<h2 id="DTpack"><em>DT</em></h2>
<h3 id="an-r-interface-to-the-datatables-library">An R interface to the DataTables library</h3>
<blockquote>
The R package <strong><em>DT</em></strong> <span class="citation" data-cites="DT">(<a href="#ref-DT" role="doc-biblioref">Xie, Cheng, and Tan 2024</a>)</span> provides an R interface to the JavaScript library DataTables. R data objects (matrices or data frames) can be displayed as tables on HTML pages, and DataTables provides filtering, pagination, sorting, and many other features in the tables.
<div class="citation">
<strong>Source:</strong> <a href="https://rstudio.github.io/DT/">DT</a>
</div>
</blockquote>
<p><a href="https://rstudio.github.io/DT/"><strong><em>DT</em> website</strong></a></p>
<p><a href="https://rstudio.github.io/DT/options.html"><strong><em>DT</em> article: options</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/DT/DT.pdf"><strong><em>DT</em> reference manual</strong></a></p>
<details>
<summary>
<strong><em>DT</em> citation</strong>
</summary>
<blockquote>
<p>Xie Y, Cheng J, Tan X (2024). <em>DT: A Wrapper of the JavaScript Library
‘DataTables’</em>. R package version 0.33, <a href="https://CRAN.R-project.org/package=DT" class="uri">https://CRAN.R-project.org/package=DT</a>.</p>
</blockquote>
<p><strong>BibTeX entry</strong></p>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<div class="citation">
<strong><a href="#tables">Return to Tables Packages</a></strong>
</div>
<h2 id="gtsummarypack"><em>gtsummary</em></h2>
<blockquote>
The <strong><em>gtsummary</em></strong> <span class="citation" data-cites="gtsummary">(<a href="#ref-gtsummary" role="doc-biblioref">D. D. Sjoberg et al. 2021</a>)</span> package provides an elegant and flexible way to create publication-ready analytical and summary tables using the R programming language. The <strong><em>gtsummary</em></strong> package summarizes data sets, regression models, and more, using sensible defaults with highly customizable capabilities.
<div class="citation">
<strong>Source:</strong> <a href="https://www.danieldsjoberg.com/gtsummary/">gtsummary</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-167"></span>
<img src="images/gtsummary.png" alt="[gtsummary](https://www.danieldsjoberg.com/gtsummary/)" width="150%" />
<p class="caption">
Figure 105: <a href="https://www.danieldsjoberg.com/gtsummary/">gtsummary</a>
</p>
</div>
</div>
<p><a href="https://www.danieldsjoberg.com/gtsummary/"><strong><em>gtsummary</em> website</strong></a></p>
<p><a href="https://www.danieldsjoberg.com/gtsummary/articles/themes.html"><strong><em>gtsummary</em> article: themes</strong></a></p>
<p><a href="https://www.danieldsjoberg.com/gtsummary/articles/rmarkdown.html"><strong><em>gtsummary</em> article: <em>gtsummary</em> + Quarto/Rmarkdown</strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/gtsummary/gtsummary.pdf"><strong><em>gtsummary</em> reference manual</strong></a></p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/gt1.png" width="150%" style="display: block; margin: auto;" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-169"></span>
<img src="images/gt2.png" alt="[***gtsummary* cheatsheet**](https://github.com/rstudio/cheatsheets/blob/main/gtsummary.pdf)" width="150%" />
<p class="caption">
Figure 106: <a href="https://github.com/rstudio/cheatsheets/blob/main/gtsummary.pdf"><strong><em>gtsummary</em> cheatsheet</strong></a>
</p>
</div>
</div>
</details>
<div class="citation">
<strong><a href="#tables">Return to Tables Packages</a></strong>
</div>
<h1 id="telem">Telemetry</h1>
<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Package</th>
<th style="text-align: center;">Section Link</th>
<th style="text-align: center;">Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong><em>actel</em></strong></td>
<td style="text-align: center;"><a href="#actelpack">link to <em>actel</em></a></td>
<td style="text-align: center;">acoustic telemetry package</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>adehabitatHR</em></strong></td>
<td style="text-align: center;"><a href="#adehabitatHRpack">link to <em>adehabitatHR</em></a></td>
<td style="text-align: center;">home range estimation</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>adehabitatHS</em></strong></td>
<td style="text-align: center;"><a href="#adehabitatHSpack">link to <em>adehabitatHS</em></a></td>
<td style="text-align: center;">habitat selection analysis</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>adehabitatLT</em></strong></td>
<td style="text-align: center;"><a href="#adehabitatLTpack">link to <em>adehabitatLT</em></a></td>
<td style="text-align: center;">animal movement analysis</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>adehabitatMA</em></strong></td>
<td style="text-align: center;"><a href="#adehabitatMApack">link to <em>adehabitatMA</em></a></td>
<td style="text-align: center;">raster map manipulation</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>amt</em></strong></td>
<td style="text-align: center;"><a href="#amtpack">link to <em>amt</em></a></td>
<td style="text-align: center;">animal movement analysis</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>m2b</em></strong></td>
<td style="text-align: center;"><a href="#m2bpack">link to <em>m2b</em></a></td>
<td style="text-align: center;">behavior prediction</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>sftrack</em></strong></td>
<td style="text-align: center;"><a href="#sftrackpack">link to <em>sftrack</em></a></td>
<td style="text-align: center;">tracking and movement data</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong><em>SiMRiv</em></strong></td>
<td style="text-align: center;"><a href="#SiMRivpack">link to <em>SiMRiv</em></a></td>
<td style="text-align: center;">simulating movements in Rivers</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong><em>wildlifeDI</em></strong></td>
<td style="text-align: center;"><a href="#wildlifeDIpack">link to <em>wildlifeDI</em></a></td>
<td style="text-align: center;">dynamic interaction indices</td>
</tr>
</tbody>
</table>
<p><br />
</p>
<h2 id="actelpack"><em>actel</em></h2>
<blockquote>
Designed for studies where animals tagged with acoustic tags are expected to move through receiver arrays. Combines the advantages of automatic sorting and checking of animal movements with the possibility for user intervention on tags that deviate from expected behaviour. <span class="citation" data-cites="actel">(<a href="#ref-actel" role="doc-biblioref">Flavio and Baktoft 2021</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://hugomflavio.github.io/actel-website/">actel</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:actel"></span>
<img src="images/actel.png" alt="[***actel***](https://hugomflavio.github.io/actel-website/)" width="150%" />
<p class="caption">
Figure 107: <a href="https://hugomflavio.github.io/actel-website/"><strong><em>actel</em></strong></a>
</p>
</div>
</div>
<p><a href="https://hugomflavio.github.io/actel-website/"><strong><em>actel</em> website</strong></a></p>
<p><a href="https://hugomflavio.github.io/actel-website/manual-index.html"><strong><em>actel</em> user guide</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/actel/actel.pdf"><strong><em>actel</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#telem">Return to Telemetry Packages</a></strong>
</div>
<h2 id="adehabitatHRpack"><em>adehabitatHR</em></h2>
<h3 id="home-range-estimation">Home Range Estimation</h3>
<blockquote>
A collection of tools for the estimation of animals home range. <span class="citation" data-cites="habhr">(<a href="#ref-habhr" role="doc-biblioref">Calenge 2024a</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://clementcalenge.r-universe.dev/adehabitatHR">adehabitatHR</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-170"></span>
<img src="images/habhr.png" alt="[***adehabitatHR* example**](https://mgritts.github.io/2016/05/13/adehabitat-visualization/)" width="150%" />
<p class="caption">
Figure 108: <a href="https://mgritts.github.io/2016/05/13/adehabitat-visualization/"><strong><em>adehabitatHR</em> example</strong></a>
</p>
</div>
</div>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatHR"><strong><em>adehabitatHR</em> website</strong></a></p>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatHR/doc/adehabitatHR.pdf"><strong><em>adehabitatHR</em> article: home range estimation</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/adehabitatHR/adehabitatHR.pdf"><strong><em>adehabitatHR</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#telem">Return to Telemetry Packages</a></strong>
</div>
<h2 id="adehabitatHSpack"><em>adehabitatHS</em></h2>
<h3 id="analysis-of-habitat-selection-by-animals">Analysis of Habitat Selection by Animals</h3>
<blockquote>
A collection of tools for the analysis of habitat selection. <span class="citation" data-cites="habhs">(<a href="#ref-habhs" role="doc-biblioref">Calenge 2024b</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://clementcalenge.r-universe.dev/adehabitatHS">adehabitatHS</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-171"></span>
<img src="images/habhs.png" alt="[***adehabitatHS* example**](https://cran.r-project.org/web/packages/adehabitatHS/vignettes/adehabitatHS.pdf)" width="150%" />
<p class="caption">
Figure 109: <a href="https://cran.r-project.org/web/packages/adehabitatHS/vignettes/adehabitatHS.pdf"><strong><em>adehabitatHS</em> example</strong></a>
</p>
</div>
</div>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatHS"><strong><em>adehabitatHS</em> website</strong></a></p>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatHS/doc/adehabitatHS.pdf"><strong><em>adehabitatHS</em> article: exploratory analysis of the habitat selection by the wildlife in R</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/adehabitatHS/adehabitatHS.pdf"><strong><em>adehabitatHS</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#telem">Return to Telemetry Packages</a></strong>
</div>
<h2 id="adehabitatLTpack"><em>adehabitatLT</em></h2>
<h3 id="analysis-of-animal-movements">Analysis of Animal Movements</h3>
<blockquote>
A collection of tools for the analysis of animal movements. <span class="citation" data-cites="hablt">(<a href="#ref-hablt" role="doc-biblioref">Calenge 2024c</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://clementcalenge.r-universe.dev/adehabitatLT">adehabitatLT</a>
</div>
</blockquote>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatLT"><strong><em>adehabitatLT</em> website</strong></a></p>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatLT/doc/manual.html"><strong><em>adehabitatLT</em> article: help index</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/adehabitatLT/adehabitatLT.pdf"><strong><em>adehabitatLT</em> reference manual</strong></a></p>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatLT/doc/adehabitatLT.pdf"><strong>Analysis of Animal Movements in R: the <em>adehabitatLT</em> package</strong></a></p>
<div class="citation">
<strong><a href="#telem">Return to Telemetry Packages</a></strong>
</div>
<h2 id="adehabitatMApack"><em>adehabitatMA</em></h2>
<h3 id="tools-to-deal-with-raster-maps">Tools to Deal with Raster Maps</h3>
<blockquote>
A collection of tools to deal with raster maps. <span class="citation" data-cites="habma">(<a href="#ref-habma" role="doc-biblioref">Calenge 2024d</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://clementcalenge.r-universe.dev/adehabitatMA">adehabitatMA</a>
</div>
</blockquote>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatMA"><strong><em>adehabitatMA</em> website</strong></a></p>
<p><a href="https://clementcalenge.r-universe.dev/adehabitatMA/doc/adehabitatMA.pdf"><strong><em>adehabitatMA</em> article: tools for the analysis of mapped data </strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/adehabitatMA/adehabitatMA.pdf"><strong><em>adehabitatMA</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#telem">Return to Telemetry Packages</a></strong>
</div>
<h2 id="amtpack"><em>amt</em></h2>
<h3 id="animal-movement-tools">Animal Movement Tools</h3>
<blockquote>
Manage and analyze animal movement data. The functionality of <strong><em>amt</em></strong> <span class="citation" data-cites="amt">(<a href="#ref-amt" role="doc-biblioref">Signer, Fieberg, and Avgar 2019</a>)</span> includes methods to calculate home ranges, track statistics (e.g. step lengths, speed, or turning angles), prepare data for fitting habitat selection analyses, and simulation of space-use from fitted step-selection functions.
<div class="citation">
<strong>Source:</strong> <a href="https://jmsigner.r-universe.dev/amt">amt</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-172"></span>
<img src="images/amt.png" alt="[***amt* example**](https://cran.r-project.org/web/packages/amt/vignettes/p1_getting_started.html)" width="150%" />
<p class="caption">
Figure 110: <a href="https://cran.r-project.org/web/packages/amt/vignettes/p1_getting_started.html"><strong><em>amt</em> example</strong></a>
</p>
</div>
</div>
<p><a href="https://jmsigner.r-universe.dev/amt"><strong><em>amt</em> website</strong></a></p>
<p><a href="https://jmsigner.r-universe.dev/articles/amt/p1_getting_started.html"><strong><em>amt</em> article: getting started</strong></a></p>
<p><a href="https://jmsigner.r-universe.dev/articles/amt/p2_hr.html"><strong><em>amt</em> article: calculating home-range overlaps with amt</strong></a></p>
<p><a href="https://jmsigner.r-universe.dev/articles/amt/p6_uhc_plots.html"><strong><em>amt</em> article: used habitat calibration plots</strong></a></p>
<p><a href="https://jmsigner.r-universe.dev/amt#manual"><strong><em>amt</em> help manuals</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/amt/amt.pdf"><strong><em>amt</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#telem">Return to Telemetry Packages</a></strong>
</div>
<h2 id="m2bpack"><em>m2b</em></h2>
<h3 id="movement-to-behaviour-inference-using-random-forest">Movement to Behaviour Inference using Random Forest</h3>
<blockquote>
Prediction of behaviour from movement characteristics using observation and random forest for the analyses of movement data in ecology. From movement information (speed, bearing…) the model predicts the observed behaviour (movement, foraging…) using random forest. The model can then extrapolate behavioural information to movement data without direct observation of behaviours. The specificity of this method relies on the derivation of multiple predictor variables from the movement data over a range of temporal windows. This procedure allows to capture as much information as possible on the changes and variations of movement and ensures the use of the random forest algorithm to its best capacity. The method is very generic, applicable to any set of data providing movement data together with observation of behaviour. <span class="citation" data-cites="m2b">(<a href="#ref-m2b" role="doc-biblioref">Dubroca and Thiebault 2017</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://ldbk.r-universe.dev/m2b">m2b</a>
</div>
</blockquote>
<p><a href="https://ldbk.r-universe.dev/m2b"><strong><em>m2b</em> website</strong></a></p>
<p><a href="https://ldbk.r-universe.dev/m2b/doc/tutorial.pdf"><strong><em>m2b</em> article: tutorial</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/m2b/m2b.pdf"><strong><em>m2b</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#overview">Return to Package Overview Table</a></strong>
</div>
<h2 id="sftrackpack"><em>sftrack</em></h2>
<blockquote>
<strong><em>sftrack</em></strong> <span class="citation" data-cites="sftrack">(<a href="#ref-sftrack" role="doc-biblioref">Boone, Joo, and Basille 2023</a>)</span> provides modern classes for <strong>tracking and movement data</strong>, relying on <strong><em>sf</em></strong> spatial infrastructure. Tracking data are made of tracks, i.e. series of locations with at least 2-dimensional spatial coordinates (x,y), a time index (t), and individual identification (id) of the object being monitored; movement data are made of trajectories, i.e. the line representation of the path, composed by steps (the straight-line segments connecting successive locations). <strong><em>sftrack</em></strong> is designed to handle movement of both living organisms and inanimate objects.
<div class="citation">
<strong>Source:</strong> <a href="https://mablab.org/sftrack/">sftrack</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-173"></span>
<img src="images/sftrack.png" alt="[***sftrack* example**](https://mablab.org/sftrack/)" width="150%" />
<p class="caption">
Figure 111: <a href="https://mablab.org/sftrack/"><strong><em>sftrack</em> example</strong></a>
</p>
</div>
</div>
<p><a href="https://mablab.org/sftrack/"><strong><em>sftrack</em> website</strong></a></p>
<p><a href="https://mablab.r-universe.dev/articles/sftrack/sftrack1_overview.html"><strong><em>sftrack</em> article: overview</strong></a></p>
<p><a href="https://mablab.org/sftrack/articles/sftrack2_reading.html"><strong><em>sftrack</em> article: reading in an <em>sftrack</em></strong></a></p>
<p><a href="https://mablab.org/sftrack/articles/sftrack3_workingwith.html"><strong><em>sftrack</em> article: structure of <em>sftrack/sftraj</em> objects</strong></a></p>
<p><a href="https://mablab.org/sftrack/articles/sftrack4_groups.html"><strong><em>sftrack</em> article: fantastic groups and how to use them</strong></a></p>
<p><a href="https://mablab.org/sftrack/articles/sftrack5_spatial.html"><strong><em>sftrack</em> article: getting spatial with <em>sftrack</em></strong></a></p>
<p><a href="https://cloud.r-project.org/web/packages/sftrack/sftrack.pdf"><strong><em>sftrack</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#telem">Return to Telemetry Packages</a></strong>
</div>
<h2 id="SiMRivpack"><em>SiMRiv</em></h2>
<h3 id="simulating-multistate-movements-in-riverheterogeneous-landscapes">Simulating Multistate Movements in River/Heterogeneous Landscapes</h3>
<blockquote>
Provides functions to generate and analyze spatially-explicit individual-based multistate movements in rivers, heterogeneous and homogeneous spaces. This is done by incorporating landscape bias on local behaviour, based on resistance rasters. Although originally conceived and designed to simulate trajectories of species constrained to linear habitats/dendritic ecological networks (e.g. river networks), the simulation algorithm is built to be highly flexible and can be applied to any (aquatic, semi-aquatic or terrestrial) organism, independently on the landscape in which it moves. Thus, the user will be able to use the package to simulate movements either in homogeneous landscapes, heterogeneous landscapes (e.g. semi-aquatic animal moving mainly along rivers but also using the matrix), or even in highly contrasted landscapes (e.g. fish in a river network). The algorithm and its input parameters are the same for all cases, so that results are comparable. Simulated trajectories can then be used as mechanistic null models (Potts &amp; Lewis 2014, <a href="doi:10.1098/rspb.2014.0231" class="uri">doi:10.1098/rspb.2014.0231</a>) to test a variety of ‘Movement Ecology’ hypotheses (Nathan et al. 2008, <a href="doi:10.1073/pnas.0800375105" class="uri">doi:10.1073/pnas.0800375105</a>), including landscape effects (e.g. resources, infrastructures) on animal movement and species site fidelity, or for predictive purposes (e.g. road mortality risk, dispersal/connectivity). The package should be relevant to explore a broad spectrum of ecological phenomena, such as those at the interface of animal behaviour, management, landscape and movement ecology, disease and invasive species spread, and population dynamics. <span class="citation" data-cites="SiMRiv">(<a href="#ref-SiMRiv" role="doc-biblioref">Quaglietta and Porto 2024</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://miguel-porto.r-universe.dev/SiMRiv">SiMRiv</a>
</div>
</blockquote>
<div class="layout-chunk" data-layout="l-body">
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-174"></span>
<img src="images/SiMRiv.png" alt="[***SiMRiv* example**](https://movementecologyjournal.biomedcentral.com/articles/10.1186/s40462-019-0154-8)" width="150%" />
<p class="caption">
Figure 112: <a href="https://movementecologyjournal.biomedcentral.com/articles/10.1186/s40462-019-0154-8"><strong><em>SiMRiv</em> example</strong></a>
</p>
</div>
</div>
<p><a href="https://miguel-porto.r-universe.dev/SiMRiv"><strong><em>SiMRiv</em> website</strong></a></p>
<p><a href="https://miguel-porto.r-universe.dev/SiMRiv/doc/SiMRiv.pdf"><strong><em>SiMRiv</em> article: usage of the <em>SiMRiv</em> package</strong></a></p>
<p><a href="https://miguel-porto.r-universe.dev/SiMRiv#manual"><strong><em>SiMRiv</em> help manuals</strong></a></p>
<p><a href="https://cran.r-project.org/web/packages/SiMRiv/SiMRiv.pdf"><strong><em>SiMRiv</em> reference manual</strong></a></p>
<div class="citation">
<strong><a href="#telem">Return to Telemetry Packages</a></strong>
</div>
<h2 id="wildlifeDIpack"><em>wildlifeDI</em></h2>
<h3 id="calculate-indices-of-dynamic-interaction-for-wildlife-tracking-data">Calculate Indices of Dynamic Interaction for Wildlife Tracking Data</h3>
<blockquote>
Dynamic interaction refers to spatial-temporal associations in the movements of two (or more) animals. This package provides tools for calculating a suite of indices used for quantifying dynamic interaction with wildlife telemetry data. For more information on each of the methods employed see the references within. The package (as of version &gt;= 0.3) also has new tools for automating contact analysis in large tracking datasets. The package (as of version 1.0) uses the ‘move2’ class of objects for working with tracking dataset. <span class="citation" data-cites="wildlifeDI1">(<a href="#ref-wildlifeDI1" role="doc-biblioref">Long et al. 2014</a>)</span> <span class="citation" data-cites="wildlifeDI2">(<a href="#ref-wildlifeDI2" role="doc-biblioref">Long et al. 2022</a>)</span>
<div class="citation">
<strong>Source:</strong> <a href="https://jedalong.r-universe.dev/wildlifeDI">wildlifeDI</a>
</div>
</blockquote>
<p><a href="https://jedalong.r-universe.dev/wildlifeDI"><strong><em>wildlifeDI</em> website</strong></a></p>
<p><a href="https://jedalong.r-universe.dev/articles/wildlifeDI/wildlifeDI-vignette-contact_analysis.html"><strong><em>wildlifeDI</em> article: contact analysis workflow</strong></a></p>
<p><a href="https://jedalong.r-universe.dev/articles/wildlifeDI/wildlifeDI-vignette.html"><strong><em>wildlifeDI</em> article: analysis of dynamic interaction patterns in wildlife tracking data</strong></a></p>
<div class="citation">
<strong><a href="#tidyverse">Return to Top</a></strong>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-rticles" class="csl-entry" role="listitem">
Allaire, JJ, Yihui Xie, Christophe Dervieux, R Foundation, Hadley Wickham, Journal of Statistical Software, Ramnath Vaidyanathan, et al. 2024. <em>Rticles: Article Formats for r Markdown</em>. <a href="https://CRAN.R-project.org/package=rticles">https://CRAN.R-project.org/package=rticles</a>.
</div>
<div id="ref-ggthemes" class="csl-entry" role="listitem">
Arnold, Jeffrey B. 2024. <em>Ggthemes: Extra Themes, Scales and Geoms for ’Ggplot2’</em>. <a href="https://jrnold.github.io/ggthemes/">https://jrnold.github.io/ggthemes/</a>.
</div>
<div id="ref-lme4" class="csl-entry" role="listitem">
Bates, Douglas, Martin Machler, Ben Bolker, and Steve Walker. 2015. <span>“Fitting Linear Mixed-Effects Models Using <span class="nocase">lme4</span>.”</span> <em>Journal of Statistical Software</em> 67 (1): 1–48. <a href="https://doi.org/10.18637/jss.v067.i01">https://doi.org/10.18637/jss.v067.i01</a>.
</div>
<div id="ref-bbmle" class="csl-entry" role="listitem">
Bolker, Ben, and R Development Core Team. 2025. <em>Bbmle: Tools for General Maximum Likelihood Estimation</em>. <a href="https://github.com/bbolker/bbmle">https://github.com/bbolker/bbmle</a>.
</div>
<div id="ref-sftrack" class="csl-entry" role="listitem">
Boone, Matthew, Rocío Joo, and Mathieu Basille. 2023. <em><span class="nocase">sftrack</span>: Modern Classes for Tracking and Movement Data</em>. <a href="https://mablab.org/sftrack/">https://mablab.org/sftrack/</a>.
</div>
<div id="ref-glmmTMB" class="csl-entry" role="listitem">
Brooks, Mollie E., Kasper Kristensen, Koen J. van Benthem, Arni Magnusson, Casper W. Berg, Anders Nielsen, Hans J. Skaug, Martin Maechler, and Benjamin M. Bolker. 2017. <span>“<span class="nocase">glmmTMB</span> Balances Speed and Flexibility Among Packages for Zero-Inflated Generalized Linear Mixed Modeling.”</span> <em>The R Journal</em> 9 (2): 378–400. <a href="https://doi.org/10.32614/RJ-2017-066">https://doi.org/10.32614/RJ-2017-066</a>.
</div>
<div id="ref-habhr" class="csl-entry" role="listitem">
Calenge, Clement. 2024a. <em>adehabitatHR: Home Range Estimation</em>. <a href="https://github.com/clementcalenge/adehabitathr">https://github.com/clementcalenge/adehabitathr</a>.
</div>
<div id="ref-habhs" class="csl-entry" role="listitem">
———. 2024b. <em>adehabitatHS: Analysis of Habitat Selection by Animals</em>. <a href="https://github.com/clementcalenge/adehabitaths">https://github.com/clementcalenge/adehabitaths</a>.
</div>
<div id="ref-hablt" class="csl-entry" role="listitem">
———. 2024c. <em>adehabitatLT: Analysis of Animal Movements</em>. <a href="https://github.com/clementcalenge/adehabitatlt">https://github.com/clementcalenge/adehabitatlt</a>.
</div>
<div id="ref-habma" class="csl-entry" role="listitem">
———. 2024d. <em>adehabitatMA: Tools to Deal with Raster Maps</em>. <a href="https://github.com/clementcalenge/adehabitatma">https://github.com/clementcalenge/adehabitatma</a>.
</div>
<div id="ref-geomtextpath" class="csl-entry" role="listitem">
Cameron, Allan, and Teun van den Brand. 2022. <em>Geomtextpath: Use Curved Text in Ggplot2</em>. <a href="https://allancameron.github.io/geomtextpath/">https://allancameron.github.io/geomtextpath/</a>.
</div>
<div id="ref-ggeasy" class="csl-entry" role="listitem">
Carroll, Jonathan, Alicia Schep, and Jonathan Sidi. 2024. <em>Ggeasy: Easy Access to ’Ggplot2’ Commands</em>. <a href="https://CRAN.R-project.org/package=ggeasy">https://CRAN.R-project.org/package=ggeasy</a>.
</div>
<div id="ref-shiny" class="csl-entry" role="listitem">
Chang, Winston, Joe Cheng, JJ Allaire, Carson Sievert, Barret Schloerke, Yihui Xie, Jeff Allen, Jonathan McPherson, Alan Dipert, and Barbara Borges. 2024. <em>Shiny: Web Application Framework for r</em>. <a href="https://CRAN.R-project.org/package=shiny">https://CRAN.R-project.org/package=shiny</a>.
</div>
<div id="ref-leaflet" class="csl-entry" role="listitem">
Cheng, Joe, Barret Schloerke, Bhaskar Karambelkar, and Yihui Xie. 2024. <em>Leaflet: Create Interactive Web Maps with the JavaScript ’Leaflet’ Library</em>. <a href="https://CRAN.R-project.org/package=leaflet">https://CRAN.R-project.org/package=leaflet</a>.
</div>
<div id="ref-ggsankey" class="csl-entry" role="listitem">
de Vries, Pepijn. 2024. <em>Ggsankeyfier: Create Sankey and Alluvial Diagrams Using ’Ggplot2’</em>. <a href="https://CRAN.R-project.org/package=ggsankeyfier">https://CRAN.R-project.org/package=ggsankeyfier</a>.
</div>
<div id="ref-m2b" class="csl-entry" role="listitem">
Dubroca, Laurent, and Andréa Thiebault. 2017. <em>M2b: Movement to Behaviour Inference Using Random Forest</em>. <a href="https://github.com/ldbk/m2b">https://github.com/ldbk/m2b</a>.
</div>
<div id="ref-ggpattern" class="csl-entry" role="listitem">
FC, Mike, Trevor L. Davis, and ggplot2 authors. 2025. <em>Ggpattern: ’Ggplot2’ Pattern Geoms</em>. <a href="https://github.com/trevorld/ggpattern">https://github.com/trevorld/ggpattern</a>.
</div>
<div id="ref-actel" class="csl-entry" role="listitem">
Flavio, Hugo, and Henrik Baktoft. 2021. <span>“<span class="nocase">actel</span>: Standardised Analysis of Acoustic Telemetry Data from Animals Moving Through Receiver Arrays.”</span> <em>Methods in Ecology and Evolution</em> 12 (1): 196–203. <a href="https://doi.org/10.1111/2041-210X.13503">https://doi.org/10.1111/2041-210X.13503</a>.
</div>
<div id="ref-fishMod" class="csl-entry" role="listitem">
Foster, Scott D. 2024. <em>fishMod: Fits Poisson-Sum-of-Gammas GLMs, Tweedie GLMs, and Delta Log-Normal Models</em>. <a href="https://CRAN.R-project.org/package=fishMod">https://CRAN.R-project.org/package=fishMod</a>.
</div>
<div id="ref-ggiraph" class="csl-entry" role="listitem">
Gohel, David, and Panagiotis Skintzos. 2024. <em>Ggiraph: Make ’Ggplot2’ Graphics Interactive</em>. <a href="https://davidgohel.github.io/ggiraph/">https://davidgohel.github.io/ggiraph/</a>.
</div>
<div id="ref-lubridate" class="csl-entry" role="listitem">
Grolemund, Garrett, and Hadley Wickham. 2011. <span>“Dates and Times Made Easy with <span class="nocase">lubridate</span>.”</span> <em>Journal of Statistical Software</em> 40 (3): 1–25. <a href="https://www.jstatsoft.org/v40/i03/">https://www.jstatsoft.org/v40/i03/</a>.
</div>
<div id="ref-EchoviewR" class="csl-entry" role="listitem">
Harrison, Lisa-Marie K, Martin J Cox, Georg Skaret, and Robert Harcourt. 2015. <span>“The r Package EchoviewR for Automated Processing of Active Acoustic Data Using Echoview.”</span> <em>Frontiers in Marine Science</em> 2 (15). <a href="https://doi.org/10.3389/fmars.2015.00015">https://doi.org/10.3389/fmars.2015.00015</a>.
</div>
<div id="ref-DHARMa" class="csl-entry" role="listitem">
Hartig, Florian. 2024. <em>DHARMa: Residual Diagnostics for Hierarchical (Multi-Level / Mixed) Regression Models</em>. <a href="https://CRAN.R-project.org/package=DHARMa">https://CRAN.R-project.org/package=DHARMa</a>.
</div>
<div id="ref-directlabels" class="csl-entry" role="listitem">
Hocking, Toby Dylan. 2025. <em>Directlabels: Direct Labels for Multicolor Plots</em>. <a href="https://github.com/tdhock/directlabels">https://github.com/tdhock/directlabels</a>.
</div>
<div id="ref-paletteer" class="csl-entry" role="listitem">
Hvitfeldt, Emil. 2021. <em>Paletteer: Comprehensive Collection of Color Palettes</em>. <a href="https://github.com/EmilHvitfeldt/paletteer">https://github.com/EmilHvitfeldt/paletteer</a>.
</div>
<div id="ref-plotrix" class="csl-entry" role="listitem">
J, Lemon. 2006. <span>“Plotrix: A Package in the Red Light District of r.”</span> <em>R-News</em> 6 (4): 8–12.
</div>
<div id="ref-ggwordcloud" class="csl-entry" role="listitem">
Le Pennec, Erwan, and Kamil Slowikowski. 2024. <em>Ggwordcloud: A Word Cloud Geom for ’Ggplot2’</em>. <a href="https://github.com/lepennec/ggwordcloud">https://github.com/lepennec/ggwordcloud</a>.
</div>
<div id="ref-emmeans" class="csl-entry" role="listitem">
Lenth, Russell V. 2024. <em>Emmeans: Estimated Marginal Means, Aka Least-Squares Means</em>. <a href="https://CRAN.R-project.org/package=emmeans">https://CRAN.R-project.org/package=emmeans</a>.
</div>
<div id="ref-CatDyn4" class="csl-entry" role="listitem">
Lin, Y-J., W-N. Tzeng, Y-S. Han, and Ruben H. Roa-Ureta. 2017. <span>“A Stock Assessment Model for Transit Stock Fisheries with Explicit Immigration and Emigration Dynamics: Application to Upstream Waves of Glass Eels.”</span> <em>Fisheries Research</em> 195: 130–40. <a href="http://dx.doi.org/10.1016/j.fishres.2017.07.009">http://dx.doi.org/10.1016/j.fishres.2017.07.009</a>.
</div>
<div id="ref-wildlifeDI1" class="csl-entry" role="listitem">
Long, Jed, Trisalyn Nelson, Stephen Webb, and Kenneth Gee. 2014. <span>“A Critical Examination of Indices of Dynamic Interaction for Wildlife Telemetry Data.”</span> <em>Journal of Animal Ecology</em> 83: 1216–33.
</div>
<div id="ref-wildlifeDI2" class="csl-entry" role="listitem">
Long, Jed, Stephen Webb, Seth Harju, and Kenneth Gee. 2022. <span>“Analyzing Contacts and Behavior from High Frequency Tracking Data Using the wildlifeDI r Package.”</span> <em>Geographical Analysis</em> 54: 648–63.
</div>
<div id="ref-AICcmodavg" class="csl-entry" role="listitem">
Mazerolle, Marc J. 2023. <em>AICcmodavg: Model Selection and Multimodel Inference Based on (q)AIC(c)</em>. <a href="https://cran.r-project.org/package=AICcmodavg">https://cran.r-project.org/package=AICcmodavg</a>.
</div>
<div id="ref-esquisse" class="csl-entry" role="listitem">
Meyer, Fanny, and Victor Perrier. 2025. <em>Esquisse: Explore and Visualize Your Data Interactively</em>. <a href="https://dreamrs.github.io/esquisse/">https://dreamrs.github.io/esquisse/</a>.
</div>
<div id="ref-rayshader" class="csl-entry" role="listitem">
Morgan-Wall, Tyler. 2025. <em>Rayshader: Create Maps and Visualize Data in 2D and 3D</em>. <a href="https://www.rayshader.com">https://www.rayshader.com</a>.
</div>
<div id="ref-tibble" class="csl-entry" role="listitem">
Müller, Kirill, and Hadley Wickham. 2023. <em>Tibble: Simple Data Frames</em>. <a href="https://CRAN.R-project.org/package=tibble">https://CRAN.R-project.org/package=tibble</a>.
</div>
<div id="ref-RFishBC" class="csl-entry" role="listitem">
Ogle, Derek H. 2023. <em>RFishBC: Back-Calculation of Fish Length</em>. <a href="https://fishr-core-team.github.io/RFishBC/">https://fishr-core-team.github.io/RFishBC/</a>.
</div>
<div id="ref-FSA" class="csl-entry" role="listitem">
Ogle, Derek H., Jason C. Doll, A. Powell Wheeler, and Alexis Dinno. 2025. <em>FSA: Simple Fisheries Stock Assessment Methods</em>. <a href="https://fishr-core-team.github.io/FSA/">https://fishr-core-team.github.io/FSA/</a>.
</div>
<div id="ref-ggstatsplot" class="csl-entry" role="listitem">
Patil, Indrajeet. 2021. <span>“<span class="nocase">Visualizations with statistical details: The <span class="nocase">’ggstatsplot’</span> approach</span>.”</span> <em><span>Journal of Open Source Software</span></em> 6 (61): 3167. <a href="https://doi.org/10.21105/joss.03167">https://doi.org/10.21105/joss.03167</a>.
</div>
<div id="ref-sf" class="csl-entry" role="listitem">
Pebesma, Edzer, and Roger Bivand. 2023. <em><span class="nocase">Spatial Data Science: With applications in R</span></em>. <span>Chapman and Hall/CRC</span>. <a href="https://doi.org/10.1201/9780429459016">https://doi.org/10.1201/9780429459016</a>.
</div>
<div id="ref-ggforce" class="csl-entry" role="listitem">
Pedersen, Thomas Lin. 2024a. <em>Ggforce: Accelerating ’Ggplot2’</em>. <a href="https://ggforce.data-imaginist.com">https://ggforce.data-imaginist.com</a>.
</div>
<div id="ref-ggraph" class="csl-entry" role="listitem">
———. 2024b. <em>Ggraph: An Implementation of Grammar of Graphics for Graphs and Networks</em>. <a href="https://ggraph.data-imaginist.com">https://ggraph.data-imaginist.com</a>.
</div>
<div id="ref-patchwork" class="csl-entry" role="listitem">
———. 2024c. <em>Patchwork: The Composer of Plots</em>. <a href="https://CRAN.R-project.org/package=patchwork">https://CRAN.R-project.org/package=patchwork</a>.
</div>
<div id="ref-gganimate" class="csl-entry" role="listitem">
Pedersen, Thomas Lin, and David Robinson. 2024. <em>Gganimate: A Grammar of Animated Graphics</em>. <a href="https://gganimate.com">https://gganimate.com</a>.
</div>
<div id="ref-showtext" class="csl-entry" role="listitem">
Qiu, Yixuan. 2024. <em>Showtext: Using Fonts More Easily in r Graphs</em>. <a href="https://CRAN.R-project.org/package=showtext">https://CRAN.R-project.org/package=showtext</a>.
</div>
<div id="ref-SiMRiv" class="csl-entry" role="listitem">
Quaglietta, Lorenzo, and Miguel Porto. 2024. <em><span>SiMRiv</span>: Individual-Based, Spatially-Explicit Simulation and Analysis of Multi-State Movements in River Networks and Heterogeneous Landscapes</em>. <a href="https://CRAN.R-project.org/package=SiMRiv">https://CRAN.R-project.org/package=SiMRiv</a>.
</div>
<div id="ref-gamlss" class="csl-entry" role="listitem">
R. A. Rigby, and D. M. Stasinopoulos. 2005. <span>“Generalized Additive Models for Location, Scale and Shape,(with Discussion).”</span> <em>Applied Statistics</em> 54.3: 507–54.
</div>
<div id="ref-mapdata" class="csl-entry" role="listitem">
Richard A. Becker, Original S code by, and Allan R. Wilks. R version by Ray Brownrigg. 2022. <em>Mapdata: Extra Map Databases</em>. <a href="https://CRAN.R-project.org/package=mapdata">https://CRAN.R-project.org/package=mapdata</a>.
</div>
<div id="ref-maps" class="csl-entry" role="listitem">
Richard A. Becker, Original S code by, Allan R. Wilks. R version by Ray Brownrigg. Enhancements by Thomas P Minka, and Alex Deckmyn. 2024. <em>Maps: Draw Geographical Maps</em>. <a href="https://CRAN.R-project.org/package=maps">https://CRAN.R-project.org/package=maps</a>.
</div>
<div id="ref-CatDyn1" class="csl-entry" role="listitem">
Roa-Ureta, Ruben H. 2011. <span>“Modeling in-Season Pulses of Recruitment and Hyperstability-Hyperdepletion in the Loligo Gahi Fishery of the Falkland Islands with Generalized Depletion Models.”</span> <em>ICES Journal of Marine Science</em> 69 (8): 1403–15. <a href="https://doi.org/10.1093/icesjms/fss110">https://doi.org/10.1093/icesjms/fss110</a>.
</div>
<div id="ref-CatDyn2" class="csl-entry" role="listitem">
———. 2015. <span>“Stock Assessment of the Spanish Mackerel (Scomberomorus Commerson) in Saudi Waters of the Arabian Gulf with Generalized Depletion Models Under Data-Limited Conditions.”</span> <em>Fisheries Research</em> 171 (Special Issue): 68–77. <a href="http://dx.doi.org/10.1016/j.fishres.2014.08.014">http://dx.doi.org/10.1016/j.fishres.2014.08.014</a>.
</div>
<div id="ref-CatDyn3" class="csl-entry" role="listitem">
Roa-Ureta, Ruben H., C. Molinet, N. Bahamonde, and P. Araya. 2015. <span>“Hierarchical Statistical Framework to Combine Generalized Depletion Models and Biomass Dynamic Models in the Stock Assessment of the Chilean Sea Urchin (Loxechinus Albus) Fishery.”</span> <em>Fisheries Research</em> 171 (Special Issue): 59–67. <a href="http://dx.doi.org/10.1016/j.fishres.2014.12.006">http://dx.doi.org/10.1016/j.fishres.2014.12.006</a>.
</div>
<div id="ref-hrbrthemes" class="csl-entry" role="listitem">
Rudis, Bob. 2024. <em>Hrbrthemes: Additional Themes, Theme Components and Utilities for ’Ggplot2’</em>. <a href="https://CRAN.R-project.org/package=hrbrthemes">https://CRAN.R-project.org/package=hrbrthemes</a>.
</div>
<div id="ref-fishualize" class="csl-entry" role="listitem">
Schiettekatte, Nina M. D., Simon J. Brandl, and Jordan M. Casey. 2022. <em>Fishualize: Color Palettes Based on Fish Species</em>. <a href="https://CRAN.R-project.org/package=fishualize">https://CRAN.R-project.org/package=fishualize</a>.
</div>
<div id="ref-ggbreak" class="csl-entry" role="listitem">
Shuangbin Xu, Meijun Chen, Tingze Feng, Li Zhan, Lang Zhou, and Guangchuang Yu. 2021. <span>“Use Ggbreak to Effectively Utilize Plotting Space to Deal with Large Datasets and Outliers.”</span> <em>Frontiers in Genetics</em> 12: 774846. <a href="https://doi.org/10.3389/fgene.2021.774846">https://doi.org/10.3389/fgene.2021.774846</a>.
</div>
<div id="ref-amt" class="csl-entry" role="listitem">
Signer, Johannes, John Fieberg, and Tal Avgar. 2019. <span>“Animal Movement Tools (Amt): R Package for Managing Tracking Data and Conducting Habitat Selection Analyses.”</span> <em>Ecology and Evolution</em> 9: 880–90.
</div>
<div id="ref-gtsummary" class="csl-entry" role="listitem">
Sjoberg, Daniel D., Karissa Whiting, Michael Curry, Jessica A. Lavery, and Joseph Larmarange. 2021. <span>“Reproducible Summary Tables with the Gtsummary Package.”</span> <em><span>The R Journal</span></em> 13: 570–80. <a href="https://doi.org/10.32614/RJ-2021-053">https://doi.org/10.32614/RJ-2021-053</a>.
</div>
<div id="ref-ggbump" class="csl-entry" role="listitem">
Sjoberg, David. 2020. <em>Ggbump: Bump Chart and Sigmoid Curves</em>. <a href="https://CRAN.R-project.org/package=ggbump">https://CRAN.R-project.org/package=ggbump</a>.
</div>
<div id="ref-ggrepel" class="csl-entry" role="listitem">
Slowikowski, Kamil. 2024. <em>Ggrepel: Automatically Position Non-Overlapping Text Labels with ’Ggplot2’</em>.
</div>
<div id="ref-BayesGrowth1" class="csl-entry" role="listitem">
Smart, Jonathan. 2023. <em>BayesGrowth: Estimate Fish Growth Using MCMC Analysis</em>. <a href="https://github.com/jonathansmart/BayesGrowth">https://github.com/jonathansmart/BayesGrowth</a>.
</div>
<div id="ref-BayesGrowth2" class="csl-entry" role="listitem">
Smart, Jonathan, and Gretchen Grammer. 2021. <span>“Modernising Fish and Shark Growth Curves with Bayesian Length-at-Age Models.”</span> <em>PLOS ONE</em> 16 (2): e0246734. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0246734">https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0246734</a>.
</div>
<div id="ref-plot3D" class="csl-entry" role="listitem">
Soetaert, Karline. 2024. <em>plot3D: Plotting Multi-Dimensional Data</em>. <a href="https://CRAN.R-project.org/package=plot3D">https://CRAN.R-project.org/package=plot3D</a>.
</div>
<div id="ref-pbapply" class="csl-entry" role="listitem">
Solymos, Peter, and Zygmunt Zawadzki. 2023. <em>Pbapply: Adding Progress Bar to Apply Functions</em>. <a href="https://CRAN.R-project.org/package=pbapply">https://CRAN.R-project.org/package=pbapply</a>.
</div>
<div id="ref-tmap" class="csl-entry" role="listitem">
Tennekes, Martijn. 2018. <span>“<span class="nocase">tmap</span>: Thematic Maps in <span>R</span>.”</span> <em>Journal of Statistical Software</em> 84 (6): 1–39. <a href="https://doi.org/10.18637/jss.v084.i06">https://doi.org/10.18637/jss.v084.i06</a>.
</div>
<div id="ref-FESta" class="csl-entry" role="listitem">
Varghese, Eldho, Sathianandan T V, and Jayasankar J. 2020. <em>FESta: Fishing Effort Standardisation</em>. <a href="https://CRAN.R-project.org/package=FESta">https://CRAN.R-project.org/package=FESta</a>.
</div>
<div id="ref-ggFishPlots" class="csl-entry" role="listitem">
Vihtakari, Mikko. 2024. <em>ggFishPlots: Visualise and Calculate Life History Parameters for Fisheries Science Using ’Ggplot2’</em>. <a href="https://deepwaterimr.github.io/ggFishPlots/">https://deepwaterimr.github.io/ggFishPlots/</a>.
</div>
<div id="ref-ggplot2" class="csl-entry" role="listitem">
Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York. <a href="https://ggplot2.tidyverse.org">https://ggplot2.tidyverse.org</a>.
</div>
<div id="ref-forcats" class="csl-entry" role="listitem">
———. 2023a. <em>Forcats: Tools for Working with Categorical Variables (Factors)</em>. <a href="https://CRAN.R-project.org/package=forcats">https://CRAN.R-project.org/package=forcats</a>.
</div>
<div id="ref-stringr" class="csl-entry" role="listitem">
———. 2023b. <em>Stringr: Simple, Consistent Wrappers for Common String Operations</em>. <a href="https://CRAN.R-project.org/package=stringr">https://CRAN.R-project.org/package=stringr</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy DAgostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span class="nocase">tidyverse</span>.”</span> <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-dplyr" class="csl-entry" role="listitem">
Wickham, Hadley, Romain François, Lionel Henry, Kirill Müller, and Davis Vaughan. 2023. <em>Dplyr: A Grammar of Data Manipulation</em>. <a href="https://CRAN.R-project.org/package=dplyr">https://CRAN.R-project.org/package=dplyr</a>.
</div>
<div id="ref-purrr" class="csl-entry" role="listitem">
Wickham, Hadley, and Lionel Henry. 2023. <em>Purrr: Functional Programming Tools</em>. <a href="https://CRAN.R-project.org/package=purrr">https://CRAN.R-project.org/package=purrr</a>.
</div>
<div id="ref-readr" class="csl-entry" role="listitem">
Wickham, Hadley, Jim Hester, and Jennifer Bryan. 2024. <em>Readr: Read Rectangular Text Data</em>. <a href="https://CRAN.R-project.org/package=readr">https://CRAN.R-project.org/package=readr</a>.
</div>
<div id="ref-tidyr" class="csl-entry" role="listitem">
Wickham, Hadley, Davis Vaughan, and Maximilian Girlich. 2024. <em>Tidyr: Tidy Messy Data</em>. <a href="https://CRAN.R-project.org/package=tidyr">https://CRAN.R-project.org/package=tidyr</a>.
</div>
<div id="ref-ggtext" class="csl-entry" role="listitem">
Wilke, Claus O., and Brenton M. Wiernik. 2025. <em>Ggtext: Improved Text Rendering Support for ’Ggplot2’</em>. <a href="https://github.com/wilkelab/ggtext">https://github.com/wilkelab/ggtext</a>.
</div>
<div id="ref-tinytex2" class="csl-entry" role="listitem">
Xie, Yihui. 2019. <span>“TinyTeX: A Lightweight, Cross-Platform, and Easy-to-Maintain LaTeX Distribution Based on TeX Live.”</span> <em>TUGboat</em> 40 (1): 30–32. <a href="https://tug.org/TUGboat/Contents/contents40-1.html">https://tug.org/TUGboat/Contents/contents40-1.html</a>.
</div>
<div id="ref-tinytex1" class="csl-entry" role="listitem">
———. 2024. <em>Tinytex: Helper Functions to Install and Maintain TeX Live, and Compile LaTeX Documents</em>. <a href="https://github.com/rstudio/tinytex">https://github.com/rstudio/tinytex</a>.
</div>
<div id="ref-DT" class="csl-entry" role="listitem">
Xie, Yihui, Joe Cheng, and Xianying Tan. 2024. <em>DT: A Wrapper of the JavaScript Library ’DataTables’</em>. <a href="https://CRAN.R-project.org/package=DT">https://CRAN.R-project.org/package=DT</a>.
</div>
<div id="ref-gghighlight" class="csl-entry" role="listitem">
Yutani, Hiroaki. 2024. <em>Gghighlight: Highlight Lines and Points in ’Ggplot2’</em>. <a href="https://yutannihilation.github.io/gghighlight/">https://yutannihilation.github.io/gghighlight/</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
